<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Eric Crandall">

<title>Isolation by Distance and Migration Analysis for Dascyllus aruanus</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>

<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#read-and-convert-data" id="toc-read-and-convert-data" class="nav-link" data-scroll-target="#read-and-convert-data">Read and convert data</a>
  <ul>
  <li><a href="#read-in-the-data" id="toc-read-in-the-data" class="nav-link" data-scroll-target="#read-in-the-data">Read in the data</a></li>
  <li><a href="#test-for-hwe" id="toc-test-for-hwe" class="nav-link" data-scroll-target="#test-for-hwe">Test for HWE</a>
  <ul>
  <li><a href="#drop-some-loci" id="toc-drop-some-loci" class="nav-link" data-scroll-target="#drop-some-loci">Drop Some Loci</a></li>
  </ul></li>
  <li><a href="#split-by-archipelago" id="toc-split-by-archipelago" class="nav-link" data-scroll-target="#split-by-archipelago">Split by Archipelago</a></li>
  </ul></li>
  <li><a href="#calculate-effective-size" id="toc-calculate-effective-size" class="nav-link" data-scroll-target="#calculate-effective-size">Calculate Effective Size</a>
  <ul>
  <li><a href="#interpreting-ld-effective-size" id="toc-interpreting-ld-effective-size" class="nav-link" data-scroll-target="#interpreting-ld-effective-size">Interpreting LD Effective Size</a></li>
  </ul></li>
  <li><a href="#new-caledonia" id="toc-new-caledonia" class="nav-link" data-scroll-target="#new-caledonia">New Caledonia</a>
  <ul>
  <li><a href="#traditional-isolation-by-distance-method" id="toc-traditional-isolation-by-distance-method" class="nav-link" data-scroll-target="#traditional-isolation-by-distance-method">Traditional Isolation by Distance Method</a>
  <ul>
  <li><a href="#calculate-distance-matrices" id="toc-calculate-distance-matrices" class="nav-link" data-scroll-target="#calculate-distance-matrices">Calculate distance matrices</a>
  <ul class="collapse">
  <li><a href="#genetic-distances" id="toc-genetic-distances" class="nav-link" data-scroll-target="#genetic-distances">Genetic Distances</a></li>
  <li><a href="#geographic-distances" id="toc-geographic-distances" class="nav-link" data-scroll-target="#geographic-distances">Geographic Distances</a></li>
  </ul></li>
  <li><a href="#calculate-linear-model" id="toc-calculate-linear-model" class="nav-link" data-scroll-target="#calculate-linear-model">Calculate linear model</a></li>
  <li><a href="#calculate-effective-size-1" id="toc-calculate-effective-size-1" class="nav-link" data-scroll-target="#calculate-effective-size-1">Calculate Effective Size</a></li>
  <li><a href="#ne-vs.-sampling-window" id="toc-ne-vs.-sampling-window" class="nav-link" data-scroll-target="#ne-vs.-sampling-window">Ne vs.&nbsp;Sampling Window</a></li>
  <li><a href="#calculate-effective-density" id="toc-calculate-effective-density" class="nav-link" data-scroll-target="#calculate-effective-density">Calculate Effective Density</a></li>
  </ul></li>
  <li><a href="#migraine-method" id="toc-migraine-method" class="nav-link" data-scroll-target="#migraine-method">MigraiNe Method</a>
  <ul>
  <li><a href="#running-migraine" id="toc-running-migraine" class="nav-link" data-scroll-target="#running-migraine">Running MigraiNe</a></li>
  <li><a href="#create-dispersal-kernels" id="toc-create-dispersal-kernels" class="nav-link" data-scroll-target="#create-dispersal-kernels">Create Dispersal Kernels</a>
  <ul class="collapse">
  <li><a href="#sigma-estimates" id="toc-sigma-estimates" class="nav-link" data-scroll-target="#sigma-estimates">Sigma estimates</a></li>
  <li><a href="#from-neighborhood-size" id="toc-from-neighborhood-size" class="nav-link" data-scroll-target="#from-neighborhood-size">From Neighborhood Size</a></li>
  <li><a href="#from-sigma2" id="toc-from-sigma2" class="nav-link" data-scroll-target="#from-sigma2">From Sigma^2</a></li>
  </ul></li>
  <li><a href="#confidence-intervals" id="toc-confidence-intervals" class="nav-link" data-scroll-target="#confidence-intervals">Confidence Intervals</a>
  <ul class="collapse">
  <li><a href="#error-in-effective-size" id="toc-error-in-effective-size" class="nav-link" data-scroll-target="#error-in-effective-size">Error in Effective Size</a></li>
  <li><a href="#error-in-effective-density" id="toc-error-in-effective-density" class="nav-link" data-scroll-target="#error-in-effective-density">Error in Effective Density</a></li>
  <li><a href="#error-in-neighborhood-size" id="toc-error-in-neighborhood-size" class="nav-link" data-scroll-target="#error-in-neighborhood-size">Error in Neighborhood Size</a></li>
  <li><a href="#plot-dispersal-kernels" id="toc-plot-dispersal-kernels" class="nav-link" data-scroll-target="#plot-dispersal-kernels">Plot Dispersal Kernels</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  <li><a href="#fiji" id="toc-fiji" class="nav-link" data-scroll-target="#fiji">Fiji</a>
  <ul>
  <li><a href="#traditional-isolation-by-distance-method-1" id="toc-traditional-isolation-by-distance-method-1" class="nav-link" data-scroll-target="#traditional-isolation-by-distance-method-1">Traditional Isolation by Distance Method</a>
  <ul>
  <li><a href="#calculate-distance-matrices-1" id="toc-calculate-distance-matrices-1" class="nav-link" data-scroll-target="#calculate-distance-matrices-1">Calculate distance matrices</a>
  <ul class="collapse">
  <li><a href="#genetic-distances-1" id="toc-genetic-distances-1" class="nav-link" data-scroll-target="#genetic-distances-1">Genetic Distances</a>
  <ul class="collapse">
  <li><a href="#geographic-distances-1" id="toc-geographic-distances-1" class="nav-link" data-scroll-target="#geographic-distances-1">Geographic Distances</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#calculate-linear-model-1" id="toc-calculate-linear-model-1" class="nav-link" data-scroll-target="#calculate-linear-model-1">Calculate linear model</a></li>
  <li><a href="#calculate-effective-size-2" id="toc-calculate-effective-size-2" class="nav-link" data-scroll-target="#calculate-effective-size-2">Calculate Effective Size</a></li>
  <li><a href="#ne-vs.-sampling-window-1" id="toc-ne-vs.-sampling-window-1" class="nav-link" data-scroll-target="#ne-vs.-sampling-window-1">Ne vs.&nbsp;Sampling Window</a></li>
  <li><a href="#calculate-effective-density-1" id="toc-calculate-effective-density-1" class="nav-link" data-scroll-target="#calculate-effective-density-1">Calculate Effective Density</a></li>
  <li><a href="#calculate-sigma" id="toc-calculate-sigma" class="nav-link" data-scroll-target="#calculate-sigma">Calculate sigma</a></li>
  </ul></li>
  <li><a href="#migraine-method-1" id="toc-migraine-method-1" class="nav-link" data-scroll-target="#migraine-method-1">MigraiNe Method</a>
  <ul>
  <li><a href="#running-migraine-1" id="toc-running-migraine-1" class="nav-link" data-scroll-target="#running-migraine-1">Running MigraiNe</a></li>
  <li><a href="#create-dispersal-kernels-1" id="toc-create-dispersal-kernels-1" class="nav-link" data-scroll-target="#create-dispersal-kernels-1">Create Dispersal Kernels</a></li>
  <li><a href="#confidence-intervals-1" id="toc-confidence-intervals-1" class="nav-link" data-scroll-target="#confidence-intervals-1">Confidence Intervals</a>
  <ul class="collapse">
  <li><a href="#error-in-effective-size-1" id="toc-error-in-effective-size-1" class="nav-link" data-scroll-target="#error-in-effective-size-1">Error in Effective Size</a></li>
  <li><a href="#error-in-effective-density-1" id="toc-error-in-effective-density-1" class="nav-link" data-scroll-target="#error-in-effective-density-1">Error in Effective Density</a></li>
  <li><a href="#error-in-slope" id="toc-error-in-slope" class="nav-link" data-scroll-target="#error-in-slope">Error in Slope</a></li>
  <li><a href="#error-in-neighborhood-size-1" id="toc-error-in-neighborhood-size-1" class="nav-link" data-scroll-target="#error-in-neighborhood-size-1">Error in Neighborhood Size</a></li>
  <li><a href="#plot-dispersal-kernels-1" id="toc-plot-dispersal-kernels-1" class="nav-link" data-scroll-target="#plot-dispersal-kernels-1">Plot Dispersal Kernels</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  <li><a href="#society-islands" id="toc-society-islands" class="nav-link" data-scroll-target="#society-islands">Society Islands</a>
  <ul>
  <li><a href="#traditional-isolation-by-distance-method-2" id="toc-traditional-isolation-by-distance-method-2" class="nav-link" data-scroll-target="#traditional-isolation-by-distance-method-2">Traditional Isolation by Distance Method</a>
  <ul>
  <li><a href="#calculate-distance-matrices-2" id="toc-calculate-distance-matrices-2" class="nav-link" data-scroll-target="#calculate-distance-matrices-2">Calculate distance matrices</a>
  <ul class="collapse">
  <li><a href="#genetic-distances-2" id="toc-genetic-distances-2" class="nav-link" data-scroll-target="#genetic-distances-2">Genetic Distances</a>
  <ul class="collapse">
  <li><a href="#geographic-distances-2" id="toc-geographic-distances-2" class="nav-link" data-scroll-target="#geographic-distances-2">Geographic Distances</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#calculate-linear-model-2" id="toc-calculate-linear-model-2" class="nav-link" data-scroll-target="#calculate-linear-model-2">Calculate linear model</a></li>
  <li><a href="#calculate-effective-size-3" id="toc-calculate-effective-size-3" class="nav-link" data-scroll-target="#calculate-effective-size-3">Calculate Effective Size</a></li>
  <li><a href="#ne-vs.-sampling-window-2" id="toc-ne-vs.-sampling-window-2" class="nav-link" data-scroll-target="#ne-vs.-sampling-window-2">Ne vs.&nbsp;Sampling Window</a></li>
  <li><a href="#calculate-effective-density-2" id="toc-calculate-effective-density-2" class="nav-link" data-scroll-target="#calculate-effective-density-2">Calculate Effective Density</a></li>
  <li><a href="#calculate-sigma-1" id="toc-calculate-sigma-1" class="nav-link" data-scroll-target="#calculate-sigma-1">Calculate sigma</a></li>
  </ul></li>
  <li><a href="#migraine-method-2" id="toc-migraine-method-2" class="nav-link" data-scroll-target="#migraine-method-2">MigraiNe Method</a>
  <ul>
  <li><a href="#running-migraine-2" id="toc-running-migraine-2" class="nav-link" data-scroll-target="#running-migraine-2">Running MigraiNe</a></li>
  <li><a href="#create-dispersal-kernels-2" id="toc-create-dispersal-kernels-2" class="nav-link" data-scroll-target="#create-dispersal-kernels-2">Create Dispersal Kernels</a>
  <ul class="collapse">
  <li><a href="#sigma-estimates-1" id="toc-sigma-estimates-1" class="nav-link" data-scroll-target="#sigma-estimates-1">Sigma estimates</a></li>
  </ul></li>
  <li><a href="#confidence-intervals-2" id="toc-confidence-intervals-2" class="nav-link" data-scroll-target="#confidence-intervals-2">Confidence Intervals</a>
  <ul class="collapse">
  <li><a href="#error-in-ne" id="toc-error-in-ne" class="nav-link" data-scroll-target="#error-in-ne">Error in Ne</a></li>
  <li><a href="#error-in-effective-density-2" id="toc-error-in-effective-density-2" class="nav-link" data-scroll-target="#error-in-effective-density-2">Error in Effective Density</a></li>
  <li><a href="#error-in-slope-1" id="toc-error-in-slope-1" class="nav-link" data-scroll-target="#error-in-slope-1">Error in Slope</a></li>
  <li><a href="#error-in-neighborhood-size-2" id="toc-error-in-neighborhood-size-2" class="nav-link" data-scroll-target="#error-in-neighborhood-size-2">Error in Neighborhood Size</a></li>
  <li><a href="#plot-dispersal-kernels-2" id="toc-plot-dispersal-kernels-2" class="nav-link" data-scroll-target="#plot-dispersal-kernels-2">Plot Dispersal Kernels</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  <li><a href="#comparing-across-archipelagos" id="toc-comparing-across-archipelagos" class="nav-link" data-scroll-target="#comparing-across-archipelagos">Comparing Across Archipelagos</a>
  <ul>
  <li><a href="#make-figures-for-the-proposal" id="toc-make-figures-for-the-proposal" class="nav-link" data-scroll-target="#make-figures-for-the-proposal">Make figures for the proposal</a></li>
  </ul></li>
  <li><a href="#treemix" id="toc-treemix" class="nav-link" data-scroll-target="#treemix">Treemix</a>
  <ul>
  <li><a href="#convert-data" id="toc-convert-data" class="nav-link" data-scroll-target="#convert-data">Convert data</a></li>
  <li><a href="#explore-treemix" id="toc-explore-treemix" class="nav-link" data-scroll-target="#explore-treemix">Explore Treemix</a>
  <ul>
  <li><a href="#plot" id="toc-plot" class="nav-link" data-scroll-target="#plot">Plot</a></li>
  </ul></li>
  <li><a href="#how-many-migration-events" id="toc-how-many-migration-events" class="nav-link" data-scroll-target="#how-many-migration-events">How many migration events?</a>
  <ul>
  <li><a href="#run-treemix-at-a-variety-of-m" id="toc-run-treemix-at-a-variety-of-m" class="nav-link" data-scroll-target="#run-treemix-at-a-variety-of-m">Run Treemix at a variety of m</a></li>
  <li><a href="#optm" id="toc-optm" class="nav-link" data-scroll-target="#optm">OptM</a></li>
  <li><a href="#plot-1" id="toc-plot-1" class="nav-link" data-scroll-target="#plot-1">Plot</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#migrate-n" id="toc-migrate-n" class="nav-link" data-scroll-target="#migrate-n">Migrate-n</a>
  <ul>
  <li><a href="#setup-1" id="toc-setup-1" class="nav-link" data-scroll-target="#setup-1">Setup</a></li>
  <li><a href="#initial-runs" id="toc-initial-runs" class="nav-link" data-scroll-target="#initial-runs">Initial Runs</a></li>
  <li><a href="#final-runs" id="toc-final-runs" class="nav-link" data-scroll-target="#final-runs">Final Runs</a>
  <ul>
  <li><a href="#models" id="toc-models" class="nav-link" data-scroll-target="#models">Models</a></li>
  <li><a href="#run-it" id="toc-run-it" class="nav-link" data-scroll-target="#run-it">Run It</a>
  <ul class="collapse">
  <li><a href="#bash-script" id="toc-bash-script" class="nav-link" data-scroll-target="#bash-script">Bash Script</a></li>
  <li><a href="#copy-it-down" id="toc-copy-it-down" class="nav-link" data-scroll-target="#copy-it-down">Copy It Down</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#function-library" id="toc-function-library" class="nav-link" data-scroll-target="#function-library">Function library</a></li>
  <li><a href="#model-marginal-likelihoods" id="toc-model-marginal-likelihoods" class="nav-link" data-scroll-target="#model-marginal-likelihoods">Model Marginal Likelihoods</a>
  <ul class="collapse">
  <li><a href="#t-test" id="toc-t-test" class="nav-link" data-scroll-target="#t-test">T-Test</a></li>
  <li><a href="#model-selection-figures" id="toc-model-selection-figures" class="nav-link" data-scroll-target="#model-selection-figures">Model Selection Figures</a></li>
  </ul></li>
  <li><a href="#parameter-estimates" id="toc-parameter-estimates" class="nav-link" data-scroll-target="#parameter-estimates">Parameter Estimates</a>
  <ul class="collapse">
  <li><a href="#priors" id="toc-priors" class="nav-link" data-scroll-target="#priors">Priors</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Isolation by Distance and Migration Analysis for Dascyllus aruanus</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Eric Crandall </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="setup" class="level1">
<h1>Setup</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(adegenet)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gdistance)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pegas)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hierfstat)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rgdal)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(graph4lg)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(coda)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(strataG)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnvStats)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genepopedit)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(perm)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"IBD_functions.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Cecile Fauvelot has kindly shared a giant (n=1358) microsatellite dataset for <em>Dascyllus aruanus</em>, the white tailed or humbug damsel. These have been sampled from Papua New Guinea, New Caledonia, Fiji and the Society Islands/Tuamotus.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figures/Humbug_dascyllus.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">White Tailed Damsel</figcaption><p></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figures/Da_samplesites.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Sample Sites</figcaption><p></p>
</figure>
</div>
</section>
<section id="read-and-convert-data" class="level1">
<h1>Read and convert data</h1>
<p>Had to do a little conversion in bbedit to ready the Excel formatted data for input into R</p>
<section id="read-in-the-data" class="level2">
<h2 class="anchored" data-anchor-id="read-in-the-data">Read in the data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>Daruanus.gen <span class="ot">&lt;-</span> <span class="fu">read.genepop</span>(<span class="st">"data/Daruanus_Fauvelot.gen"</span>, <span class="at">ncode =</span> 3L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
 Converting data from a Genepop .gen file to a genind object... 


File description:  Dascyllus aruanus all populations 1358 individuals 

...done.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rename the populations to just the text values</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>Daruanus.gen<span class="sc">@</span>pop <span class="ot">&lt;-</span> Daruanus.gen<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">str_extract</span>(<span class="st">"[A-Za-z]+"</span>) <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the locality information</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>Daruanus.sites <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"data/Daruanus_sites.xlsx"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>Daruanus.gen<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">tibble</span>(<span class="at">pop =</span> <span class="fu">as.character</span>(.)) <span class="sc">%&gt;%</span> <span class="fu">count</span>(pop)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["pop"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Go","2":"55"},{"1":"Hie","2":"47"},{"1":"Lar","2":"51"},{"1":"Lion","2":"49"},{"1":"Lolo","2":"9"},{"1":"Manu","2":"14"},{"1":"Mara","2":"52"},{"1":"Mata","2":"48"},{"1":"MBO","2":"50"},{"1":"Mo","2":"264"},{"1":"Muaiv","2":"54"},{"1":"Nadi","2":"51"},{"1":"Puna","2":"25"},{"1":"QBW","2":"54"},{"1":"Rangi","2":"43"},{"1":"Suva","2":"56"},{"1":"Taga","2":"52"},{"1":"Tahaa","2":"36"},{"1":"Taka","2":"30"},{"1":"Tave","2":"50"},{"1":"Tem","2":"64"},{"1":"Ten","2":"52"},{"1":"Tetia","2":"52"},{"1":"Tote","2":"53"},{"1":"Yanu","2":"47"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="test-for-hwe" class="level2">
<h2 class="anchored" data-anchor-id="test-for-hwe">Test for HWE</h2>
<p>Test for HWE. First look at the number of populations that have HWE departures for each locus. Then look at the distribution of p-values following <span class="citation" data-cites="waplesTestingHardyWeinberg2014">Waples (<a href="#ref-waplesTestingHardyWeinberg2014" role="doc-biblioref">2014</a>)</span>. A flat distribution is fine, but enrichment for low p-values suggests that the locus is not globally at HWE.</p>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">Code</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Hardy-Weinberg Stats</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">P-Values Plot</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>separated_pops <span class="ot">&lt;-</span> <span class="fu">seppop</span>(Daruanus.gen)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># perform HWE test</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>hw <span class="ot">&lt;-</span> <span class="fu">map</span>(separated_pops, hw.test)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>hw2 <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,hw) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"locus"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                <span class="fu">group_by</span>(locus) <span class="sc">%&gt;%</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarize</span>(<span class="at">outofhwe =</span> <span class="fu">length</span>(<span class="fu">which</span>(Pr.exact <span class="sc">&lt;</span> <span class="fl">0.05</span>)), </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">outofhwe_prop =</span> <span class="fu">length</span>(<span class="fu">which</span>(Pr.exact <span class="sc">&lt;</span> <span class="fl">0.05</span>))<span class="sc">/</span><span class="fu">length</span>(hw),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">meanp =</span> <span class="fu">mean</span>(Pr.exact))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>pvalues<span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,hw) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"locus"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>          <span class="fu">group_by</span>(locus) <span class="sc">%&gt;%</span> </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Pr.exact)) <span class="sc">+</span> </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">facet_wrap</span>(<span class="sc">~</span>locus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>hw2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["locus"],"name":[1],"type":["chr"],"align":["left"]},{"label":["outofhwe"],"name":[2],"type":["int"],"align":["right"]},{"label":["outofhwe_prop"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["meanp"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"304","2":"2","3":"0.08","4":"0.47184"},{"1":"331","2":"1","3":"0.04","4":"0.45820"},{"1":"360","2":"0","3":"0.00","4":"0.55168"},{"1":"408","2":"19","3":"0.76","4":"0.03880"},{"1":"494","2":"15","3":"0.60","4":"0.16492"},{"1":"523","2":"4","3":"0.16","4":"0.32964"},{"1":"542","2":"4","3":"0.16","4":"0.42476"},{"1":"565","2":"1","3":"0.04","4":"0.32932"},{"1":"589","2":"3","3":"0.12","4":"0.43324"},{"1":"590","2":"3","3":"0.12","4":"0.46220"},{"1":"593","2":"5","3":"0.20","4":"0.32276"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pvalues</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</div>
<p>Itâ€™s pretty clear we need to zap loci 408, 494, and 593. 565 is a little weird, but letâ€™s keep it.</p>
<section id="drop-some-loci" class="level3">
<h3 class="anchored" data-anchor-id="drop-some-loci">Drop Some Loci</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">locNames</span>(Daruanus.gen)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "304" "408" "331" "589" "494" "523" "360" "565" "542" "593" "590"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Daruanus.gen <span class="ot">&lt;-</span> Daruanus.gen[loc<span class="ot">=</span><span class="sc">-</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">10</span>)]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#convert to strataG gtypes</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>Daruanus.gtypes <span class="ot">&lt;-</span> <span class="fu">genind2gtypes</span>(Daruanus.gen)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.gen, output = "Daruanus/Daruanus_All_8locus.txt")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="split-by-archipelago" class="level2">
<h2 class="anchored" data-anchor-id="split-by-archipelago">Split by Archipelago</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#split them again</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>separated_pops <span class="ot">&lt;-</span> <span class="fu">seppop</span>(Daruanus.gen)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#split off the Fiji samples</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>fijipops <span class="ot">&lt;-</span> Daruanus.sites <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Region<span class="sc">==</span><span class="st">"Fiji"</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji <span class="ot">&lt;-</span> <span class="fu">repool</span>(separated_pops[fijipops<span class="sc">$</span>Abbr])</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#split off the NC samples</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>NCpops <span class="ot">&lt;-</span> Daruanus.sites <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Region<span class="sc">==</span><span class="st">"NC"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Abbr<span class="sc">!=</span><span class="st">"Hie"</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>Daruanus.NC <span class="ot">&lt;-</span> <span class="fu">repool</span>(separated_pops[NCpops<span class="sc">$</span>Abbr])</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co">#split off the FP samples</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>FPpops <span class="ot">&lt;-</span> Daruanus.sites <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Region <span class="sc">==</span> <span class="st">"FP"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Abbr <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Puna"</span>,<span class="st">"Tetia"</span>, <span class="st">"Tem"</span>,<span class="st">"Mo"</span>,<span class="st">"Tahaa"</span>))</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>Daruanus.FP <span class="ot">&lt;-</span> <span class="fu">repool</span>(separated_pops[FPpops<span class="sc">$</span>Abbr])</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.NC,output = "Daruanus/NC/Daruanus_NC.txt")</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.Fiji,output = "Daruanus/Fiji/Daruanus_Fiji.txt")</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.FP,output = "Daruanus/FP/Daruanus_FP.txt")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="calculate-effective-size" class="level1">
<h1>Calculate Effective Size</h1>
<p><span class="citation" data-cites="neelEstimationEffectivePopulation2013">Neel et al. (<a href="#ref-neelEstimationEffectivePopulation2013" role="doc-biblioref">2013</a>)</span> say:</p>
<blockquote class="blockquote">
<p>Our results show that the LD method provides a good approximation of the NS as long as the scale of sampling is commensurate with the scale of local breeding.</p>
</blockquote>
<p>Treating the whole dataset as one population yields <span class="math inline">\(N_e\)</span> of -27423.9 (in other words too large) at pcrit of 0.02. But the problem is determining the size of the genetic neighborhood, because Fauvelot et al.â€™s samples were not as regularly spaced as Dâ€™Aloiaâ€™s.</p>
<p>Going to use the LD method as most recently discussed by <span class="citation" data-cites="waplesLinkageDisequilibriumEstimates2010">Waples and Do (<a href="#ref-waplesLinkageDisequilibriumEstimates2010" role="doc-biblioref">2010</a>)</span>, and implemented in NeEstimator v2 <span class="citation" data-cites="doNeEstimatorV2Reimplementation2014">(<a href="#ref-doNeEstimatorV2Reimplementation2014" role="doc-biblioref">Do et al. 2014</a>)</span>. Iâ€™m going to remove alleles following Waples and Doâ€™s rule of thumb. Parameter settings are in <code>"Daruanus/Ne_estimator/"</code> Best to run this from the command line actually. I am having it calculate <span class="math inline">\(N_b\)</span> (number of breeders) for monogamy, as the protogynous mating system of <em>Dascyllus aruanus</em> seems closer to monogamy than random mating. Also had to edit the table output of NeEstimator to make it legible to R because it had lots of spaces and empty cells ðŸ‘Ž</p>
<p>Hereâ€™s the parameter file for LDNe</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/Applications/NeEstimator/Ne2-1M</span> i:/Users/eric/github/IBD_Kernels/Daruanus/NeEstimator/info o:/Users/eric/github/IBD_Kernels/Daruanus/NeEstimator/option</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="ex">INFO</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>   <span class="pp">*</span> A number = sum of method<span class="er">(</span><span class="ex">s</span><span class="kw">)</span> <span class="ex">to</span> run: LD<span class="er">(</span><span class="ex">=1</span><span class="kw">)</span><span class="ex">,</span> Het<span class="er">(</span><span class="ex">=2</span><span class="kw">)</span><span class="ex">,</span> Coan<span class="er">(</span><span class="ex">=4</span><span class="kw">)</span><span class="ex">,</span> Temporal<span class="er">(</span><span class="ex">=8</span><span class="kw">)</span><span class="bu">.</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/eric/github/IBD_Kernels/Daruanus/</span>  <span class="pp">*</span> Input Directory</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="ex">Daruanus_All_8locus.txt</span> <span class="pp">*</span> Input file name</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="ex">2</span>    <span class="pp">*</span> 1 = FSTAT format, 2 = GENEPOP format</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/eric/github/IBD_Kernels/Daruanus/NeEstimator/</span>  <span class="pp">*</span> Output Directory</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="ex">Daruanus_LDNe.txt</span>  <span class="pp">*</span> Output file name <span class="er">(</span><span class="ex">put</span> asterisk adjacent to the name to append<span class="kw">)</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="ex">6</span>    <span class="pp">*</span> Number of critical values, added 1 if a run by rejecting only singleton alleles is included</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span> 0.01 0.02 0.03 0.04 0.05    <span class="pp">*</span> Critical values, a special value <span class="st">'1'</span> is for rejecting only singleton alleles</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>   <span class="pp">*</span> 0: Random mating, 1: Monogamy <span class="er">(</span><span class="ex">LD</span> method<span class="kw">)</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="ex">OPTION</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>  1  5  1  <span class="pp">*</span> First number = sum of method<span class="er">(</span><span class="ex">s</span><span class="kw">)</span> <span class="ex">to</span> have extra output: LD<span class="er">(</span><span class="ex">=1</span><span class="kw">)</span><span class="ex">,</span> Het<span class="er">(</span><span class="ex">=2</span><span class="kw">)</span><span class="ex">,</span> Coan<span class="er">(</span><span class="ex">=4</span><span class="kw">)</span><span class="ex">,</span> Temporal<span class="er">(</span><span class="ex">=8</span><span class="kw">)</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>  <span class="pp">*</span> Maximum individuals/pop. If 0: no limit</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>  <span class="pp">*</span> First entry n1 = 0: No Freq output. If n1 = <span class="at">-1:</span> Freq. output up to population 50. Two entries n1, n2 with n1 <span class="op">&lt;</span>= n2: Freq output for populations from n1 to n2. Max. populations to have Freq output is set at 50</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>  <span class="pp">*</span> For Burrow output file <span class="er">(</span><span class="ex">up</span> to 50 populations can have output<span class="kw">)</span><span class="bu">.</span> See remark below</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>  <span class="pp">*</span> Parameter CI: 1 for Yes, 0 for No</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>  <span class="pp">*</span> Jackknife CI: 1 for Yes, 0 for No</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>  <span class="pp">*</span> Up to population, or range of populations to run <span class="er">(</span><span class="cf">if</span> <span class="ex">2</span> entries<span class="kw">)</span><span class="bu">.</span> If first entry = 0: no restriction</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>  <span class="pp">*</span> All loci are accepted</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>  <span class="pp">*</span> Enter 1: A file is created to document missing data if there are any in input file. Enter 0: no file created</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>   <span class="pp">*</span> Line for chromosomes/loci option and file</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>I implemented a filtering step that follows <span class="citation" data-cites="waplesLinkageDisequilibriumEstimates2010">Waples and Do (<a href="#ref-waplesLinkageDisequilibriumEstimates2010" role="doc-biblioref">2010</a>)</span> :</p>
<blockquote class="blockquote">
<p>For S &gt; 100: choose Pcrit = 0.01 For S &gt; 25: choose Pcrit = 0.02.<br>
For S &lt; 25: choose so that 1/(2S) &lt; Pcrit &lt; 1/S.</p>
</blockquote>
<p>And I am only keeping estimates from samples with n &gt;= 10.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"./NeEstimator/Daruanus_LDNe_xLD.txt"</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering based on rule of thumb from Waples &amp; Do</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>Ne_estimates_f <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates, <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Population =</span> <span class="fu">str_replace</span>(Population,<span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">:(</span><span class="sc">\\</span><span class="st">w+)_[</span><span class="sc">\\</span><span class="st">w-]+"</span>, </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">replacement =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span> ))</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>Ne_estimates_f</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Population"],"name":[1],"type":["chr"],"align":["left"]},{"label":["SampSize"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pcrit"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["WeightedMean"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["IndAlleles"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["r2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["ExpR2"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Ne"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["ParametricLow"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["ParametricHigh"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["JackknifeLow"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["JackknifeHigh"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["EffDF"],"name":[13],"type":["dbl"],"align":["right"]}],"data":[{"1":"Go","2":"55","3":"0.02","4":"55.0","5":"4331","6":"0.018751","7":"0.019236","8":"-1377.4","9":"2013.9","10":"NA","11":"1282.0","12":"NA","13":"2900"},{"1":"Hie","2":"47","3":"0.02","4":"47.0","5":"7358","6":"0.022882","7":"0.022721","8":"4137.0","9":"722.9","10":"NA","11":"606.1","12":"NA","13":"4898"},{"1":"Lar","2":"51","3":"0.02","4":"51.0","5":"3484","6":"0.020885","7":"0.020834","8":"13190.4","9":"622.0","10":"NA","11":"480.6","12":"NA","13":"2083"},{"1":"Lion","2":"49","3":"0.02","4":"49.0","5":"6502","6":"0.021510","7":"0.021737","8":"-2941.3","9":"1250.1","10":"NA","11":"954.9","12":"NA","13":"4445"},{"1":"Manu","2":"14","3":"0.04","4":"14.0","5":"1005","6":"0.089004","7":"0.089239","8":"-2636.6","9":"74.2","10":"NA","11":"63.4","12":"NA","13":"763"},{"1":"Mara","2":"52","3":"0.02","4":"52.0","5":"4081","6":"0.020336","7":"0.020411","8":"-8977.0","9":"793.1","10":"NA","11":"553.8","12":"NA","13":"2151"},{"1":"Mata","2":"48","3":"0.02","4":"48.0","5":"4191","6":"0.021966","7":"0.022218","8":"-2652.5","9":"923.0","10":"NA","11":"634.0","12":"NA","13":"2397"},{"1":"MBO","2":"50","3":"0.02","4":"50.0","5":"8333","6":"0.021633","7":"0.021276","8":"1863.7","9":"644.9","10":"NA","11":"455.8","12":"NA","13":"3220"},{"1":"Mo","2":"264","3":"0.01","4":"264.0","5":"4260","6":"0.003764","7":"0.003834","8":"-9543.0","9":"6998.6","10":"NA","11":"3618.6","12":"NA","13":"1862"},{"1":"Muaiv","2":"54","3":"0.02","4":"54.0","5":"2752","6":"0.019327","7":"0.019612","8":"-2340.3","9":"854.1","10":"NA","11":"349.2","12":"NA","13":"710"},{"1":"Nadi","2":"51","3":"0.02","4":"50.5","5":"2608","6":"0.021235","7":"0.021033","8":"3305.3","9":"472.5","10":"NA","11":"328.5","12":"NA","13":"1195"},{"1":"Puna","2":"25","3":"0.02","4":"25.0","5":"3421","6":"0.045698","7":"0.045184","8":"1199.8","9":"221.8","10":"NA","11":"166.1","12":"NA","13":"1778"},{"1":"QBW","2":"54","3":"0.02","4":"53.7","5":"4489","6":"0.024834","7":"0.019734","8":"128.0","9":"105.5","10":"159.7","11":"68.7","12":"381.7","13":"336"},{"1":"Rangi","2":"43","3":"0.02","4":"43.0","5":"2845","6":"0.025128","7":"0.024981","8":"4541.9","9":"440.1","10":"NA","11":"297.9","12":"NA","13":"1275"},{"1":"Suva","2":"56","3":"0.02","4":"56.0","5":"2222","6":"0.019452","7":"0.018874","8":"1150.6","9":"373.0","10":"NA","11":"291.1","12":"NA","13":"1153"},{"1":"Taga","2":"52","3":"0.02","4":"52.0","5":"2346","6":"0.020192","7":"0.020411","8":"-3057.6","9":"671.3","10":"NA","11":"383.8","12":"NA","13":"953"},{"1":"Tahaa","2":"36","3":"0.02","4":"36.0","5":"2525","6":"0.032537","7":"0.030239","8":"287.4","9":"157.1","10":"1156.0","11":"127.8","12":"NA","13":"1169"},{"1":"Taka","2":"30","3":"0.02","4":"29.8","5":"1585","6":"0.038172","7":"0.037159","8":"607.9","9":"159.8","10":"NA","11":"137.7","12":"NA","13":"1098"},{"1":"Tave","2":"50","3":"0.02","4":"50.0","5":"3464","6":"0.022223","7":"0.021276","8":"701.1","9":"325.3","10":"NA","11":"293.8","12":"NA","13":"2443"},{"1":"Tem","2":"64","3":"0.02","4":"64.0","5":"2593","6":"0.016139","7":"0.016404","8":"-2518.0","9":"1021.7","10":"NA","11":"770.0","12":"NA","13":"1742"},{"1":"Ten","2":"52","3":"0.02","4":"52.0","5":"3724","6":"0.020779","7":"0.020411","8":"1805.7","9":"492.7","10":"NA","11":"382.7","12":"NA","13":"1971"},{"1":"Tetia","2":"52","3":"0.02","4":"52.0","5":"2208","6":"0.021145","7":"0.020411","8":"905.4","9":"324.2","10":"NA","11":"262.5","12":"NA","13":"1225"},{"1":"Tote","2":"53","3":"0.02","4":"53.0","5":"3833","6":"0.020282","7":"0.020004","8":"2395.4","9":"544.8","10":"NA","11":"409.5","12":"NA","13":"1940"},{"1":"Yanu","2":"47","3":"0.02","4":"47.0","5":"3459","6":"0.022957","7":"0.022721","8":"2822.5","9":"488.4","10":"NA","11":"426.0","12":"NA","13":"2532"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<section id="interpreting-ld-effective-size" class="level3">
<h3 class="anchored" data-anchor-id="interpreting-ld-effective-size">Interpreting LD Effective Size</h3>
<p>These estimates are true Ne estimates because these samples were taken across the age structure of the population. So there wonâ€™t be any conversion from Nb to Ne.</p>
<p>Cecile says:</p>
<blockquote class="blockquote">
<p>For Fiji and NC, multiple individuals at a coral colony were indeed sampled as we used clove oil around coral colonies covered by a plastic bagâ€¦ so yes too, across age structure. I do not have the size of individuals sampled</p>
</blockquote>
<p><span class="citation" data-cites="waplesEffectsOverlappingGenerations2014">Waples, Antao, and Luikart (<a href="#ref-waplesEffectsOverlappingGenerations2014" role="doc-biblioref">2014</a>)</span> say:</p>
<blockquote class="blockquote">
<p>Our empirical results provide some qualified support for the hypothesis (Waples and Do 2010) that a sample that includes as many cohorts as there are in a generation should produce an estimate approximately equal to Neâ€¦.All estimates based on random samples of adults were smaller than true Ne â€¦, but there was a tendency for the bias to be less when the number of cohorts included in the adult sample corresponded more closely to the generation length.</p>
</blockquote>
<p><em>Dascyllus aruanus</em> strikes me as one of those species where youâ€™ll have as many cohorts as generations, although protogyny kind of messes with this. In any case, we can expect our estimates of Ne (and thus De) to be downwardly biased, and therefore our estimates of <span class="math inline">\(\sigma\)</span> to be upwardly biased, by hopefully less than 10%?</p>
</section>
</section>
<section id="new-caledonia" class="level1">
<h1>New Caledonia</h1>
<section id="traditional-isolation-by-distance-method" class="level2">
<h2 class="anchored" data-anchor-id="traditional-isolation-by-distance-method">Traditional Isolation by Distance Method</h2>
<p>Based on the OG <span class="citation" data-cites="roussetGeneticDifferentiationEstimation1997">(<a href="#ref-roussetGeneticDifferentiationEstimation1997" role="doc-biblioref">Rousset 1997</a>)</span> estimator from slope of the IBD regression.</p>
<section id="calculate-distance-matrices" class="level3">
<h3 class="anchored" data-anchor-id="calculate-distance-matrices">Calculate distance matrices</h3>
<section id="genetic-distances" class="level4">
<h4 class="anchored" data-anchor-id="genetic-distances">Genetic Distances</h4>
<p>Weir and Cockerhamâ€™s Fst and other basic stats</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>Daruanus.NC.hfst <span class="ot">&lt;-</span> <span class="fu">genind2hierfstat</span>(Daruanus.NC)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>Daruanus.NC.loci <span class="ot">&lt;-</span> <span class="fu">genind2loci</span>(Daruanus.NC)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gen.loci &lt;- genind2loci(gen)</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>stats.NC <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.NC)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>theta.NC <span class="ot">&lt;-</span> <span class="fu">theta.msat</span>(Daruanus.NC.loci)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">#mean theta</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(theta.NC[,<span class="dv">2</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 156.622</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>fst.NC <span class="ot">&lt;-</span> <span class="fu">genet.dist</span>(Daruanus.NC.hfst, <span class="at">method =</span> <span class="st">"WC84"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mean Fis values</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>stats.NC<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Mara"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Ten"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["MBO"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Lar"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Go"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["QBW"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Tote"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0146","2":"0.0300625","3":"0.0232","4":"0.0067125","5":"0.0191","6":"0.0227625","7":"0.003"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>meanFis <span class="ot">&lt;-</span> stats.NC<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>                <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># linearize</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>fst.NC <span class="ot">&lt;-</span> fst.NC<span class="sc">/</span>(<span class="dv">1</span><span class="sc">-</span>fst.NC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="geographic-distances" class="level4">
<h4 class="anchored" data-anchor-id="geographic-distances">Geographic Distances</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate great circle distance</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>gcdists_NC <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="fu">pointDistance</span>(NCpops[,<span class="dv">5</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">lonlat=</span>T)<span class="sc">/</span><span class="dv">1000</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(gcdists_NC, <span class="st">"Labels"</span>) <span class="ot">&lt;-</span> NCpops<span class="sc">$</span>Abbr</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>gcdists.mat_NC <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(gcdists_NC)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(gen.fst), "Daruanus_linearizedFst.csv", row.names = F, quote=F)</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(gcdists), "Daruanus_gcdists.csv", row.names = F, quote=F)</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#pull out a few other distances we'll need</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>neighbordists_NC <span class="ot">&lt;-</span> gcdists.mat_NC[<span class="fu">row</span>(gcdists.mat_NC) <span class="sc">==</span> <span class="fu">col</span>(gcdists.mat_NC) <span class="sc">+</span> <span class="dv">1</span>]</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>distfromP1_NC <span class="ot">&lt;-</span> gcdists.mat_NC[,<span class="dv">1</span>]</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>maxdist_NC <span class="ot">&lt;-</span> <span class="fu">max</span>(gcdists.mat_NC)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>meandists_NC <span class="ot">&lt;-</span> <span class="fu">mean</span>(neighbordists_NC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="calculate-linear-model" class="level3">
<h3 class="anchored" data-anchor-id="calculate-linear-model">Calculate linear model</h3>
<p>First to get the slope <span class="math inline">\(m\)</span> we need to make a simple linear model. I donâ€™t think significance is really important here, but we can calculate that with a Mantel test.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mantel test</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>mantelt <span class="ot">&lt;-</span> <span class="fu">mantel.randtest</span>(fst.NC,gcdists_NC, <span class="at">nrepet =</span> <span class="dv">10000</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>distances <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">distance=</span><span class="fu">as.vector</span>(gcdists_NC),<span class="at">fst=</span><span class="fu">as.vector</span>(fst.NC))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>lmodel_NC <span class="ot">&lt;-</span> <span class="fu">lm</span>(fst <span class="sc">~</span> distance , distances)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>slope_NC <span class="ot">&lt;-</span> <span class="fu">round</span>(lmodel_NC<span class="sc">$</span>coefficients[<span class="dv">2</span>],<span class="dv">7</span>)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>mantelr <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>obs, <span class="dv">2</span>)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>pvalue <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>pvalue, <span class="dv">5</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>lmodel_plot_NC <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(distances,<span class="fu">aes</span>(<span class="at">x=</span>distance,<span class="at">y=</span>fst)) <span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span> </span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">xlab</span>(<span class="st">"Geographic Distance (km)"</span>) <span class="sc">+</span> </span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">ylab</span>(<span class="fu">expression</span>(F[<span class="st">"ST"</span>]<span class="sc">/</span><span class="dv">1</span><span class="sc">-</span>F[<span class="st">"ST"</span>])) <span class="sc">+</span> </span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>             <span class="fu">geom_text</span>(<span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"m ="</span>, slope_NC, </span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"; Mantel r ="</span>, mantelr,</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>                         <span class="st">", p ="</span>, pvalue ), </span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">80</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.002</span>))</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>lmodel_plot_NC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("NC_IBD.pdf", plot=clown_plot,device="pdf", width=7, height=5,units="in")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Yowza. Negative slope! As Cecile had already measured. But the Fst values are really teeny, and thereâ€™s a lot of error in their measurement.</p>
</section>
<section id="calculate-effective-size-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-size-1">Calculate Effective Size</h3>
<p>Take the harmonic mean of the Ne across all pops</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC <span class="ot">&lt;-</span> Ne_estimates_f <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Population <span class="sc">%in%</span> NCpops<span class="sc">$</span>Abbr)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#write_csv(Nb_estimates_f,"NeEstimator/Nb_estimates_recruits_NeTable.csv")</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">8</span>,<span class="dv">11</span>,<span class="dv">12</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Population"],"name":[1],"type":["chr"],"align":["left"]},{"label":["SampSize"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pcrit"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["WeightedMean"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Ne"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["JackknifeLow"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["JackknifeHigh"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"Go","2":"55","3":"0.02","4":"55.0","5":"-1377.4","6":"1282.0","7":"NA"},{"1":"Lar","2":"51","3":"0.02","4":"51.0","5":"13190.4","6":"480.6","7":"NA"},{"1":"Mara","2":"52","3":"0.02","4":"52.0","5":"-8977.0","6":"553.8","7":"NA"},{"1":"MBO","2":"50","3":"0.02","4":"50.0","5":"1863.7","6":"455.8","7":"NA"},{"1":"QBW","2":"54","3":"0.02","4":"53.7","5":"128.0","6":"68.7","7":"381.7"},{"1":"Ten","2":"52","3":"0.02","4":"52.0","5":"1805.7","6":"382.7","7":"NA"},{"1":"Tote","2":"53","3":"0.02","4":"53.0","5":"2395.4","6":"409.5","7":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># harmonic mean of Nb, following Waples and Do 2010</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_NC<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ne-vs.-sampling-window" class="level3">
<h3 class="anchored" data-anchor-id="ne-vs.-sampling-window">Ne vs.&nbsp;Sampling Window</h3>
<p>Letâ€™s cluster the sites by UPGMA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">hclust</span>(gcdists_NC,<span class="st">"average"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Letâ€™s explore Ne and <span class="math inline">\(F_{is}\)</span> when lumping populationsâ€¦ Lump populations that are 10, 20, 40, 100 and 200 km apart.</p>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">10km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">20km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false">40km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-4" role="tab" aria-controls="tabset-2-4" aria-selected="false">100km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-5" role="tab" aria-controls="tabset-2-5" aria-selected="false">200km (all pops as one)</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-6" role="tab" aria-controls="tabset-2-6" aria-selected="false">Figure</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>stats.NC<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Mara"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Ten"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["MBO"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Lar"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Go"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["QBW"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Tote"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0146","2":"0.0300625","3":"0.0232","4":"0.0067125","5":"0.0191","6":"0.0227625","7":"0.003"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.10</span>km <span class="ot">&lt;-</span> Daruanus.NC</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.NC<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fct_collapse</span>(</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">center =</span> <span class="fu">c</span>(<span class="st">"Go"</span>,<span class="st">"Lar"</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.10</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.NC<span class="fl">.10</span>km)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.10</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Mara"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Ten"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["MBO"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["center"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["QBW"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["Tote"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0146","2":"0.0300625","3":"0.0232","4":"0.012425","5":"0.0227625","6":"0.003"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>meanFis_NC_10km <span class="ot">&lt;-</span> Daruanus.NC<span class="fl">.10</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.NC.10km,output = "Daruanus/NC/Daruanus_NC_10km.txt")</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC10km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="st">"NeEstimator/Daruanus_LDNe_NC_10kmxLD.txt"</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC10km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_NC10km, <span class="dv">10</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC10km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_NC10km<span class="sc">$</span>Ne)</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC10km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 638.502</code></pre>
</div>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.20</span>km <span class="ot">&lt;-</span> Daruanus.NC</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.20</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.NC<span class="fl">.20</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">center =</span> <span class="fu">c</span>(<span class="st">"MBO"</span>,<span class="st">"Lar"</span>,<span class="st">"Go"</span>,<span class="st">"QBW"</span>) )</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.20</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.NC<span class="fl">.20</span>km)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.20</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Mara"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Ten"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["center"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Tote"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0146","2":"0.0300625","3":"0.0181875","4":"0.003"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>meanFis_NC_20km <span class="ot">&lt;-</span> Daruanus.NC<span class="fl">.20</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.NC.20km,output = "NC/Daruanus_NC_20km.txt")</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC20km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"NeEstimator/Daruanus_LDNe_NC_20kmxLD.txt"</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC20km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_NC20km, <span class="dv">10</span>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC20km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_NC20km<span class="sc">$</span>Ne)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC20km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3062.967</code></pre>
</div>
</div>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.40</span>km <span class="ot">&lt;-</span> Daruanus.NC</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.40</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.NC<span class="fl">.40</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">north =</span> <span class="fu">c</span>(<span class="st">"Mara"</span>,<span class="st">"Ten"</span>),</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">center =</span> <span class="fu">c</span>(<span class="st">"MBO"</span>,<span class="st">"Lar"</span>,<span class="st">"Go"</span>,<span class="st">"QBW"</span>) )</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.40</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.NC<span class="fl">.40</span>km)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.40</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["north"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["center"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Tote"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0211875","2":"0.0181875","3":"0.003"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>meanFis_NC_40km <span class="ot">&lt;-</span> Daruanus.NC<span class="fl">.40</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.NC.40km,output = "NC/Daruanus_NC_40km.txt")</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC40km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"NeEstimator/Daruanus_LDNe_NC_40kmxLD.txt"</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC40km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_NC40km, <span class="dv">10</span>)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC40km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_NC40km<span class="sc">$</span>Ne)</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC40km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2373.392</code></pre>
</div>
</div>
</div>
<div id="tabset-2-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.100</span>km <span class="ot">&lt;-</span> Daruanus.NC</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.100</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.NC<span class="fl">.100</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">north =</span> <span class="fu">c</span>(<span class="st">"Mara"</span>),</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">east =</span> <span class="fu">c</span>(<span class="st">"MBO"</span>,<span class="st">"Lar"</span>,<span class="st">"Go"</span>,<span class="st">"QBW"</span>,<span class="st">"Tote"</span>,<span class="st">"Ten"</span>) )</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.100</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.NC<span class="fl">.100</span>km)</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>Daruanus.NC<span class="fl">.100</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["north"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["east"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0146","2":"0.0174875"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>meanFis_NC_100km <span class="ot">&lt;-</span> Daruanus.NC<span class="fl">.100</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.NC.100km,output = "NC/Daruanus_NC_100km.txt")</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC100km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>                                           <span class="st">"NeEstimator/Daruanus_LDNe_NC_100kmxLD.txt"</span>)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC100km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_NC100km, <span class="dv">10</span>)</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="co">#replace one very large estimate of Ne with 20,0000</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC100km<span class="sc">$</span>Ne[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC100km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_NC100km<span class="sc">$</span>Ne)</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC100km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12268.54</code></pre>
</div>
</div>
</div>
<div id="tabset-2-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-5-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC200km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">"NeEstimator/Daruanus_LDNe_NC_1popxLD.txt"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC200km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_NC200km, <span class="dv">10</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>Ne_hm_NC200km <span class="ot">&lt;-</span> Ne_estimates_NC200km<span class="sc">$</span>Ne</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>Ne_all_NC <span class="ot">&lt;-</span> Ne_estimates_NC200km<span class="sc">$</span>Ne</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>Ne_all_NC_CI <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_estimates_NC200km<span class="sc">$</span>ParametricLow,</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>                  Ne_estimates_NC200km<span class="sc">$</span>ParametricHigh)</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>Ne_all_NC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6294.9</code></pre>
</div>
</div>
</div>
<div id="tabset-2-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-6-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>NCwindows <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">SampleWindow =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">40</span>,<span class="dv">100</span>,<span class="dv">200</span>),</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">hm_Ne =</span> <span class="fu">c</span>(Ne_hm_NC,Ne_hm_NC10km,Ne_hm_NC20km,Ne_hm_NC40km,</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                                Ne_hm_NC100km,Ne_hm_NC200km))</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(NCwindows, <span class="fu">aes</span>(<span class="at">x =</span> SampleWindow, <span class="at">y =</span> hm_Ne)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">20000</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("NC_Ne_v_SampDistance.pdf",height = 7, width = 7)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</section>
<section id="calculate-effective-density" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-density">Calculate Effective Density</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide by mean distance between sampling sites to get density</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>De_NC <span class="ot">&lt;-</span> Ne_hm_NC<span class="sc">/</span>meandists_NC</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>De_all_NC <span class="ot">&lt;-</span> Ne_all_NC <span class="sc">/</span> maxdist_NC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean density is 32.6102505 individuals/km, or if we do the whole sample as a single population 41.9068643 individuals/km</p>
</section>
</section>
<section id="migraine-method" class="level2">
<h2 class="anchored" data-anchor-id="migraine-method">MigraiNe Method</h2>
<section id="running-migraine" class="level3">
<h3 class="anchored" data-anchor-id="running-migraine">Running MigraiNe</h3>
<p>I modified the genepop file by adding sampling coordinates as the name of the last individual in each population. These coordinates were distances in km along a the mostly linear SW coastline of New Caledonia, which runs a total of ~612km. It is ~418km to the first population at Mara, so I am adding that value to the coordinates in the file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>distfromP1_NC<span class="sc">+</span><span class="dv">418</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    Mara      Ten      MBO      Lar       Go      QBW     Tote 
418.0000 454.7026 494.1474 507.2045 514.9837 527.4177 568.2117 </code></pre>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true">First Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false">Second Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-3" role="tab" aria-controls="tabset-3-3" aria-selected="false">Third Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-4" role="tab" aria-controls="tabset-3-4" aria-selected="false">Fourth Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-5" role="tab" aria-controls="tabset-3-5" aria-selected="false">Seventh Run</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_NC.txt</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="co"># last individual in each population. These coordinates were distances in km along </span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a the mostly linear SW coastline of New Caledonia, </span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="co"># which runs a total of ~612km. It is ~418km to the first population at Mara, </span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="co"># so I am adding that value to the coordinates in the file.</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>612 <span class="ex">0</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 25.08</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a><span class="co">#612/25.08 = 24.40 so I will use 25 bins</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>25</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus</span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>26,35,11,57,47,32,30,23,57,34,44</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>1,2500,1</span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_NC.txt</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># last individual in each population. These coordinates were distances in km along </span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a the mostly linear SW coastline of New Caledonia, </span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co"># which runs a total of ~612km. It is ~418km to the first population at Mara, </span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co"># so I am adding that value to the coordinates in the file.</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMin</span><span class="op">=</span>0 <span class="ex">0</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>612 <span class="ex">0</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 25.08</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">#612/25.08 = 24.40 so I will use 26 bins</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>26</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>26,35,11,57,47,32,30,23,57,34,44</span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,10000,1</span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This finished in 250 minutes, and had very similar results to the first run</p>
</div>
<div id="tabset-3-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-3-tab">
<p>After removing 3 loci</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../../Daruanus_NC.txt</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># last individual in each population. These coordinates were distances in km along </span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a the mostly linear SW coastline of New Caledonia, </span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># which runs a total of ~612km. It is ~418km to the first population at Mara, </span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co"># so I am adding that value to the coordinates in the file.</span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMin</span><span class="op">=</span>0 <span class="ex">0</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>612 <span class="ex">0</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 25.08</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="co">#612/25.08 = 24.40 so I will use 25 bins</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>25</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs</span></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>26,11,57,32,30,23,57,44</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,10000,1</span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a><span class="va">Plots</span><span class="op">=</span> <span class="ex">all1DProfiles</span></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a><span class="co">#1DProfiles=2Nmu, 2Nm, Nb, condS2, g</span></span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-3-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-4-tab">
<p>And, lo, I forgot to use the condS2 parameterization that is recommended for weak IBD signals by <span class="citation" data-cites="lebloisMigraiNeManual2020">Leblois and Rousset (<a href="#ref-lebloisMigraiNeManual2020" role="doc-biblioref">2020</a>)</span>!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../../Daruanus_NC.txt</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># last individual in each population. These coordinates were distances in km along </span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a the mostly linear SW coastline of New Caledonia, </span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co"># which runs a total of ~612km. It is ~418km to the first population at Mara, </span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co"># so I am adding that value to the coordinates in the file.</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMin</span><span class="op">=</span>0 <span class="ex">0</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>612 <span class="ex">0</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 25.08</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="co">#612/25.08 = 24.40 so I will use 25 bins</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>25</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>26,11,57,32,30,23,57,44</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="co">#sampling - this performs uniform sampling of ln(sigma^2), which is the quantity we are interested in</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,condS2</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,logscale</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,1</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,10000,100000</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a><span class="va">Plots</span><span class="op">=</span> <span class="ex">all1DProfiles</span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a><span class="co">#1DProfiles=2Nmu, 2Nm, Nb, condS2, g</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finishing in 171.6 minutes</p>
</div>
<div id="tabset-3-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-5-tab">
<p>Brought in the priors a little. Didnâ€™t change the results much.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../../Daruanus_NC.txt</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># last individual in each population. These coordinates were distances in km along </span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># a the mostly linear SW coastline of New Caledonia, </span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="co"># which runs a total of ~612km. It is ~418km to the first population at Mara, </span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co"># so I am adding that value to the coordinates in the file.</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMin</span><span class="op">=</span>0 <span class="ex">0</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>612 <span class="ex">0</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 25.08</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="co">#612/25.08 = 24.40 so I will use 26 bins</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>26</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>26,11,57,32,30,23,57,44</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co">#sampling - this performs uniform sampling of ln(sigma^2)</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,condS2</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,logscale</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and conds2</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,1</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,6000,10000</span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a><span class="co">#oneDimCI= All</span></span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a><span class="va">Plots</span><span class="op">=</span> <span class="ex">all1DProfiles</span></span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a><span class="co">#1DProfiles=2Nmu, 2Nm, Nb, condS2, g</span></span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a><span class="va">writeAdHocFiles</span><span class="op">=</span>T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finished in 187.55 minutes.</p>
</div>
</div>
</div>
</section>
<section id="create-dispersal-kernels" class="level3">
<h3 class="anchored" data-anchor-id="create-dispersal-kernels">Create Dispersal Kernels</h3>
<section id="sigma-estimates" class="level4">
<h4 class="anchored" data-anchor-id="sigma-estimates">Sigma estimates</h4>
<p>So we have two routes to estimate <span class="math inline">\(\sigma\)</span> here.</p>
</section>
<section id="from-neighborhood-size" class="level4">
<h4 class="anchored" data-anchor-id="from-neighborhood-size">From Neighborhood Size</h4>
<p><span class="math display">\[
\sigma = \sqrt\frac{NS}{4D_e}
\]</span></p>
</section>
<section id="from-sigma2" class="level4">
<h4 class="anchored" data-anchor-id="from-sigma2">From Sigma^2</h4>
<p>After converting it from lattice units to km</p>
<p><span class="math display">\[
\sigma = \sqrt{\sigma^2}
\]</span></p>
<p>This got me the following estimates.</p>
<p>Output from run 7.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>runDir <span class="ot">&lt;-</span> <span class="st">"./NC/Migraine/run7"</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co">#runDir &lt;- "/Users/edc5240/github/IBD_Kernels/Daruanus/NC/Migraine/run7"</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_migraine</span>(runDir)</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>NS_NC <span class="ot">&lt;-</span> result[<span class="st">"NS"</span>]</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>NSCI_NC <span class="ot">&lt;-</span> <span class="fu">c</span>(result[<span class="st">"NSCI1"</span>],result[<span class="st">"NSCI2"</span>])</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>Nmu_NC <span class="ot">&lt;-</span> result[<span class="st">"Nmu"</span>]</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>Nm_NC <span class="ot">&lt;-</span> result[<span class="st">"Nm"</span>]</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>g_NC <span class="ot">&lt;-</span> result[<span class="st">"g"</span>]</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>lattice2geog_NC <span class="ot">&lt;-</span> result[<span class="st">"lattice2geog"</span>]</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>sigma2_NC <span class="ot">&lt;-</span> <span class="fu">g_to_sigma2</span>(g_NC)</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>sigma_fromsigma2_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(sigma2_NC<span class="sc">*</span>lattice2geog_NC)</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>sigma_fromNS_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NS_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_NC))</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>sigmaCI_fromNS_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NSCI_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_NC))</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>sigma_fromNS_all_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NS_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_all_NC))</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>sigmaCI_fromNS_all_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NSCI_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_all_NC))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The <span class="math inline">\(\sigma\)</span> we get from Neighborhood Size <span class="math inline">\(\sigma\)</span> is 227. We get a much lower estimate from <span class="math inline">\(\sigma^2\)</span>, with <span class="math inline">\(\sigma\)</span> = 1750</p>
</section>
</section>
<section id="confidence-intervals" class="level3">
<h3 class="anchored" data-anchor-id="confidence-intervals">Confidence Intervals</h3>
<p>Propagating error sorta following Pinsky et al.&nbsp;table S2</p>
<section id="error-in-effective-size" class="level4">
<h4 class="anchored" data-anchor-id="error-in-effective-size">Error in Effective Size</h4>
<p>Following <span class="citation" data-cites="pinskyUsingIsolationDistance2010">Pinsky, Montes, and Palumbi (<a href="#ref-pinskyUsingIsolationDistance2010" role="doc-biblioref">2010</a>)</span> I am going to bootstrap across the confidence intervals for each Nb estimate. Unfortunately, the new jackknife method of <span class="citation" data-cites="jonesImprovedConfidenceIntervals2016">Jones, Ovenden, and Wang (<a href="#ref-jonesImprovedConfidenceIntervals2016" role="doc-biblioref">2016</a>)</span> often results in infinite upper bounds with marine data (but then, so does the parametric method). Iâ€™m also going to use a uniform distribution for the error because approximating the error structure with ChiSq or Normal distributions is not a simple task and Iâ€™m just trying to get a sketch of the error to compare with MigraiNe anyway. Iâ€™m going to set â€œInfiniteâ€ values in the upper CI to 20,000 since I rarely see upper bounds that high.</p>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-4-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-1" role="tab" aria-controls="tabset-4-1" aria-selected="true">For Harmonic Mean Method</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-2" role="tab" aria-controls="tabset-4-2" aria-selected="false">For Whole Sample Method</a></li></ul>
<div class="tab-content">
<div id="tabset-4-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-4-1-tab">
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC<span class="sc">$</span>JackknifeHigh[<span class="fu">which</span>(<span class="fu">is.na</span>(Ne_estimates_NC<span class="sc">$</span>JackknifeHigh))] <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC<span class="sc">$</span>JackknifeHigh <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_NC<span class="sc">$</span>JackknifeHigh)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_NC<span class="sc">$</span>JackknifeLow <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_NC<span class="sc">$</span>JackknifeLow)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co"># couldn't get purrr:map to do this, so resorted to mapply to set upper and lower bounds</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>Ne_error_NC <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(n <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100000</span>){</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  hm <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(<span class="fu">mapply</span>(runif, <span class="at">n=</span><span class="dv">1</span>, </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">min=</span>Ne_estimates_NC<span class="sc">$</span>JackknifeLow,</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max=</span>Ne_estimates_NC<span class="sc">$</span>JackknifeHigh))</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>  Ne_error_NC <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_error_NC,hm)</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ne_error_NC)<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(Ne_error_NC, "NC/Ne_error_NC.csv",quote=F,row.names = F)</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Ne_error_NC), <span class="fu">aes</span>(<span class="at">x=</span>Ne_error_NC)) <span class="sc">+</span> <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-4-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-2-tab">
<p><span class="citation" data-cites="naaykensIsolationDistanceGenetic2022">Naaykens and Dâ€™Aloia (<a href="#ref-naaykensIsolationDistanceGenetic2022" role="doc-biblioref">2022</a>)</span> showed that using the whole sample to estimate density gives pretty similar results to the harmonic mean method, so Iâ€™m also going to try that.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>Nbl95 <span class="ot">&lt;-</span> Ne_all_NC_CI[<span class="dv">1</span>]</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>Nbu95 <span class="ot">&lt;-</span> Ne_all_NC_CI[<span class="dv">2</span>]</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>Nb <span class="ot">&lt;-</span> Ne_all_NC</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>r2_NC <span class="ot">&lt;-</span> Ne_estimates_NC200km<span class="sc">$</span>r2</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>er2_NC <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>Ne_estimates_NC200km<span class="sc">$</span>SampSize <span class="sc">+</span> <span class="fl">3.19</span><span class="sc">/</span>Ne_estimates_NC200km<span class="sc">$</span>SampSize <span class="sc">^</span><span class="dv">2</span> <span class="co">#from Waples 2006 table 2</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>df_NC <span class="ot">&lt;-</span> Ne_estimates_NC200km<span class="sc">$</span>IndAlleles</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="fu">WaplesMonoNe</span>(<span class="fu">r2p</span>(r2_NC,er2_NC))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6140.55</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this shows that we can get approximately what NeEstimator gives us... not sure why its not exact... must be missing some correction</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>rCI_NC <span class="ot">&lt;-</span> df_NC<span class="sc">*</span>r2_NC <span class="sc">/</span> (<span class="fu">qchisq</span>(<span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>), <span class="at">df =</span> df_NC))</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="fu">WaplesMonoNe</span>(rCI_NC <span class="sc">-</span> er2_NC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3729.688 16288.111</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co">#and now to get and plot the error distribution</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>Ne_error_all_NC <span class="ot">&lt;-</span> <span class="fu">WaplesMonoNe</span>(((df_NC<span class="sc">*</span>r2_NC)<span class="sc">/</span>(<span class="fu">rchisq</span>(<span class="dv">10000</span>, <span class="at">df =</span> df_NC))) <span class="sc">-</span> er2_NC)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Ne_error_all_NC), <span class="fu">aes</span>(<span class="at">x=</span>Ne_error_all_NC)) <span class="sc">+</span> <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">20000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="error-in-effective-density" class="level4">
<h4 class="anchored" data-anchor-id="error-in-effective-density">Error in Effective Density</h4>
<p>One issue with this analysis is that, while <span class="citation" data-cites="neelEstimationEffectivePopulation2013">Neel et al. (<a href="#ref-neelEstimationEffectivePopulation2013" role="doc-biblioref">2013</a>)</span> make a good case that we are estimating the Ne of the local neighborhood, we donâ€™t actually know what the size of the neighborhood is. Indeed, thatâ€™s actually what we are trying to estimate. <span class="citation" data-cites="pinskyUsingIsolationDistance2010">(<a href="#ref-pinskyUsingIsolationDistance2010" role="doc-biblioref">Pinsky, Montes, and Palumbi 2010</a>)</span> used neighborhoods that were 1/2 the distance to the next neighborhood on either side of the sampling site, while <span class="citation" data-cites="pinskyMarineDispersalScales2017">(<a href="#ref-pinskyMarineDispersalScales2017" role="doc-biblioref">Pinsky et al. 2017</a>)</span> didnâ€™t even attempt this and just used the Ne of the whole sampled population, and divided by the length of the whole sampled area.</p>
<p>This is another area of uncertainty, so we should model the uncertainty in neighborhood length. We know its between 10 and 40 km based on the Ne vs.&nbsp;Sampling Window analysis aboveâ€¦</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>Ne_error_NC <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">"NC/Ne_error_NC.csv"</span>)<span class="sc">$</span>x <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>De_error_NC <span class="ot">&lt;-</span> Ne_error_NC<span class="sc">/</span> <span class="fu">rnorm</span>(<span class="dv">100000</span>,<span class="at">mean=</span>meandists_NC,<span class="at">sd =</span> <span class="dv">15</span><span class="sc">/</span><span class="fl">1.96</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>De_error_all_NC <span class="ot">&lt;-</span> Ne_error_all_NC <span class="sc">/</span> maxdist_NC</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(De_error_all_NC), <span class="fu">aes</span>(<span class="at">x=</span>De_error_all_NC)) <span class="sc">+</span> <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">250</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(De_error_NC), <span class="fu">aes</span>(<span class="at">x=</span>De_error_NC)) <span class="sc">+</span> <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">250</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-31-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="error-in-neighborhood-size" class="level4">
<h4 class="anchored" data-anchor-id="error-in-neighborhood-size">Error in Neighborhood Size</h4>
<p>Using a uniform distribution is out because there is clearly a peaked distribution in the Migraine output. So I am using a quick fit to a truncated lognormal distribution.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figures/Da_NC_Neighborhood.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Migraine_Run2_Neighborhood_Theta</figcaption><p></p>
</figure>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>NSCI_NC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     NSCI1      NSCI2 
9.8516e+04 1.0000e+12 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">curve</span>(<span class="fu">dlnorm</span>(x, <span class="at">meanlog =</span> <span class="fu">log</span>(NS_NC), <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">2.75e5</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qlnorm</span>(<span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>),<span class="at">meanlog =</span> <span class="fu">log</span>(NS_NC), <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">2.75e5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.464785e-04 3.073026e+17</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>Neighborhood_error_NC <span class="ot">&lt;-</span> <span class="fu">rlnormTrunc</span>(<span class="at">n =</span> <span class="dv">100000</span>, <span class="at">meanlog =</span> <span class="fu">log</span>(NS_NC), </span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">2.75e5</span>), <span class="at">min =</span> NSCI_NC[<span class="dv">1</span>], <span class="at">max =</span> NSCI_NC[<span class="dv">2</span>])</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Neighborhood_error_NC), <span class="fu">aes</span>(<span class="at">x=</span>Neighborhood_error_NC)) <span class="sc">+</span> </span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>sigma_error_fromNS_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(Neighborhood_error_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_error_NC))</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sigma_error_fromNS_NC) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>sigma_error_fromNS_all_NC <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(Neighborhood_error_NC<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_error_all_NC))</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sigma_error_fromNS_all_NC) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromNS_NC), </span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromNS_NC)) <span class="sc">+</span></span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-33-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromNS_all_NC), </span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromNS_all_NC)) <span class="sc">+</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-33-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quantile</span>(sigma_error_fromNS_NC, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>),<span class="at">na.rm=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>       2.5%       97.5% 
   25.80886 56043.57610 </code></pre>
</div>
</div>
</section>
<section id="plot-dispersal-kernels" class="level4">
<h4 class="anchored" data-anchor-id="plot-dispersal-kernels">Plot Dispersal Kernels</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>kernelplot_NC <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">500</span>)),<span class="fu">aes</span>(x)) <span class="sc">+</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromNS_NC,<span class="dv">1000</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, </span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colour =</span> <span class="st">"lightblue"</span>,                                  <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">linewidth=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.2</span>) }) <span class="sc">+</span></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromsigma2_NC), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Sigma2"</span>), <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_NC), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>          <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Neighborhood_Size"</span>), <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_all_NC), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Neighborhood_Size_OnePop"</span>), <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Alongshore Distance (km)"</span>) <span class="sc">+</span> </span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Dispersal probability density"</span>) <span class="sc">+</span></span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"Kernel"</span>,<span class="at">values =</span> </span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>           <span class="fu">c</span>(<span class="at">Migraine_Sigma2=</span><span class="st">"darkblue"</span>, </span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">Migraine_Neighborhood_Size =</span><span class="st">"blue"</span>,</span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">Migraine_Neighborhood_Size_OnePop =</span> <span class="st">"darkcyan"</span>)) <span class="sc">+</span> </span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.01</span>) </span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>kernelplot_NC</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="fiji" class="level1">
<h1>Fiji</h1>
<section id="traditional-isolation-by-distance-method-1" class="level2">
<h2 class="anchored" data-anchor-id="traditional-isolation-by-distance-method-1">Traditional Isolation by Distance Method</h2>
<p>Based on the OG <span class="citation" data-cites="roussetGeneticDifferentiationEstimation1997">(<a href="#ref-roussetGeneticDifferentiationEstimation1997" role="doc-biblioref">Rousset 1997</a>)</span> estimator from slope of the IBD regression.</p>
<section id="calculate-distance-matrices-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-distance-matrices-1">Calculate distance matrices</h3>
<section id="genetic-distances-1" class="level4">
<h4 class="anchored" data-anchor-id="genetic-distances-1">Genetic Distances</h4>
<p>Weir and Cockerhamâ€™s Fst and other basic stats.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji.hfst <span class="ot">&lt;-</span> <span class="fu">genind2hierfstat</span>(Daruanus.Fiji)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji.loci <span class="ot">&lt;-</span> <span class="fu">genind2loci</span>(Daruanus.Fiji)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gen.loci &lt;- genind2loci(gen)</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>stats.Fiji <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.Fiji)</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>theta.Fiji <span class="ot">&lt;-</span> <span class="fu">theta.msat</span>(Daruanus.Fiji.loci)</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="co">#mean theta</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(theta.Fiji[,<span class="dv">2</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 61.06467</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>fst.Fiji <span class="ot">&lt;-</span> <span class="fu">genet.dist</span>(Daruanus.Fiji.hfst, <span class="at">method =</span> <span class="st">"WC84"</span>)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mean Fis values</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>stats.Fiji<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Suva"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Muaiv"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Yanu"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Taga"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Nadi"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["Mata"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Tave"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"-0.01635","2":"0.0421","3":"0.019275","4":"-0.0143375","5":"0.032325","6":"-0.0137875","7":"0.00805"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># linearize</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>fst.Fiji <span class="ot">&lt;-</span> fst.Fiji<span class="sc">/</span>(<span class="dv">1</span><span class="sc">-</span>fst.Fiji)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="geographic-distances-1" class="level5">
<h5 class="anchored" data-anchor-id="geographic-distances-1">Geographic Distances</h5>
<p>Also, given the circular nature of Viti Levu, the Euclidean distances measured with <code>pointDistance()</code> are going to be short. So I measured distances between each neighboring locality along the reef in Google Earth, as given in <code>fijidistances</code></p>
<p>This code creates a kml file for import into Google Earth.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>fijipops.sp<span class="ot">&lt;-</span>fijipops[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">4</span>)]</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">coordinates</span>(fijipops.sp) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"decimalLongitude"</span>,<span class="st">"decimalLatitude"</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(fijipops.sp) <span class="ot">&lt;-</span> <span class="fu">CRS</span>(<span class="st">"+proj=longlat +datum=WGS84"</span>)</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="co">#writeOGR(fijipops.sp, dsn="Fiji/fijipops.kml", layer = "Daruanus samples", driver = "KML")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>I used these points to measure distances in Google Earth. The total sampled length from Ovalu in the east to the Yasawas in the northwest was a total of ~414km.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate great circle distance</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>gcdists_Fiji <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="fu">pointDistance</span>(fijipops.sp, <span class="at">lonlat=</span>T)<span class="sc">/</span><span class="dv">1000</span>)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(gcdists_Fiji, <span class="st">"Labels"</span>) <span class="ot">&lt;-</span> fijipops<span class="sc">$</span>Abbr</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>gcdists.mat_Fiji <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(gcdists_Fiji)</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(gen.fst_Fiji), "Daruanus_linearizedFst.csv", row.names = F, quote=F)</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(gcdists_Fiji), "Daruanus_gcdists.csv", row.names = F, quote=F)</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="co">#pull out a few other distances we'll need</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>neighbordists_Fiji <span class="ot">&lt;-</span> gcdists.mat_Fiji[<span class="fu">row</span>(gcdists.mat_Fiji) <span class="sc">==</span> <span class="fu">col</span>(gcdists.mat_Fiji) <span class="sc">+</span> <span class="dv">1</span>]</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a><span class="co">#distfromP1 &lt;- gcdists.mat[,1]</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a><span class="co">#meandists &lt;- mean(neighbordists)</span></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>fijidistances <span class="ot">&lt;-</span> <span class="fu">c</span>((<span class="fl">106.03-98.16</span>), (<span class="fl">155.22-106.03</span>), (<span class="fl">188.58-155.22</span>), (<span class="fl">273.50-188.58</span>),</span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">370</span><span class="fl">-273.50</span>), (<span class="fl">378.13-370.0</span>))</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>meandists_Fiji <span class="ot">&lt;-</span> <span class="fu">mean</span>(fijidistances)</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>maxdist_Fiji <span class="ot">&lt;-</span> <span class="fl">378.13-98.16</span></span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>fijidistances</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  7.87 49.19 33.36 84.92 96.50  8.13</code></pre>
</div>
</div>
<p>This gives us a mean sampling distance of 46.6616667</p>
</section>
</section>
</section>
<section id="calculate-linear-model-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-linear-model-1">Calculate linear model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mantel test</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>mantelt<span class="ot">&lt;-</span><span class="fu">mantel.randtest</span>(fst.Fiji,gcdists_Fiji, <span class="at">nrepet =</span> <span class="dv">10000</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>distances <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">distance=</span><span class="fu">as.vector</span>(gcdists_Fiji),<span class="at">fst=</span><span class="fu">as.vector</span>(fst.Fiji))</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>lmodel_Fiji <span class="ot">&lt;-</span> <span class="fu">lm</span>(fst <span class="sc">~</span> distance , distances)</span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>slope_Fiji <span class="ot">&lt;-</span> lmodel_Fiji<span class="sc">$</span>coefficients[<span class="dv">2</span>]</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>mantelr <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>obs, <span class="dv">2</span>)</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>pvalue <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>pvalue, <span class="dv">5</span>)</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>lmodel_plot_Fiji <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(distances,<span class="fu">aes</span>(<span class="at">x=</span>distance,<span class="at">y=</span>fst)) <span class="sc">+</span></span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span> </span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">xlab</span>(<span class="st">"Geographic Distance (km)"</span>) <span class="sc">+</span> </span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylab</span>(<span class="fu">expression</span>(F[<span class="st">"ST"</span>]<span class="sc">/</span><span class="dv">1</span><span class="sc">-</span>F[<span class="st">"ST"</span>])) <span class="sc">+</span> </span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_text</span>(<span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"m ="</span>, <span class="fu">round</span>(slope_Fiji,<span class="dv">8</span>),</span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"; Mantel r ="</span>, mantelr,</span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>                            <span class="st">", p ="</span>, pvalue ), </span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">80</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.002</span>))</span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a>lmodel_plot_Fiji</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Fiji_IBD.pdf", plot=lmodel_plot,device="pdf", width=7, height=5,units="in")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And after removing the 3 wonky loci, the slope is very slightly positive!</p>
</section>
<section id="calculate-effective-size-2" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-size-2">Calculate Effective Size</h3>
<p>Pull out just the relevant Fiji estimates of Ne. The negative numbers reflect very high values of Ne!</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji <span class="ot">&lt;-</span> Ne_estimates_f <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Population <span class="sc">%in%</span> fijipops<span class="sc">$</span>Abbr)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">8</span>,<span class="dv">11</span>,<span class="dv">12</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Population"],"name":[1],"type":["chr"],"align":["left"]},{"label":["SampSize"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pcrit"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["WeightedMean"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Ne"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["JackknifeLow"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["JackknifeHigh"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"Mata","2":"48","3":"0.02","4":"48.0","5":"-2652.5","6":"634.0","7":"NA"},{"1":"Muaiv","2":"54","3":"0.02","4":"54.0","5":"-2340.3","6":"349.2","7":"NA"},{"1":"Nadi","2":"51","3":"0.02","4":"50.5","5":"3305.3","6":"328.5","7":"NA"},{"1":"Suva","2":"56","3":"0.02","4":"56.0","5":"1150.6","6":"291.1","7":"NA"},{"1":"Taga","2":"52","3":"0.02","4":"52.0","5":"-3057.6","6":"383.8","7":"NA"},{"1":"Tave","2":"50","3":"0.02","4":"50.0","5":"701.1","6":"293.8","7":"NA"},{"1":"Yanu","2":"47","3":"0.02","4":"47.0","5":"2822.5","6":"426.0","7":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># harmonic mean of Ne, following Waples and Do 2010</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_Fiji<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="ne-vs.-sampling-window-1" class="level3">
<h3 class="anchored" data-anchor-id="ne-vs.-sampling-window-1">Ne vs.&nbsp;Sampling Window</h3>
<p>Letâ€™s cluster the sites by UPGMA (using Euclidean distances)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">hclust</span>(gcdists_Fiji,<span class="st">"average"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-5-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-1" role="tab" aria-controls="tabset-5-1" aria-selected="true">10km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-2" role="tab" aria-controls="tabset-5-2" aria-selected="false">40km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-3" role="tab" aria-controls="tabset-5-3" aria-selected="false">100km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-4" role="tab" aria-controls="tabset-5-4" aria-selected="false">300km (all pops as one)</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-5" role="tab" aria-controls="tabset-5-5" aria-selected="false">Figure</a></li></ul>
<div class="tab-content">
<div id="tabset-5-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-5-1-tab">
<p>Letâ€™s explore Ne and <span class="math inline">\(F_{is}\)</span> when lumping populationsâ€¦ First lump populations that are less than 10 km apart</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>stats.Fiji<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Suva"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Muaiv"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Yanu"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Taga"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Nadi"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["Mata"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Tave"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"-0.01635","2":"0.0421","3":"0.019275","4":"-0.0143375","5":"0.032325","6":"-0.0137875","7":"0.00805"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.10</span>km <span class="ot">&lt;-</span> Daruanus.Fiji</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.Fiji<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">east =</span> <span class="fu">c</span>(<span class="st">"Suva"</span>,<span class="st">"Muaiv"</span>),</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">west =</span> <span class="fu">c</span>(<span class="st">"Mata"</span>,<span class="st">"Tave"</span>)</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.10</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.Fiji<span class="fl">.10</span>km)</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.10</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["east"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Yanu"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Taga"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Nadi"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["west"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.013475","2":"0.019275","3":"-0.0143375","4":"0.032325","5":"-0.0025875"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.Fiji.10km,output = "Daruanus/Fiji/Daruanus_Fiji_10km.txt")</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji10km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"NeEstimator/Daruanus_LDNe_Fiji_10kmxLD.txt"</span>)</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji10km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_Fiji10km, <span class="dv">10</span>)</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji10km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_Fiji10km<span class="sc">$</span>Ne)</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji10km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6700.065</code></pre>
</div>
</div>
</div>
<div id="tabset-5-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.40</span>km <span class="ot">&lt;-</span> Daruanus.Fiji</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.40</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.Fiji<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">east =</span> <span class="fu">c</span>(<span class="st">"Suva"</span>,<span class="st">"Muaiv"</span>),</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">center =</span> <span class="fu">c</span>(<span class="st">"Yanu"</span>,<span class="st">"Taga"</span>),</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">west =</span> <span class="fu">c</span>(<span class="st">"Mata"</span>,<span class="st">"Tave"</span>)</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.40</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.Fiji<span class="fl">.40</span>km)</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.40</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["east"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["center"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Nadi"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["west"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.013475","2":"-0.00065","3":"0.032325","4":"-0.0025875"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.Fiji.40km,output = "Daruanus/Fiji/Daruanus_Fiji_40km.txt")</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji40km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"NeEstimator/Daruanus_LDNe_Fiji_40kmxLD.txt"</span>)</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji40km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_Fiji40km, <span class="dv">10</span>)</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji40km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_Fiji40km<span class="sc">$</span>Ne)</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji40km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4601.126</code></pre>
</div>
</div>
</div>
<div id="tabset-5-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.100</span>km <span class="ot">&lt;-</span> Daruanus.Fiji</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.100</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.Fiji<span class="fl">.100</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">east =</span> <span class="fu">c</span>(<span class="st">"Suva"</span>,<span class="st">"Muaiv"</span>,<span class="st">"Yanu"</span>,<span class="st">"Taga"</span>),</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">west =</span> <span class="fu">c</span>(<span class="st">"Mata"</span>,<span class="st">"Tave"</span>,<span class="st">"Nadi"</span>)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.100</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.Fiji<span class="fl">.100</span>km)</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>Daruanus.Fiji<span class="fl">.100</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                                                                   mean, </span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>                                                                   <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["east"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["west"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.00815","2":"0.0100375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.Fiji.100km,output = "Fiji/Daruanus_Fiji_100km.txt")</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji100km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"NeEstimator/Daruanus_LDNe_Fiji_100kmxLD.txt"</span>)</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji100km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_Fiji100km, <span class="dv">10</span>)</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji100km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_Fiji100km<span class="sc">$</span>Ne)</span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji100km</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5822.584</code></pre>
</div>
</div>
</div>
<div id="tabset-5-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji300km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"NeEstimator/Daruanus_LDNe_Fiji_1popxLD.txt"</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji300km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_Fiji300km, <span class="dv">10</span>)</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>Ne_hm_Fiji300km <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>Ne_all_Fiji <span class="ot">&lt;-</span> Ne_estimates_Fiji300km<span class="sc">$</span>Ne</span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>Ne_all_Fiji_CI <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_estimates_Fiji300km<span class="sc">$</span>ParametricLow,</span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>                    Ne_estimates_Fiji300km<span class="sc">$</span>ParametricHigh)</span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a>Ne_all_Fiji</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -207211.8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>Ne_all_Fiji_CI</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6613.9     NA</code></pre>
</div>
</div>
</div>
<div id="tabset-5-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-5-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>fijiwindows <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">SampleWindow =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">40</span>,<span class="dv">100</span>,<span class="dv">300</span>),</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">hm_Ne =</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(Ne_hm_Fiji,Ne_hm_Fiji10km,Ne_hm_Fiji40km,</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>                          Ne_hm_Fiji100km,Ne_hm_Fiji300km))</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fijiwindows, <span class="fu">aes</span>(<span class="at">x =</span> SampleWindow, <span class="at">y =</span> hm_Ne)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">20000</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Fiji_Ne_v_SampDistance.pdf",height = 7, width = 7)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</section>
<section id="calculate-effective-density-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-density-1">Calculate Effective Density</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide by mean distance between sampling sites to get density</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>De_Fiji <span class="ot">&lt;-</span> Ne_hm_Fiji<span class="sc">/</span>meandists_Fiji</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="co">#De_all_Fiji &lt;- Ne_all_Fiji / maxdist_Fiji</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean density is 82.3843306 individuals/km. If we do the whole sample as a single population, Ne is â€œinfiniteâ€ so this isnâ€™t useful.</p>
</section>
<section id="calculate-sigma" class="level3">
<h3 class="anchored" data-anchor-id="calculate-sigma">Calculate sigma</h3>
<p>Following Roussetâ€™s <span class="citation" data-cites="roussetGeneticDifferentiationEstimation1997">(<a href="#ref-roussetGeneticDifferentiationEstimation1997" role="doc-biblioref">1997</a>)</span> equation:</p>
<p><span class="math display">\[
\frac{1}{m} = 4D_e\sigma^2
\]</span></p>
<p>Which <span class="citation" data-cites="pinskyMarineDispersalScales2017">(<a href="#ref-pinskyMarineDispersalScales2017" role="doc-biblioref">Pinsky et al. 2017</a>)</span> re-arranged to give:</p>
<p><span class="math display">\[
\sigma = \sqrt{\frac{1}{4D_em}}
\]</span></p>
<p>So now letâ€™s plug that into the first equation!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>sigma_fromSlope_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_Fiji<span class="sc">*</span>slope_Fiji))</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="co">#sigma_fromSlope_all_Fiji &lt;- sqrt(1 / (4*De_all_Fiji*slope_Fiji))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><span class="math inline">\(\sigma\)</span> estimated from this slope is 63.67 km if I use the harmonic mean Ne for density. I canâ€™t use the Ne from the whole population because it is infinite.</p>
</section>
</section>
<section id="migraine-method-1" class="level2">
<h2 class="anchored" data-anchor-id="migraine-method-1">MigraiNe Method</h2>
<section id="running-migraine-1" class="level3">
<h3 class="anchored" data-anchor-id="running-migraine-1">Running MigraiNe</h3>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-6-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-6-1" role="tab" aria-controls="tabset-6-1" aria-selected="true">First Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-6-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-6-2" role="tab" aria-controls="tabset-6-2" aria-selected="false">Second Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-6-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-6-3" role="tab" aria-controls="tabset-6-3" aria-selected="false">Third Run</a></li></ul>
<div class="tab-content">
<div id="tabset-6-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-6-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_Fiji.txt</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span> <span class="ex">LinearIBD</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. These coordinates were distances in km </span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="co">#along the Viti Levu reef and coastline from Ovalu in the east to the Yasawas </span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="co">#in the northwest: a total of ~414km. I measured all distances in Google Earth</span></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>414 <span class="ex">0</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 46.66</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a><span class="co">#414/46.66 = 8.87 so I will use 10 bins</span></span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>10</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.Fiji@loc.n.all)</span></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>22,27,8,46,47,30,24,20,52,31,38</span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>1,2500,1</span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This run completed in 147 minutes</p>
</div>
<div id="tabset-6-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-6-2-tab">
<p>A second run where I widen the prior on theta and Nm.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_Fiji.txt</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. These coordinates were distances in km </span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="co">#along the Viti Levu reef and coastline from Ovalu in the east to the Yasawas </span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a><span class="co">#in the northwest: a total of ~414km. I measured all distances in Google Earth</span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>414 <span class="ex">0</span></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 46.66</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a><span class="co">#414/46.66 = 8.87 so I will use 10 bins</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>10</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.Fiji@loc.n.all)</span></span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>22,27,8,46,47,30,24,20,52,31,38</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb100-22"><a href="#cb100-22" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb100-23"><a href="#cb100-23" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb100-24"><a href="#cb100-24" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb100-25"><a href="#cb100-25" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb100-26"><a href="#cb100-26" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb100-27"><a href="#cb100-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb100-28"><a href="#cb100-28" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb100-29"><a href="#cb100-29" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>1,2500,1</span>
<span id="cb100-30"><a href="#cb100-30" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb100-31"><a href="#cb100-31" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb100-32"><a href="#cb100-32" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb100-33"><a href="#cb100-33" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb100-34"><a href="#cb100-34" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb100-35"><a href="#cb100-35" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb100-36"><a href="#cb100-36" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This result finished in the same amount of time and with very similar results to the first!</p>
</div>
<div id="tabset-6-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-6-3-tab">
<p>Removing 3 loci and using the condS2 search parameterization</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../../Daruanus_Fiji.txt</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span>LinearIBD</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. These coordinates were distances in km </span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="co">#along the Viti Levu reef and coastline from Ovalu in the east to the Yasawas </span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a><span class="co">#in the northwest: a total of ~414km. I measured all distances in Google Earth</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>414 <span class="ex">0</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 46.66</span></span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a><span class="co">#414/46.66 = 8.87 so I will use 10 bins</span></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>10</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.Fiji@loc.n.all)</span></span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>22,8,46,30,24,20,52,38</span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,condS2</span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,logscale</span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,1</span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,10000,100000</span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a><span class="va">writeAdHocFiles</span><span class="op">=</span>T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finished in 112.9333333 minutes.</p>
</div>
</div>
</div>
</section>
<section id="create-dispersal-kernels-1" class="level3">
<h3 class="anchored" data-anchor-id="create-dispersal-kernels-1">Create Dispersal Kernels</h3>
<p>This got me the following estimates.</p>
<p>Output from run 2.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>runDir <span class="ot">&lt;-</span> <span class="st">"./Fiji/Migraine/run3"</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_migraine</span>(runDir)</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>NS_Fiji <span class="ot">&lt;-</span> result[<span class="st">"NS"</span>]</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>NSCI_Fiji <span class="ot">&lt;-</span> <span class="fu">c</span>(result[<span class="st">"NSCI1"</span>],result[<span class="st">"NSCI2"</span>])</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>Nmu_Fiji <span class="ot">&lt;-</span> result[<span class="st">"Nmu"</span>]</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>Nm_Fiji <span class="ot">&lt;-</span> result[<span class="st">"Nm"</span>]</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>g_Fiji <span class="ot">&lt;-</span> result[<span class="st">"g"</span>]</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>lattice2geog_Fiji <span class="ot">&lt;-</span> result[<span class="st">"lattice2geog"</span>]</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>sigma2_Fiji <span class="ot">&lt;-</span> <span class="fu">g_to_sigma2</span>(g_Fiji)</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>sigma_fromsigma2_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(sigma2_Fiji<span class="sc">*</span>lattice2geog_Fiji)</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>sigma_fromNS_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NS_Fiji<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_Fiji))</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>sigmaCI_fromNS_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NSCI_Fiji<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_Fiji))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The <span class="math inline">\(\sigma\)</span> we get from Neighborhood Size <span class="math inline">\(\sigma\)</span> is 31.2. We get a much lower estimate from <span class="math inline">\(\sigma^2\)</span>, with <span class="math inline">\(\sigma\)</span> = 5.92</p>
</section>
<section id="confidence-intervals-1" class="level3">
<h3 class="anchored" data-anchor-id="confidence-intervals-1">Confidence Intervals</h3>
<p>Propagating error sorta following Pinsky et al.&nbsp;table S2</p>
<section id="error-in-effective-size-1" class="level4">
<h4 class="anchored" data-anchor-id="error-in-effective-size-1">Error in Effective Size</h4>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-7-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-7-1" role="tab" aria-controls="tabset-7-1" aria-selected="true">For Harmonic Mean Method</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-7-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-7-2" role="tab" aria-controls="tabset-7-2" aria-selected="false">For Whole Sample Method</a></li></ul>
<div class="tab-content">
<div id="tabset-7-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-7-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji<span class="sc">$</span>JackknifeHigh[<span class="fu">which</span>(<span class="fu">is.na</span>(Ne_estimates_Fiji<span class="sc">$</span>JackknifeHigh))] <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji<span class="sc">$</span>JackknifeHigh <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_Fiji<span class="sc">$</span>JackknifeHigh)</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_Fiji<span class="sc">$</span>JackknifeLow <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_Fiji<span class="sc">$</span>JackknifeLow)</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>Ne_error_Fiji <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(n <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100000</span>){</span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>  hm <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(<span class="fu">mapply</span>(runif, <span class="at">n=</span><span class="dv">1</span>, </span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">min=</span>Ne_estimates_Fiji<span class="sc">$</span>JackknifeLow,</span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max=</span>Ne_estimates_Fiji<span class="sc">$</span>JackknifeHigh))</span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>  Ne_error_Fiji <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_error_Fiji,hm)</span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ne_error_Fiji)<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Ne_error_NC &lt;- read_csv(file = "NC/Ne_error_NC.csv")$x %&gt;% as.vector</span></span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Ne_error_Fiji), <span class="fu">aes</span>(<span class="at">x=</span>Ne_error_Fiji)) <span class="sc">+</span> <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-7-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-7-2-tab">
<p>Canâ€™t do the whole sample method, because Ne estimate is â€œinfiniteâ€</p>
</div>
</div>
</div>
</section>
<section id="error-in-effective-density-1" class="level4">
<h4 class="anchored" data-anchor-id="error-in-effective-density-1">Error in Effective Density</h4>
<p>Will model the error in <code>meandist</code> as wellâ€¦</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>Ne_error_Fiji <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">"Fiji/Ne_error_Fiji.csv"</span>)<span class="sc">$</span>x <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>De_error_Fiji <span class="ot">&lt;-</span> Ne_error_Fiji<span class="sc">/</span> <span class="fu">rnormTrunc</span>(<span class="dv">100000</span>,<span class="at">mean=</span>meandists_Fiji,</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">sd =</span> meandists_Fiji<span class="sc">/</span><span class="fl">1.96</span>, <span class="at">min =</span> <span class="fl">1e-10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="error-in-slope" class="level4">
<h4 class="anchored" data-anchor-id="error-in-slope">Error in Slope</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>slope_se_Fiji <span class="ot">&lt;-</span> <span class="fu">summary</span>(lmodel_Fiji)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1e-5</span>)), <span class="fu">aes</span>(<span class="at">x=</span>x)) <span class="sc">+</span> <span class="fu">stat_function</span>(<span class="at">fun=</span>dnormTrunc, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> slope_Fiji, <span class="at">sd=</span>slope_se_Fiji, <span class="at">min =</span> <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>slope_error_Fiji <span class="ot">&lt;-</span> <span class="fu">rnormTrunc</span>(<span class="dv">100000</span>, <span class="at">mean =</span> slope_Fiji, <span class="at">sd =</span> slope_se_Fiji,<span class="at">min =</span> <span class="fl">1e-10</span>)</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(slope_error_Fiji), <span class="fu">aes</span>(<span class="at">x=</span>slope_error_Fiji)) <span class="sc">+</span> <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-54-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>sigma_error_fromSlope_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_error_Fiji<span class="sc">*</span>slope_error_Fiji))</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromSlope_Fiji), <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromSlope_Fiji)) <span class="sc">+</span></span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">1000</span>)  <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-54-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quantile</span>(sigma_error_fromSlope_Fiji, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     2.5%     97.5% 
 5.858928 90.741609 </code></pre>
</div>
</div>
</section>
<section id="error-in-neighborhood-size-1" class="level4">
<h4 class="anchored" data-anchor-id="error-in-neighborhood-size-1">Error in Neighborhood Size</h4>
<p>Using a uniform distribution is out because there is clearly a peaked distribution in the Migraine output. So I am using a quick fit to a lognormal distribution</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figures/Da_Fiji_Neighborhood.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Migraine_Run2_Neighborhood_Theta</figcaption><p></p>
</figure>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>NSCI_Fiji</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    NSCI1     NSCI2 
   122779 101654085 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qlnorm</span>(<span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>),<span class="at">meanlog =</span> <span class="fu">log</span>(NS_Fiji), <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">18.89</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.010201e+03 1.016589e+08</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>Neighborhood_error_Fiji <span class="ot">&lt;-</span> <span class="fu">rlnormTrunc</span>(<span class="at">n =</span> <span class="dv">100000</span>, <span class="at">meanlog =</span> <span class="fu">log</span>(NS_Fiji), </span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">18.89</span>), <span class="at">min =</span> NSCI_Fiji[<span class="dv">1</span>], <span class="at">max =</span> NSCI_Fiji[<span class="dv">2</span>])</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Neighborhood_error_Fiji), <span class="fu">aes</span>(<span class="at">x=</span>Neighborhood_error_Fiji)) <span class="sc">+</span> </span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>sigma_error_fromNS_Fiji <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(Neighborhood_error_Fiji<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_error_Fiji))</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sigma_error_fromNS_Fiji) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromNS_Fiji), </span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromNS_Fiji)) <span class="sc">+</span></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-56-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="plot-dispersal-kernels-1" class="level4">
<h4 class="anchored" data-anchor-id="plot-dispersal-kernels-1">Plot Dispersal Kernels</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>kernelplot_Fiji <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)),<span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromNS_Fiji,<span class="dv">1000</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> <span class="st">"lightblue"</span>,</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.2</span>) }) <span class="sc">+</span></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromSlope_Fiji,<span class="dv">1000</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">colour =</span> <span class="st">"lightgreen"</span>,</span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.2</span>) }) <span class="sc">+</span></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromSlope_Fiji), </span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">linetype=</span><span class="dv">2</span>,<span class="fu">aes</span>(<span class="at">color=</span><span class="st">"IBD_Slope"</span>),<span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromsigma2_Fiji), </span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">linetype=</span><span class="dv">2</span>,<span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Sigma2"</span>), </span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_Fiji), </span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>                              <span class="at">linetype=</span><span class="dv">2</span>,<span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Neighborhood_Size"</span>), </span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">show.legend =</span> T) <span class="sc">+</span> </span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">xlab</span>(<span class="st">"Alongshore Distance (km)"</span>) <span class="sc">+</span></span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">ylab</span>(<span class="st">"Dispersal probability density"</span>) <span class="sc">+</span></span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">scale_color_manual</span>(<span class="st">"Kernel"</span>,<span class="at">values =</span> </span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(<span class="at">Migraine_Sigma2=</span><span class="st">"darkblue"</span>, </span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Migraine_Neighborhood_Size =</span><span class="st">"blue"</span>,</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">IBD_Slope =</span> <span class="st">"green"</span>)) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.05</span>)</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a>kernelplot_Fiji</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The reason that the error is generally smaller than the estimate is because the upper bounds of Ne are often unbounded, and treated as 20,000.</p>
</section>
</section>
</section>
</section>
<section id="society-islands" class="level1">
<h1>Society Islands</h1>
<section id="traditional-isolation-by-distance-method-2" class="level2">
<h2 class="anchored" data-anchor-id="traditional-isolation-by-distance-method-2">Traditional Isolation by Distance Method</h2>
<p>Based on the OG <span class="citation" data-cites="roussetGeneticDifferentiationEstimation1997">(<a href="#ref-roussetGeneticDifferentiationEstimation1997" role="doc-biblioref">Rousset 1997</a>)</span> estimator from slope of the IBD regression.</p>
<section id="calculate-distance-matrices-2" class="level3">
<h3 class="anchored" data-anchor-id="calculate-distance-matrices-2">Calculate distance matrices</h3>
<section id="genetic-distances-2" class="level4">
<h4 class="anchored" data-anchor-id="genetic-distances-2">Genetic Distances</h4>
<p>Weir and Cockerhamâ€™s Fst and other basic stats.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>Daruanus.FP.hfst <span class="ot">&lt;-</span> <span class="fu">genind2hierfstat</span>(Daruanus.FP)</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>Daruanus.FP.loci <span class="ot">&lt;-</span> <span class="fu">genind2loci</span>(Daruanus.FP)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gen.loci &lt;- genind2loci(gen)</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>stats.FP <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.FP)</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>theta.FP <span class="ot">&lt;-</span> <span class="fu">theta.msat</span>(Daruanus.FP.loci)</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a><span class="co">#mean theta</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(theta.FP[,<span class="dv">2</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 53.37003</code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>fst.FP <span class="ot">&lt;-</span> <span class="fu">genet.dist</span>(Daruanus.FP.hfst, <span class="at">method =</span> <span class="st">"WC84"</span>)</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mean Fis values</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>stats.FP<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Puna"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Tem"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mo"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Tetia"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Tahaa"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.00615","2":"0.0063125","3":"-0.001225","4":"0.028625","5":"0.0123375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># linearize</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>fst.FP <span class="ot">&lt;-</span> fst.FP<span class="sc">/</span>(<span class="dv">1</span><span class="sc">-</span>fst.FP)</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(fst.FP), "Daruanus_FP_linearizedFst.csv", row.names = F, quote=F)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="geographic-distances-2" class="level5">
<h5 class="anchored" data-anchor-id="geographic-distances-2">Geographic Distances</h5>
<p>Calculate great circle distance</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>gcdists_FP <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="fu">pointDistance</span>(FPpops[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">4</span>)], <span class="at">lonlat=</span>T)<span class="sc">/</span><span class="dv">1000</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(gcdists_FP, <span class="st">"Labels"</span>) <span class="ot">&lt;-</span> FPpops<span class="sc">$</span>Abbr</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>gcdists.mat_FP <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(gcdists_FP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now to follow what Malin did, and create a principal components axis, and measure distance along it.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># because we only care about the x axis, we need to reorder,so that Tetiaroa comes after Puna and before Temae</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>FPpops <span class="ot">&lt;-</span> FPpops[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>),]</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>FPpops.sp <span class="ot">&lt;-</span> <span class="fu">SpatialPointsDataFrame</span>(FPpops[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">4</span>)], <span class="at">data =</span> FPpops, </span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">proj4string =</span> <span class="fu">CRS</span>(<span class="st">"+proj=longlat  +datum=WGS84"</span>))</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>FPpops.utm <span class="ot">&lt;-</span> <span class="fu">spTransform</span>(FPpops.sp, <span class="fu">CRS</span>(<span class="st">"+proj=utm +zone=56 +datum=WGS84"</span>))</span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a><span class="co">#as.dist(pointDistance(localities.utm,latlon=F)/1000)</span></span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="co">#principal components without scaling or centering, we just want the rotation</span></span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>pc_FP <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(FPpops.utm<span class="sc">@</span>coords, <span class="at">retx=</span>T, <span class="at">scale=</span>F,<span class="at">center=</span>F)</span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pc_FP<span class="sc">$</span>x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set PC2 axis to zero</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>pc1_FP<span class="ot">&lt;-</span><span class="fu">cbind</span>(pc_FP<span class="sc">$</span>x[,<span class="dv">1</span>],<span class="dv">0</span>)</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>pcdists_FP <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="fu">pointDistance</span>(pc1_FP,<span class="at">lonlat=</span>F)<span class="sc">/</span><span class="dv">1000</span>)</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(pcdists_FP, <span class="st">"Labels"</span>) <span class="ot">&lt;-</span> FPpops<span class="sc">$</span>Abbr</span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(as.matrix(pcdists), "Apercula_pcdists.csv", row.names = T, quote=F)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>pcdists.mat_FP <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(pcdists_FP)</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a><span class="co">#pull out a few other distances we'll need</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>neighbordists_FP <span class="ot">&lt;-</span> pcdists.mat_FP[<span class="fu">row</span>(pcdists.mat_FP) <span class="sc">==</span> <span class="fu">col</span>(pcdists.mat_FP) <span class="sc">+</span> <span class="dv">1</span>]</span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>distfromP1_FP <span class="ot">&lt;-</span> pcdists.mat_FP[,<span class="dv">1</span>]</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>maxdist_FP <span class="ot">&lt;-</span> <span class="fu">max</span>(pcdists.mat_FP)</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>meandists_FP <span class="ot">&lt;-</span> <span class="fu">mean</span>(neighbordists_FP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean sampling distance is 86.1473678 km. But note that Tetiaroa occurs only a couple of kilometers from the Moorea population because they are being forced onto the rotated X axis.</p>
</section>
</section>
</section>
<section id="calculate-linear-model-2" class="level3">
<h3 class="anchored" data-anchor-id="calculate-linear-model-2">Calculate linear model</h3>
<p>First to get the slope <span class="math inline">\(m\)</span> we need to make a simple linear model. I donâ€™t think significance is really important here, but we can calculate that with a Mantel test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mantel test</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>mantelt<span class="ot">&lt;-</span><span class="fu">mantel.randtest</span>(fst.FP,pcdists_FP, <span class="at">nrepet =</span> <span class="dv">10000</span>)</span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>distances <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">distance=</span><span class="fu">as.vector</span>(pcdists_FP),<span class="at">fst=</span><span class="fu">as.vector</span>(fst.FP))</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>lmodel_FP <span class="ot">&lt;-</span> <span class="fu">lm</span>(fst <span class="sc">~</span> distance , distances)</span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a>slope_FP <span class="ot">&lt;-</span> lmodel_FP<span class="sc">$</span>coefficients[<span class="dv">2</span>]</span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a>mantelr <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>obs, <span class="dv">2</span>)</span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a>pvalue <span class="ot">&lt;-</span> <span class="fu">round</span>(mantelt<span class="sc">$</span>pvalue, <span class="dv">5</span>)</span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a>lmodel_plot_FP <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(distances,<span class="fu">aes</span>(<span class="at">x=</span>distance,<span class="at">y=</span>fst)) <span class="sc">+</span></span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span> </span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">xlab</span>(<span class="st">"Geographic Distance (km)"</span>) <span class="sc">+</span> </span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ylab</span>(<span class="fu">expression</span>(F[<span class="st">"ST"</span>]<span class="sc">/</span><span class="dv">1</span><span class="sc">-</span>F[<span class="st">"ST"</span>])) <span class="sc">+</span> </span>
<span id="cb125-16"><a href="#cb125-16" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_text</span>(<span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"m ="</span>, slope_FP, <span class="st">"; Mantel r ="</span>, </span>
<span id="cb125-17"><a href="#cb125-17" aria-hidden="true" tabindex="-1"></a>                                  mantelr,<span class="st">", p ="</span>, pvalue ), </span>
<span id="cb125-18"><a href="#cb125-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">80</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.002</span>))</span>
<span id="cb125-19"><a href="#cb125-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-20"><a href="#cb125-20" aria-hidden="true" tabindex="-1"></a>lmodel_plot_FP</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("FP_IBD.pdf", plot=lmodel_plot,device="pdf", width=7, height=5,units="in")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>E voila, câ€™est positive! But not significantly so.</p>
</section>
<section id="calculate-effective-size-3" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-size-3">Calculate Effective Size</h3>
<p>Pull out just the relevant FP estimates of Ne. The negative numbers reflect very high values of Ne!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP <span class="ot">&lt;-</span> Ne_estimates_f <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Population <span class="sc">%in%</span> FPpops<span class="sc">$</span>Abbr)</span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">8</span>,<span class="dv">11</span>,<span class="dv">12</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Population"],"name":[1],"type":["chr"],"align":["left"]},{"label":["SampSize"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pcrit"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["WeightedMean"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Ne"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["JackknifeLow"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["JackknifeHigh"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"Mo","2":"264","3":"0.01","4":"264","5":"-9543.0","6":"3618.6","7":"NA"},{"1":"Puna","2":"25","3":"0.02","4":"25","5":"1199.8","6":"166.1","7":"NA"},{"1":"Tahaa","2":"36","3":"0.02","4":"36","5":"287.4","6":"127.8","7":"NA"},{"1":"Tem","2":"64","3":"0.02","4":"64","5":"-2518.0","6":"770.0","7":"NA"},{"1":"Tetia","2":"52","3":"0.02","4":"52","5":"905.4","6":"262.5","7":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="co"># harmonic mean of Ne, following Waples and Do 2010</span></span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>Ne_hm_FP <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_FP<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ne-vs.-sampling-window-2" class="level3">
<h3 class="anchored" data-anchor-id="ne-vs.-sampling-window-2">Ne vs.&nbsp;Sampling Window</h3>
<p>Letâ€™s cluster the sites by UPGMA (using Euclidean distances)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">hclust</span>(pcdists_FP,<span class="st">"average"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-8-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-1" role="tab" aria-controls="tabset-8-1" aria-selected="true">10km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-2" role="tab" aria-controls="tabset-8-2" aria-selected="false">20km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-3" role="tab" aria-controls="tabset-8-3" aria-selected="false">40km</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-4" role="tab" aria-controls="tabset-8-4" aria-selected="false">300km (all pops as one)</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-8-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-8-5" role="tab" aria-controls="tabset-8-5" aria-selected="false">Figure</a></li></ul>
<div class="tab-content">
<div id="tabset-8-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-8-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>stats.FP<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Puna"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Tem"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mo"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Tetia"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Tahaa"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.00615","2":"0.0063125","3":"-0.001225","4":"0.028625","5":"0.0123375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>meanFis_FP <span class="ot">&lt;-</span> stats.FP<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.10</span>km <span class="ot">&lt;-</span> Daruanus.FP</span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.FP<span class="fl">.10</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">TemTetia =</span> <span class="fu">c</span>(<span class="st">"Tetia"</span>,<span class="st">"Tem"</span>)</span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.10</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.FP<span class="fl">.10</span>km)</span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.10</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Puna"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["TemTetia"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mo"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Tahaa"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.00615","2":"0.016375","3":"-0.001225","4":"0.0123375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>meanFis_FP_10km <span class="ot">&lt;-</span> Daruanus.FP<span class="fl">.10</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.FP.10km,output = "Daruanus/FP/Daruanus_FP_10km.txt")</span></span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP10km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(</span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"NeEstimator/Daruanus_LDNe_FP_10kmxLD.txt"</span>)</span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP10km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_FP10km, <span class="dv">10</span>)</span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-11"><a href="#cb132-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-12"><a href="#cb132-12" aria-hidden="true" tabindex="-1"></a>Ne_hm_FP10km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_FP10km<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-8-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.20</span>km <span class="ot">&lt;-</span> Daruanus.FP</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.20</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.FP<span class="fl">.20</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">TemTetiaMo =</span> <span class="fu">c</span>(<span class="st">"Tetia"</span>,<span class="st">"Tem"</span>,<span class="st">"Mo"</span>)</span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.20</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.FP<span class="fl">.20</span>km)</span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.20</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Puna"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["TemTetiaMo"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Tahaa"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.00615","2":"0.0047","3":"0.0123375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>meanFis_FP_20km <span class="ot">&lt;-</span> Daruanus.FP<span class="fl">.20</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.FP.20km,output = "Daruanus/FP/Daruanus_FP_20km.txt")</span></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP20km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"NeEstimator/Daruanus_LDNe_FP_20kmxLD.txt"</span>)</span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP20km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_FP20km, <span class="dv">10</span>)</span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-9"><a href="#cb134-9" aria-hidden="true" tabindex="-1"></a>Ne_hm_FP20km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_FP20km<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-8-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.40</span>km <span class="ot">&lt;-</span> Daruanus.FP</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.40</span>km<span class="sc">@</span>pop <span class="ot">&lt;-</span>  Daruanus.FP<span class="fl">.40</span>km<span class="sc">@</span>pop <span class="sc">%&gt;%</span> <span class="fu">fct_collapse</span>(</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">East =</span> <span class="fu">c</span>(<span class="st">"Tetia"</span>,<span class="st">"Tem"</span>,<span class="st">"Mo"</span>,<span class="st">"Puna"</span>)</span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a> )</span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.40</span>km.stats <span class="ot">&lt;-</span> <span class="fu">basic.stats</span>(Daruanus.FP<span class="fl">.40</span>km)</span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>Daruanus.FP<span class="fl">.40</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["East"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Tahaa"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.004825","2":"0.0123375"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>meanFis_FP_40km <span class="ot">&lt;-</span> Daruanus.FP<span class="fl">.40</span>km.stats<span class="sc">$</span>Fis <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">meanFis =</span> <span class="fu">rowMeans</span>(.))</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a><span class="co">#genind_to_genepop(Daruanus.FP.40km,output = "Daruanus/FP/Daruanus_FP_40km.txt")</span></span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP40km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span> <span class="st">"NeEstimator/Daruanus_LDNe_FP_40kmxLD.txt"</span>)</span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP40km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_FP40km, <span class="dv">10</span>)</span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a>Ne_hm_FP40km <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(Ne_estimates_FP40km<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-8-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP300km <span class="ot">&lt;-</span> <span class="fu">read_NeEstimator</span>(<span class="at">file =</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"NeEstimator/Daruanus_LDNe_FP_1popxLD.txt"</span>)</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP300km <span class="ot">&lt;-</span> <span class="fu">WDFilter</span>(Ne_estimates_FP300km, <span class="dv">10</span>)</span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>Ne_hm_FP300km <span class="ot">&lt;-</span> Ne_estimates_FP300km<span class="sc">$</span>Ne</span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a>Ne_all_FP <span class="ot">&lt;-</span> Ne_estimates_FP300km<span class="sc">$</span>Ne</span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>Ne_all_FP_CI <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_estimates_FP300km<span class="sc">$</span>ParametricLow,<span class="dv">20000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-8-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-8-5-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>FPwindows <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">SampleWindow =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">40</span>,<span class="dv">300</span>),</span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">hm_Ne =</span> <span class="fu">c</span>(Ne_hm_FP,Ne_hm_FP10km,Ne_hm_FP40km,Ne_hm_FP300km))</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a>FPNevDistance <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(FPwindows, <span class="fu">aes</span>(<span class="at">x =</span> SampleWindow, <span class="at">y =</span> hm_Ne)) <span class="sc">+</span> </span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">20000</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">300</span>)</span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a>FPNevDistance</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("FP_Ne_v_SampDistance.pdf", height = 7, width = 7)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</section>
<section id="calculate-effective-density-2" class="level3">
<h3 class="anchored" data-anchor-id="calculate-effective-density-2">Calculate Effective Density</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide by mean distance between sampling sites to get density</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>De_FP <span class="ot">&lt;-</span> Ne_hm_FP<span class="sc">/</span>meandists_FP</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>De_all_FP <span class="ot">&lt;-</span> Ne_all_FP<span class="sc">/</span>maxdist_FP</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean density in the Societies is 11.807568 inds/km, or really really large if we consider the whole sample as a single population 34.9942393</p>
</section>
<section id="calculate-sigma-1" class="level3">
<h3 class="anchored" data-anchor-id="calculate-sigma-1">Calculate sigma</h3>
<p>Following Roussetâ€™s <span class="citation" data-cites="roussetGeneticDifferentiationEstimation1997">(<a href="#ref-roussetGeneticDifferentiationEstimation1997" role="doc-biblioref">1997</a>)</span> equation:</p>
<p><span class="math display">\[
\frac{1}{m} = 4D_e\sigma^2
\]</span></p>
<p>Which <span class="citation" data-cites="pinskyMarineDispersalScales2017">(<a href="#ref-pinskyMarineDispersalScales2017" role="doc-biblioref">Pinsky et al. 2017</a>)</span> re-arranged to give:</p>
<p><span class="math display">\[
\sigma = \sqrt{\frac{1}{4D_em}}
\]</span></p>
<p>So now letâ€™s plug that into the first equation!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a>sigma_fromSlope_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_FP<span class="sc">*</span>slope_FP))</span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>sigma_fromSlope_all_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_all_FP<span class="sc">*</span>slope_FP))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>So effective density is 11.81 individuals per km, and <span class="math inline">\(\sigma\)</span> is 64.23 km if we calculate density based on harmonic mean of Ne, or <code>r</code>signif(sigma_fromSlope_all_FP,4)` km if we calculate it across the whole sample.</p>
</section>
</section>
<section id="migraine-method-2" class="level2">
<h2 class="anchored" data-anchor-id="migraine-method-2">MigraiNe Method</h2>
<section id="running-migraine-2" class="level3">
<h3 class="anchored" data-anchor-id="running-migraine-2">Running MigraiNe</h3>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-9-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-9-1" role="tab" aria-controls="tabset-9-1" aria-selected="true">First Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-9-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-9-2" role="tab" aria-controls="tabset-9-2" aria-selected="false">Second Run</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-9-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-9-3" role="tab" aria-controls="tabset-9-3" aria-selected="false">Fifth Run</a></li></ul>
<div class="tab-content">
<div id="tabset-9-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-9-1-tab">
<p>I modified the genepop file by adding sampling coordinates as the name of the last individual in each population. There is no coastline for these samples, which each come from different islands. I just measured from Tahiti to Maupiti Itâ€™s 65 km from the tip of Tahiti Iti to Puna, so I added that to each coordinate</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>distfromP1_FP<span class="sc">+</span><span class="dv">65</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Puna     Tetia       Tem        Mo     Tahaa 
 65.00000  73.17410  90.15654 116.74566 393.24127 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_FP.txt</span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span> <span class="ex">LinearIBD</span></span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. There is no coastline for these samples,</span></span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a><span class="co"># which each come from different islands. I just measured from Tahiti to Maupiti</span></span>
<span id="cb144-6"><a href="#cb144-6" aria-hidden="true" tabindex="-1"></a><span class="co"># It's 65 km from the tip of Tahiti Iti to Puna, so I added that to </span></span>
<span id="cb144-7"><a href="#cb144-7" aria-hidden="true" tabindex="-1"></a><span class="co"># each coordinate.</span></span>
<span id="cb144-8"><a href="#cb144-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>355 <span class="ex">0</span></span>
<span id="cb144-9"><a href="#cb144-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 75.78</span></span>
<span id="cb144-10"><a href="#cb144-10" aria-hidden="true" tabindex="-1"></a><span class="co">#355/75.78 = 4.684 so I will use 6 bins</span></span>
<span id="cb144-11"><a href="#cb144-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>6</span>
<span id="cb144-12"><a href="#cb144-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb144-13"><a href="#cb144-13" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb144-14"><a href="#cb144-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb144-15"><a href="#cb144-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb144-16"><a href="#cb144-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb144-17"><a href="#cb144-17" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.FP@loc.n.all)</span></span>
<span id="cb144-18"><a href="#cb144-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb144-19"><a href="#cb144-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>21,24,8,37,43,26,19,16,43,31,40</span>
<span id="cb144-20"><a href="#cb144-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb144-21"><a href="#cb144-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb144-22"><a href="#cb144-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb144-23"><a href="#cb144-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb144-24"><a href="#cb144-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb144-25"><a href="#cb144-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb144-26"><a href="#cb144-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb144-27"><a href="#cb144-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb144-28"><a href="#cb144-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb144-29"><a href="#cb144-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb144-30"><a href="#cb144-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>2,10000,1</span>
<span id="cb144-31"><a href="#cb144-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb144-32"><a href="#cb144-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb144-33"><a href="#cb144-33" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb144-34"><a href="#cb144-34" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb144-35"><a href="#cb144-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb144-36"><a href="#cb144-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb144-37"><a href="#cb144-37" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The first run finished with an error, but a second identical run completed in 50.8666667 minutes. Both runs had very similar results.</p>
</div>
<div id="tabset-9-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-9-2-tab">
<p>Widened the prior on theta because the first two runs were over</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../Daruanus_FP.txt</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span> <span class="ex">LinearIBD</span></span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. There is no coastline for these samples,</span></span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a><span class="co"># which each come from different islands. I just measured from Tahiti to Maupiti</span></span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a><span class="co"># It's 65 km from the tip of Tahiti Iti to Puna, so I added that to </span></span>
<span id="cb145-7"><a href="#cb145-7" aria-hidden="true" tabindex="-1"></a><span class="co"># each coordinate.</span></span>
<span id="cb145-8"><a href="#cb145-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>355 <span class="ex">0</span></span>
<span id="cb145-9"><a href="#cb145-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 75.78</span></span>
<span id="cb145-10"><a href="#cb145-10" aria-hidden="true" tabindex="-1"></a><span class="co">#355/75.78 = 4.684 so I will use 6 bins</span></span>
<span id="cb145-11"><a href="#cb145-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>6</span>
<span id="cb145-12"><a href="#cb145-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb145-13"><a href="#cb145-13" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb145-14"><a href="#cb145-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb145-15"><a href="#cb145-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb145-16"><a href="#cb145-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb145-17"><a href="#cb145-17" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.FP@loc.n.all)</span></span>
<span id="cb145-18"><a href="#cb145-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb145-19"><a href="#cb145-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>21,24,8,37,43,26,19,16,43,31,40</span>
<span id="cb145-20"><a href="#cb145-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,</span>
<span id="cb145-21"><a href="#cb145-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,</span>
<span id="cb145-22"><a href="#cb145-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb145-23"><a href="#cb145-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb145-24"><a href="#cb145-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb145-25"><a href="#cb145-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb145-26"><a href="#cb145-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb145-27"><a href="#cb145-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb145-28"><a href="#cb145-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb145-29"><a href="#cb145-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,0</span>
<span id="cb145-30"><a href="#cb145-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>4,10000,1</span>
<span id="cb145-31"><a href="#cb145-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2</span>
<span id="cb145-32"><a href="#cb145-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb145-33"><a href="#cb145-33" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots= all1DProfiles</span></span>
<span id="cb145-34"><a href="#cb145-34" aria-hidden="true" tabindex="-1"></a><span class="ex">1DProfiles=2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb145-35"><a href="#cb145-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb145-36"><a href="#cb145-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb145-37"><a href="#cb145-37" aria-hidden="true" tabindex="-1"></a><span class="co">#writeAdHocFiles=T</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-9-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-9-3-tab">
<p>Removed 3 loci, used condS2 parameterization</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="va">GenepopFileName</span><span class="op">=</span>../../Daruanus_FP.txt</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="va">DemographicModel</span><span class="op">=</span> <span class="ex">LinearIBD</span></span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a><span class="co">#I modified the genepop file by adding sampling coordinates as the name of the </span></span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="co">#last individual in each population. There is no coastline for these samples,</span></span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a><span class="co"># which each come from different islands. I just measured from Tahiti to Maupiti</span></span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a><span class="co"># It's 65 km from the tip of Tahiti Iti to Puna, so I added that to </span></span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="co"># each coordinate.</span></span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a><span class="va">PSONMax</span><span class="op">=</span>355 <span class="ex">0</span></span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Neighborhood size is based on mean distance between populations = 75.78</span></span>
<span id="cb146-10"><a href="#cb146-10" aria-hidden="true" tabindex="-1"></a><span class="co">#355/75.78 = 4.684 so I will use 6 bins</span></span>
<span id="cb146-11"><a href="#cb146-11" aria-hidden="true" tabindex="-1"></a><span class="va">GeoBinNbr</span><span class="op">=</span>6</span>
<span id="cb146-12"><a href="#cb146-12" aria-hidden="true" tabindex="-1"></a><span class="va">GeoUnit</span><span class="op">=</span> <span class="ex">ind.km</span></span>
<span id="cb146-13"><a href="#cb146-13" aria-hidden="true" tabindex="-1"></a><span class="co">#alternate way of specifying the habitat, not used for now</span></span>
<span id="cb146-14"><a href="#cb146-14" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars= 0.5 0.5 400 1 0</span></span>
<span id="cb146-15"><a href="#cb146-15" aria-hidden="true" tabindex="-1"></a><span class="co">#habitatPars=0 0 0 300 0</span></span>
<span id="cb146-16"><a href="#cb146-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Mutation Model is K-allele = PIM, with k=2 for SNPs. GivenK is number of alleles</span></span>
<span id="cb146-17"><a href="#cb146-17" aria-hidden="true" tabindex="-1"></a><span class="co"># at each locus (Daruanus.FP@loc.n.all)</span></span>
<span id="cb146-18"><a href="#cb146-18" aria-hidden="true" tabindex="-1"></a><span class="va">MutationModel</span><span class="op">=</span>PIM</span>
<span id="cb146-19"><a href="#cb146-19" aria-hidden="true" tabindex="-1"></a><span class="va">GivenK</span><span class="op">=</span>21,8,37,26,19,16,43,40</span>
<span id="cb146-20"><a href="#cb146-20" aria-hidden="true" tabindex="-1"></a><span class="va">samplingSpace</span><span class="op">=</span>,,condS2</span>
<span id="cb146-21"><a href="#cb146-21" aria-hidden="true" tabindex="-1"></a><span class="va">samplingScale</span><span class="op">=</span>,,logscale</span>
<span id="cb146-22"><a href="#cb146-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis - this will do 5 runs of 100 points and</span></span>
<span id="cb146-23"><a href="#cb146-23" aria-hidden="true" tabindex="-1"></a><span class="co">#overwrite those with 10 runs of 250 points</span></span>
<span id="cb146-24"><a href="#cb146-24" aria-hidden="true" tabindex="-1"></a><span class="va">writeSequence</span><span class="op">=</span> <span class="ex">Over,Over,Over,Over,Over,Append,10</span></span>
<span id="cb146-25"><a href="#cb146-25" aria-hidden="true" tabindex="-1"></a><span class="va">StatisticSequence</span><span class="op">=</span>PAC</span>
<span id="cb146-26"><a href="#cb146-26" aria-hidden="true" tabindex="-1"></a><span class="va">PointNumber</span><span class="op">=</span>100,100,100,100,100,250</span>
<span id="cb146-27"><a href="#cb146-27" aria-hidden="true" tabindex="-1"></a><span class="va">Nrunsperpoint</span><span class="op">=</span>30,30,30,30,30,50</span>
<span id="cb146-28"><a href="#cb146-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Wide priors on Neu, Nem and g</span></span>
<span id="cb146-29"><a href="#cb146-29" aria-hidden="true" tabindex="-1"></a><span class="va">LowerBound</span><span class="op">=</span>0.1,1,1</span>
<span id="cb146-30"><a href="#cb146-30" aria-hidden="true" tabindex="-1"></a><span class="va">Upperbound</span><span class="op">=</span>4,10000,100000</span>
<span id="cb146-31"><a href="#cb146-31" aria-hidden="true" tabindex="-1"></a><span class="va">oneDimCI</span><span class="op">=</span> <span class="ex">2Nmu,</span> 2Nm, Nb, condS2, g</span>
<span id="cb146-32"><a href="#cb146-32" aria-hidden="true" tabindex="-1"></a><span class="va">CoreNbrForR</span><span class="op">=</span>4</span>
<span id="cb146-33"><a href="#cb146-33" aria-hidden="true" tabindex="-1"></a><span class="va">Plots</span><span class="op">=</span> <span class="ex">allProfiles</span></span>
<span id="cb146-34"><a href="#cb146-34" aria-hidden="true" tabindex="-1"></a><span class="co">#1DProfiles=2Nmu, 2Nm, Nb, condS2, g</span></span>
<span id="cb146-35"><a href="#cb146-35" aria-hidden="true" tabindex="-1"></a><span class="va">extrascale</span><span class="op">=</span>Nb=logscale</span>
<span id="cb146-36"><a href="#cb146-36" aria-hidden="true" tabindex="-1"></a><span class="va">graphicFormat</span><span class="op">=</span>pdf</span>
<span id="cb146-37"><a href="#cb146-37" aria-hidden="true" tabindex="-1"></a><span class="va">writeAdHocFiles</span><span class="op">=</span>T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finished in 56.45 minutes.</p>
</div>
</div>
</div>
</section>
<section id="create-dispersal-kernels-2" class="level3">
<h3 class="anchored" data-anchor-id="create-dispersal-kernels-2">Create Dispersal Kernels</h3>
<section id="sigma-estimates-1" class="level4">
<h4 class="anchored" data-anchor-id="sigma-estimates-1">Sigma estimates</h4>
<p>This got me the following estimates.</p>
<p>Output from run 5.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a>runDir <span class="ot">&lt;-</span> <span class="st">"./FP/Migraine/run5"</span></span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">read_migraine</span>(runDir)</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a>NS_FP <span class="ot">&lt;-</span> result[<span class="st">"NS"</span>]</span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a>NSCI_FP <span class="ot">&lt;-</span> <span class="fu">c</span>(result[<span class="st">"NSCI1"</span>],result[<span class="st">"NSCI2"</span>])</span>
<span id="cb147-6"><a href="#cb147-6" aria-hidden="true" tabindex="-1"></a>Nmu_FP <span class="ot">&lt;-</span> result[<span class="st">"Nmu"</span>]</span>
<span id="cb147-7"><a href="#cb147-7" aria-hidden="true" tabindex="-1"></a>Nm_FP <span class="ot">&lt;-</span> result[<span class="st">"Nm"</span>]</span>
<span id="cb147-8"><a href="#cb147-8" aria-hidden="true" tabindex="-1"></a>g_FP <span class="ot">&lt;-</span> result[<span class="st">"g"</span>]</span>
<span id="cb147-9"><a href="#cb147-9" aria-hidden="true" tabindex="-1"></a>lattice2geog_FP <span class="ot">&lt;-</span> result[<span class="st">"lattice2geog"</span>]</span>
<span id="cb147-10"><a href="#cb147-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-11"><a href="#cb147-11" aria-hidden="true" tabindex="-1"></a>sigma2_FP <span class="ot">&lt;-</span> <span class="fu">g_to_sigma2</span>(g_FP)</span>
<span id="cb147-12"><a href="#cb147-12" aria-hidden="true" tabindex="-1"></a>sigma_fromsigma2_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(sigma2_FP<span class="sc">*</span>lattice2geog_FP)</span>
<span id="cb147-13"><a href="#cb147-13" aria-hidden="true" tabindex="-1"></a>sigma_fromNS_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NS_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_FP))</span>
<span id="cb147-14"><a href="#cb147-14" aria-hidden="true" tabindex="-1"></a>sigmaCI_fromNS_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NSCI_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_FP))</span>
<span id="cb147-15"><a href="#cb147-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-16"><a href="#cb147-16" aria-hidden="true" tabindex="-1"></a>sigma_fromNS_all_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NS_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_all_FP))</span>
<span id="cb147-17"><a href="#cb147-17" aria-hidden="true" tabindex="-1"></a>sigmaCI_fromNS_all_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(NSCI_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_all_FP))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The <span class="math inline">\(\sigma\)</span> we get from <span class="math inline">\(F_{ST}\)</span> ~ Distance is 64.2, and from Neighborhood Size <span class="math inline">\(\sigma\)</span> is 88. We again get a lower estimate from <span class="math inline">\(\sigma^2\)</span>, with <span class="math inline">\(\sigma\)</span> = 15</p>
</section>
</section>
<section id="confidence-intervals-2" class="level3">
<h3 class="anchored" data-anchor-id="confidence-intervals-2">Confidence Intervals</h3>
<p>Propagating error sorta following Pinsky et al.&nbsp;table S2</p>
<section id="error-in-ne" class="level4">
<h4 class="anchored" data-anchor-id="error-in-ne">Error in Ne</h4>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-10-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-10-1" role="tab" aria-controls="tabset-10-1" aria-selected="true">For Harmonic Mean Method</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-10-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-10-2" role="tab" aria-controls="tabset-10-2" aria-selected="false">For Whole Sample Method</a></li></ul>
<div class="tab-content">
<div id="tabset-10-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-10-1-tab">
<p>Following <span class="citation" data-cites="pinskyUsingIsolationDistance2010">Pinsky, Montes, and Palumbi (<a href="#ref-pinskyUsingIsolationDistance2010" role="doc-biblioref">2010</a>)</span> I am going to bootstrap across the confidence intervals for each Nb estimate. Unfortunately, the new jackknife method of <span class="citation" data-cites="jonesImprovedConfidenceIntervals2016">Jones, Ovenden, and Wang (<a href="#ref-jonesImprovedConfidenceIntervals2016" role="doc-biblioref">2016</a>)</span> often results in infinite upper bounds with marine data. Iâ€™m going to use the jackknife CIs. Iâ€™m also going to use a uniform distribution for the error because approximating the error structure with ChiSq or Normal distributions is not a simple task and Iâ€™m just trying to get a sketch of the error to compare with MigraiNe anyway. Iâ€™m going to set â€œInfiniteâ€ values in the upper CI to 20,000 since I rarely see upper bounds that high.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP<span class="sc">$</span>JackknifeHigh[<span class="fu">which</span>(<span class="fu">is.na</span>(Ne_estimates_FP<span class="sc">$</span>JackknifeHigh))] <span class="ot">&lt;-</span> <span class="dv">20000</span></span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP<span class="sc">$</span>JackknifeHigh <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_FP<span class="sc">$</span>JackknifeHigh)</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>Ne_estimates_FP<span class="sc">$</span>JackknifeLow <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Ne_estimates_FP<span class="sc">$</span>JackknifeLow)</span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>Ne_error_FP <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(n <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100000</span>){</span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>  hm <span class="ot">&lt;-</span> <span class="fu">harm_mean</span>(<span class="fu">mapply</span>(runif, <span class="at">n=</span><span class="dv">1</span>, </span>
<span id="cb148-8"><a href="#cb148-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">min=</span>Ne_estimates_FP<span class="sc">$</span>JackknifeLow,</span>
<span id="cb148-9"><a href="#cb148-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max=</span>Ne_estimates_FP<span class="sc">$</span>JackknifeHigh))</span>
<span id="cb148-10"><a href="#cb148-10" aria-hidden="true" tabindex="-1"></a>  Ne_error_FP <span class="ot">&lt;-</span> <span class="fu">c</span>(Ne_error_FP,hm)</span>
<span id="cb148-11"><a href="#cb148-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb148-12"><a href="#cb148-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Ne_error_FP) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb148-13"><a href="#cb148-13" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(Ne_error_FP, "FP/Ne_error_FP.csv",quote=F,row.names = F)</span></span>
<span id="cb148-14"><a href="#cb148-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Ne_error_FP), <span class="fu">aes</span>(<span class="at">x=</span>Ne_error_FP)) <span class="sc">+</span> <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-77-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-10-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-10-2-tab">
<p>Naaykens and Dâ€™Aloia showed that using the whole sample to estimate density gives pretty similar results to the harmonic mean method, so Iâ€™m also going to try that.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>Nb <span class="ot">&lt;-</span> Ne_all_FP</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>r2_FP <span class="ot">&lt;-</span> Ne_estimates_FP300km<span class="sc">$</span>r2</span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="co">#from Waples 2006 table 2</span></span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a>er2_FP <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>Ne_estimates_FP300km<span class="sc">$</span>SampSize <span class="sc">+</span> <span class="fl">3.19</span><span class="sc">/</span>Ne_estimates_FP300km<span class="sc">$</span>SampSize <span class="sc">^</span><span class="dv">2</span> </span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a>df_FP <span class="ot">&lt;-</span> Ne_estimates_FP300km<span class="sc">$</span>IndAlleles</span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-7"><a href="#cb149-7" aria-hidden="true" tabindex="-1"></a><span class="fu">WaplesMonoNe</span>(r2_FP <span class="sc">-</span> er2_FP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11688.35</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this shows that we can get approximately what NeEstimator gives us... </span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="co">#not sure why its not exact... must be missing some correction</span></span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>rCI_FP <span class="ot">&lt;-</span> df_NC<span class="sc">*</span>r2_FP <span class="sc">/</span> (<span class="fu">qchisq</span>(<span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>), <span class="at">df =</span> df_FP))</span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a><span class="fu">WaplesMonoNe</span>(rCI_NC <span class="sc">-</span> er2_NC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3729.688 16288.111</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co">#and now to get and plot the error distribution</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>Ne_error_all_FP <span class="ot">&lt;-</span> <span class="fu">WaplesMonoNe</span>(((df_FP<span class="sc">*</span>r2_FP)<span class="sc">/</span>(<span class="fu">rchisq</span>(<span class="dv">100000</span>, <span class="at">df =</span> df_FP))) <span class="sc">-</span> er2_FP)</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Ne_error_all_FP), <span class="fu">aes</span>(<span class="at">x=</span>Ne_error_all_FP)) <span class="sc">+</span> </span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">30000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-78-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="error-in-effective-density-2" class="level4">
<h4 class="anchored" data-anchor-id="error-in-effective-density-2">Error in Effective Density</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>Ne_error_FP <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">"FP/Ne_error_FP.csv"</span>)<span class="sc">$</span>x <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>De_error_FP <span class="ot">&lt;-</span> Ne_error_FP <span class="sc">/</span><span class="fu">rnormTrunc</span>(<span class="dv">100000</span>, <span class="at">mean =</span> meandists_FP,</span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sd =</span> (<span class="dv">65-10</span>)<span class="sc">/</span><span class="fl">1.96</span>, <span class="at">min =</span> <span class="fl">1e-10</span>)</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>De_error_all_FP <span class="ot">&lt;-</span> Ne_error_all_FP <span class="sc">/</span> maxdist_FP</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(De_error_FP), <span class="fu">aes</span>(<span class="at">x=</span>De_error_FP)) <span class="sc">+</span></span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">1000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-79-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(De_error_all_FP), <span class="fu">aes</span>(<span class="at">x=</span>De_error_all_FP)) <span class="sc">+</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">1000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-79-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="error-in-slope-1" class="level4">
<h4 class="anchored" data-anchor-id="error-in-slope-1">Error in Slope</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>slope_se_FP <span class="ot">&lt;-</span> <span class="fu">summary</span>(lmodel_FP)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="fl">1e-6</span>,<span class="fl">1e-4</span>)), <span class="fu">aes</span>(<span class="at">x=</span>x)) <span class="sc">+</span> <span class="fu">stat_function</span>(<span class="at">fun=</span>dnormTrunc, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> slope_FP, <span class="at">sd=</span>slope_se_FP, <span class="at">min =</span> <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>slope_error_FP <span class="ot">&lt;-</span> <span class="fu">rnormTrunc</span>(<span class="dv">100000</span>, <span class="at">mean =</span> slope_FP, <span class="at">sd =</span> slope_se_FP,<span class="at">min =</span> <span class="fl">1e-10</span>)</span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(slope_error_Fiji), <span class="fu">aes</span>(<span class="at">x=</span>slope_error_Fiji)) <span class="sc">+</span> <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-80-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>sigma_error_fromSlope_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_error_FP<span class="sc">*</span>slope_error_FP))</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>sigma_error_fromSlope_all_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">4</span><span class="sc">*</span>De_error_all_FP<span class="sc">*</span>slope_error_FP))</span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromSlope_FP), <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromSlope_FP)) <span class="sc">+</span></span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">1000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-80-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromSlope_all_FP), <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromSlope_all_FP)) <span class="sc">+</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">1000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-80-4.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quantile</span>(sigma_error_fromSlope_FP, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     2.5%     97.5% 
 10.36456 100.46865 </code></pre>
</div>
</div>
</section>
<section id="error-in-neighborhood-size-2" class="level4">
<h4 class="anchored" data-anchor-id="error-in-neighborhood-size-2">Error in Neighborhood Size</h4>
<p>Using a uniform distribution is out because there is clearly a peaked distribution in the Migraine output. So I am using a quick fit to a truncated lognormal distribution</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figures/Da_FP_Neighborhood.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Migraine_Run2_Neighborhood_Theta</figcaption><p></p>
</figure>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>NSCI_FP</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       NSCI1        NSCI2 
       67763 531462708061 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">curve</span>(<span class="fu">dlnorm</span>(x, <span class="at">meanlog =</span> <span class="fu">log</span>(NS_FP), <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">7.9</span>), <span class="at">log =</span> T))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-81-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qlnorm</span>(<span class="fu">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>),<span class="at">meanlog =</span> <span class="fu">log</span>(NS_FP), <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="fl">206.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.062342e+01 1.260620e+10</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>Neighborhood_error_FP <span class="ot">&lt;-</span> <span class="fu">rlnormTrunc</span>(<span class="at">n =</span> <span class="dv">100000</span>, <span class="at">meanlog =</span> <span class="fu">log</span>(NS_FP), </span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">sdlog =</span> <span class="fu">log</span>(<span class="dv">210</span>), <span class="at">min =</span> NSCI_FP[<span class="dv">1</span>], <span class="at">max =</span> <span class="cn">Inf</span>)</span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(Neighborhood_error_FP), <span class="fu">aes</span>(<span class="at">x=</span>Neighborhood_error_FP)) <span class="sc">+</span> </span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-82-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a>sigma_error_fromNS_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(Neighborhood_error_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_error_FP))</span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a>sigma_error_fromNS_all_FP <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(Neighborhood_error_FP<span class="sc">/</span>(<span class="dv">4</span><span class="sc">*</span>De_error_all_FP))</span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-5"><a href="#cb168-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sigma_error_fromNS_FP) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb168-6"><a href="#cb168-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-7"><a href="#cb168-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">tibble</span>(sigma_error_fromNS_FP), </span>
<span id="cb168-8"><a href="#cb168-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>sigma_error_fromNS_FP)) <span class="sc">+</span></span>
<span id="cb168-9"><a href="#cb168-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-82-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="plot-dispersal-kernels-2" class="level4">
<h4 class="anchored" data-anchor-id="plot-dispersal-kernels-2">Plot Dispersal Kernels</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>kernelplot_FP <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)),<span class="fu">aes</span>(x)) <span class="sc">+</span> </span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromNS_FP,<span class="dv">1000</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb169-3"><a href="#cb169-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb169-4"><a href="#cb169-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"lightblue"</span>,                                       <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.2</span>) }) <span class="sc">+</span></span>
<span id="cb169-5"><a href="#cb169-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromSlope_FP,<span class="dv">1000</span>), </span>
<span id="cb169-6"><a href="#cb169-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb169-7"><a href="#cb169-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb169-8"><a href="#cb169-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">colour =</span> <span class="st">"lightgreen"</span>,                                       <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.2</span>) }) <span class="sc">+</span></span>
<span id="cb169-9"><a href="#cb169-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromSlope_FP), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb169-10"><a href="#cb169-10" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"IBD_Slope"</span>), <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb169-11"><a href="#cb169-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromSlope_all_FP), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb169-12"><a href="#cb169-12" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"IBD_Slope_1pop"</span>),</span>
<span id="cb169-13"><a href="#cb169-13" aria-hidden="true" tabindex="-1"></a>                                                          <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb169-14"><a href="#cb169-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromsigma2_FP), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb169-15"><a href="#cb169-15" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Sigma2"</span>), </span>
<span id="cb169-16"><a href="#cb169-16" aria-hidden="true" tabindex="-1"></a>                                                          <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb169-17"><a href="#cb169-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_FP), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb169-18"><a href="#cb169-18" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Neighborhood_Size"</span>), </span>
<span id="cb169-19"><a href="#cb169-19" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb169-20"><a href="#cb169-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_all_FP), <span class="at">linetype=</span><span class="dv">2</span>,</span>
<span id="cb169-21"><a href="#cb169-21" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Migraine_Neighborhood_Size_1pop"</span>),</span>
<span id="cb169-22"><a href="#cb169-22" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb169-23"><a href="#cb169-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">xlab</span>(<span class="st">"Alongshore Distance (km)"</span>) <span class="sc">+</span> </span>
<span id="cb169-24"><a href="#cb169-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">"Dispersal probability density"</span>) <span class="sc">+</span></span>
<span id="cb169-25"><a href="#cb169-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(<span class="st">"Kernel"</span>,<span class="at">values =</span> </span>
<span id="cb169-26"><a href="#cb169-26" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(<span class="at">IBD_Slope =</span> <span class="st">"green"</span>,</span>
<span id="cb169-27"><a href="#cb169-27" aria-hidden="true" tabindex="-1"></a>                     <span class="at">IBD_Slope_1pop =</span> <span class="st">"darkgreen"</span>,</span>
<span id="cb169-28"><a href="#cb169-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">Migraine_Sigma2=</span><span class="st">"darkblue"</span>, </span>
<span id="cb169-29"><a href="#cb169-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Migraine_Neighborhood_Size =</span><span class="st">"blue"</span>,</span>
<span id="cb169-30"><a href="#cb169-30" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Migraine_Neighborhood_Size_1pop =</span> <span class="st">"cyan4"</span>)) <span class="sc">+</span></span>
<span id="cb169-31"><a href="#cb169-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.1</span>)</span>
<span id="cb169-32"><a href="#cb169-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-33"><a href="#cb169-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-34"><a href="#cb169-34" aria-hidden="true" tabindex="-1"></a>kernelplot_FP</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-83-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="comparing-across-archipelagos" class="level1">
<h1>Comparing Across Archipelagos</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a>NC <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">Archipelago =</span> <span class="st">"New Caledonia"</span>,</span>
<span id="cb170-2"><a href="#cb170-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">Estimate =</span> <span class="fu">c</span>(<span class="st">"Mean sampling distance"</span>,</span>
<span id="cb170-3"><a href="#cb170-3" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Fst ~ Distance Slope"</span>,</span>
<span id="cb170-4"><a href="#cb170-4" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne Harmonic Mean"</span>,</span>
<span id="cb170-5"><a href="#cb170-5" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne as one population"</span>,</span>
<span id="cb170-6"><a href="#cb170-6" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density"</span>,</span>
<span id="cb170-7"><a href="#cb170-7" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density one pop"</span>,</span>
<span id="cb170-8"><a href="#cb170-8" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Theta"</span>,</span>
<span id="cb170-9"><a href="#cb170-9" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Nm"</span>,</span>
<span id="cb170-10"><a href="#cb170-10" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"g"</span>,</span>
<span id="cb170-11"><a href="#cb170-11" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size"</span>,</span>
<span id="cb170-12"><a href="#cb170-12" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size Low CI"</span>,</span>
<span id="cb170-13"><a href="#cb170-13" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size High CI"</span>,</span>
<span id="cb170-14"><a href="#cb170-14" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Bin Width"</span>,</span>
<span id="cb170-15"><a href="#cb170-15" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope"</span>,</span>
<span id="cb170-16"><a href="#cb170-16" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope; Ne as one pop"</span>,</span>
<span id="cb170-17"><a href="#cb170-17" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS"</span>,</span>
<span id="cb170-18"><a href="#cb170-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS; Ne as one pop"</span>,</span>
<span id="cb170-19"><a href="#cb170-19" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from sigma2"</span>),</span>
<span id="cb170-20"><a href="#cb170-20" aria-hidden="true" tabindex="-1"></a>             <span class="at">Values =</span> <span class="fu">c</span>(meandists_NC,</span>
<span id="cb170-21"><a href="#cb170-21" aria-hidden="true" tabindex="-1"></a>                        slope_NC,</span>
<span id="cb170-22"><a href="#cb170-22" aria-hidden="true" tabindex="-1"></a>                        Ne_hm_NC,</span>
<span id="cb170-23"><a href="#cb170-23" aria-hidden="true" tabindex="-1"></a>                        Ne_all_NC,</span>
<span id="cb170-24"><a href="#cb170-24" aria-hidden="true" tabindex="-1"></a>                        De_NC,</span>
<span id="cb170-25"><a href="#cb170-25" aria-hidden="true" tabindex="-1"></a>                        De_all_NC,</span>
<span id="cb170-26"><a href="#cb170-26" aria-hidden="true" tabindex="-1"></a>                        Nmu_NC,</span>
<span id="cb170-27"><a href="#cb170-27" aria-hidden="true" tabindex="-1"></a>                        Nm_NC,</span>
<span id="cb170-28"><a href="#cb170-28" aria-hidden="true" tabindex="-1"></a>                        g_NC,</span>
<span id="cb170-29"><a href="#cb170-29" aria-hidden="true" tabindex="-1"></a>                        NS_NC,</span>
<span id="cb170-30"><a href="#cb170-30" aria-hidden="true" tabindex="-1"></a>                        NSCI_NC[<span class="dv">1</span>],</span>
<span id="cb170-31"><a href="#cb170-31" aria-hidden="true" tabindex="-1"></a>                        NSCI_NC[<span class="dv">2</span>],</span>
<span id="cb170-32"><a href="#cb170-32" aria-hidden="true" tabindex="-1"></a>                        lattice2geog_NC,</span>
<span id="cb170-33"><a href="#cb170-33" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">NA</span>,</span>
<span id="cb170-34"><a href="#cb170-34" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">NA</span>,</span>
<span id="cb170-35"><a href="#cb170-35" aria-hidden="true" tabindex="-1"></a>                        sigma_fromNS_NC,</span>
<span id="cb170-36"><a href="#cb170-36" aria-hidden="true" tabindex="-1"></a>                        sigma_fromNS_all_NC,</span>
<span id="cb170-37"><a href="#cb170-37" aria-hidden="true" tabindex="-1"></a>                        sigma_fromsigma2_NC</span>
<span id="cb170-38"><a href="#cb170-38" aria-hidden="true" tabindex="-1"></a>                        ))</span>
<span id="cb170-39"><a href="#cb170-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-40"><a href="#cb170-40" aria-hidden="true" tabindex="-1"></a>Fiji <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">Archipelago =</span> <span class="st">"Fiji"</span>,</span>
<span id="cb170-41"><a href="#cb170-41" aria-hidden="true" tabindex="-1"></a>            <span class="at">Estimate =</span> <span class="fu">c</span>(<span class="st">"Mean sampling distance"</span>,</span>
<span id="cb170-42"><a href="#cb170-42" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Fst ~ Distance Slope"</span>,</span>
<span id="cb170-43"><a href="#cb170-43" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne Harmonic Mean"</span>,</span>
<span id="cb170-44"><a href="#cb170-44" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne as one population"</span>,</span>
<span id="cb170-45"><a href="#cb170-45" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density"</span>,</span>
<span id="cb170-46"><a href="#cb170-46" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density one pop"</span>,</span>
<span id="cb170-47"><a href="#cb170-47" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Theta"</span>,</span>
<span id="cb170-48"><a href="#cb170-48" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Nm"</span>,</span>
<span id="cb170-49"><a href="#cb170-49" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"g"</span>,</span>
<span id="cb170-50"><a href="#cb170-50" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size"</span>,</span>
<span id="cb170-51"><a href="#cb170-51" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size Low CI"</span>,</span>
<span id="cb170-52"><a href="#cb170-52" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size High CI"</span>,</span>
<span id="cb170-53"><a href="#cb170-53" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Bin Width"</span>,</span>
<span id="cb170-54"><a href="#cb170-54" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope"</span>,</span>
<span id="cb170-55"><a href="#cb170-55" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope; Ne as one pop"</span>,</span>
<span id="cb170-56"><a href="#cb170-56" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS"</span>,</span>
<span id="cb170-57"><a href="#cb170-57" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS; Ne as one pop"</span>,</span>
<span id="cb170-58"><a href="#cb170-58" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from sigma2"</span>),</span>
<span id="cb170-59"><a href="#cb170-59" aria-hidden="true" tabindex="-1"></a>             <span class="at">Values =</span> <span class="fu">c</span>(meandists_Fiji,</span>
<span id="cb170-60"><a href="#cb170-60" aria-hidden="true" tabindex="-1"></a>                        slope_Fiji,</span>
<span id="cb170-61"><a href="#cb170-61" aria-hidden="true" tabindex="-1"></a>                        Ne_hm_Fiji,</span>
<span id="cb170-62"><a href="#cb170-62" aria-hidden="true" tabindex="-1"></a>                        Ne_all_Fiji,</span>
<span id="cb170-63"><a href="#cb170-63" aria-hidden="true" tabindex="-1"></a>                        De_Fiji,</span>
<span id="cb170-64"><a href="#cb170-64" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">NA</span>,</span>
<span id="cb170-65"><a href="#cb170-65" aria-hidden="true" tabindex="-1"></a>                        Nmu_Fiji,</span>
<span id="cb170-66"><a href="#cb170-66" aria-hidden="true" tabindex="-1"></a>                        Nm_Fiji,</span>
<span id="cb170-67"><a href="#cb170-67" aria-hidden="true" tabindex="-1"></a>                        g_Fiji,</span>
<span id="cb170-68"><a href="#cb170-68" aria-hidden="true" tabindex="-1"></a>                        NS_Fiji,</span>
<span id="cb170-69"><a href="#cb170-69" aria-hidden="true" tabindex="-1"></a>                        NSCI_Fiji[<span class="dv">1</span>],</span>
<span id="cb170-70"><a href="#cb170-70" aria-hidden="true" tabindex="-1"></a>                        NSCI_Fiji[<span class="dv">2</span>],</span>
<span id="cb170-71"><a href="#cb170-71" aria-hidden="true" tabindex="-1"></a>                        lattice2geog_Fiji,</span>
<span id="cb170-72"><a href="#cb170-72" aria-hidden="true" tabindex="-1"></a>                        sigma_fromSlope_Fiji,</span>
<span id="cb170-73"><a href="#cb170-73" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">NA</span>,</span>
<span id="cb170-74"><a href="#cb170-74" aria-hidden="true" tabindex="-1"></a>                        sigma_fromNS_Fiji,</span>
<span id="cb170-75"><a href="#cb170-75" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">NA</span>,</span>
<span id="cb170-76"><a href="#cb170-76" aria-hidden="true" tabindex="-1"></a>                        sigma_fromsigma2_Fiji</span>
<span id="cb170-77"><a href="#cb170-77" aria-hidden="true" tabindex="-1"></a>                        ))</span>
<span id="cb170-78"><a href="#cb170-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-79"><a href="#cb170-79" aria-hidden="true" tabindex="-1"></a>FP <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">Archipelago =</span> <span class="st">"Societies"</span>,</span>
<span id="cb170-80"><a href="#cb170-80" aria-hidden="true" tabindex="-1"></a>            <span class="at">Estimate =</span> <span class="fu">c</span>(<span class="st">"Mean sampling distance"</span>,</span>
<span id="cb170-81"><a href="#cb170-81" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Fst ~ Distance Slope"</span>,</span>
<span id="cb170-82"><a href="#cb170-82" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne Harmonic Mean"</span>,</span>
<span id="cb170-83"><a href="#cb170-83" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Ne as one population"</span>,</span>
<span id="cb170-84"><a href="#cb170-84" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density"</span>,</span>
<span id="cb170-85"><a href="#cb170-85" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Effective Density one pop"</span>,</span>
<span id="cb170-86"><a href="#cb170-86" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Theta"</span>,</span>
<span id="cb170-87"><a href="#cb170-87" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Nm"</span>,</span>
<span id="cb170-88"><a href="#cb170-88" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"g"</span>,</span>
<span id="cb170-89"><a href="#cb170-89" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size"</span>,</span>
<span id="cb170-90"><a href="#cb170-90" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size Low CI"</span>,</span>
<span id="cb170-91"><a href="#cb170-91" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Neighborhood Size High CI"</span>,</span>
<span id="cb170-92"><a href="#cb170-92" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Bin Width"</span>,</span>
<span id="cb170-93"><a href="#cb170-93" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope"</span>,</span>
<span id="cb170-94"><a href="#cb170-94" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from slope; Ne as one pop"</span>,</span>
<span id="cb170-95"><a href="#cb170-95" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS"</span>,</span>
<span id="cb170-96"><a href="#cb170-96" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from NS; Ne as one pop"</span>,</span>
<span id="cb170-97"><a href="#cb170-97" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"sigma from sigma2"</span>),</span>
<span id="cb170-98"><a href="#cb170-98" aria-hidden="true" tabindex="-1"></a>             <span class="at">Values =</span> <span class="fu">c</span>(meandists_FP,</span>
<span id="cb170-99"><a href="#cb170-99" aria-hidden="true" tabindex="-1"></a>                        slope_FP,</span>
<span id="cb170-100"><a href="#cb170-100" aria-hidden="true" tabindex="-1"></a>                        Ne_hm_FP,</span>
<span id="cb170-101"><a href="#cb170-101" aria-hidden="true" tabindex="-1"></a>                        Ne_all_FP,</span>
<span id="cb170-102"><a href="#cb170-102" aria-hidden="true" tabindex="-1"></a>                        De_FP,</span>
<span id="cb170-103"><a href="#cb170-103" aria-hidden="true" tabindex="-1"></a>                        De_all_FP,</span>
<span id="cb170-104"><a href="#cb170-104" aria-hidden="true" tabindex="-1"></a>                        Nmu_FP,</span>
<span id="cb170-105"><a href="#cb170-105" aria-hidden="true" tabindex="-1"></a>                        Nm_FP,</span>
<span id="cb170-106"><a href="#cb170-106" aria-hidden="true" tabindex="-1"></a>                        g_FP,</span>
<span id="cb170-107"><a href="#cb170-107" aria-hidden="true" tabindex="-1"></a>                        NS_FP,</span>
<span id="cb170-108"><a href="#cb170-108" aria-hidden="true" tabindex="-1"></a>                        NSCI_FP[<span class="dv">1</span>],</span>
<span id="cb170-109"><a href="#cb170-109" aria-hidden="true" tabindex="-1"></a>                        NSCI_FP[<span class="dv">2</span>],</span>
<span id="cb170-110"><a href="#cb170-110" aria-hidden="true" tabindex="-1"></a>                        lattice2geog_FP,</span>
<span id="cb170-111"><a href="#cb170-111" aria-hidden="true" tabindex="-1"></a>                        sigma_fromSlope_FP,</span>
<span id="cb170-112"><a href="#cb170-112" aria-hidden="true" tabindex="-1"></a>                        sigma_fromSlope_all_FP,</span>
<span id="cb170-113"><a href="#cb170-113" aria-hidden="true" tabindex="-1"></a>                        sigma_fromNS_FP,</span>
<span id="cb170-114"><a href="#cb170-114" aria-hidden="true" tabindex="-1"></a>                        sigma_fromNS_all_FP,</span>
<span id="cb170-115"><a href="#cb170-115" aria-hidden="true" tabindex="-1"></a>                        sigma_fromsigma2_FP</span>
<span id="cb170-116"><a href="#cb170-116" aria-hidden="true" tabindex="-1"></a>                        ))            </span>
<span id="cb170-117"><a href="#cb170-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-118"><a href="#cb170-118" aria-hidden="true" tabindex="-1"></a>across.archipelagos <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(NC,Fiji,FP)</span>
<span id="cb170-119"><a href="#cb170-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-120"><a href="#cb170-120" aria-hidden="true" tabindex="-1"></a>across.archipelagos.df <span class="ot">&lt;-</span> <span class="fu">dcast</span>(across.archipelagos,Archipelago<span class="sc">~</span>Estimate)</span>
<span id="cb170-121"><a href="#cb170-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-122"><a href="#cb170-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-123"><a href="#cb170-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-124"><a href="#cb170-124" aria-hidden="true" tabindex="-1"></a><span class="co">#write_csv(across.archipelagos.df,"Across_archipelago_statistics.csv")</span></span>
<span id="cb170-125"><a href="#cb170-125" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb170-126"><a href="#cb170-126" aria-hidden="true" tabindex="-1"></a>errors <span class="ot">&lt;-</span> <span class="fu">enframe</span>(<span class="fu">c</span>(<span class="at">NC_NS =</span> sigma_error_fromNS_NC,</span>
<span id="cb170-127"><a href="#cb170-127" aria-hidden="true" tabindex="-1"></a>                    <span class="at">NC_NS_all =</span> sigma_error_fromNS_all_NC,</span>
<span id="cb170-128"><a href="#cb170-128" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Fiji_NS =</span> sigma_error_fromNS_Fiji,</span>
<span id="cb170-129"><a href="#cb170-129" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Fiji_Slope =</span> sigma_error_fromSlope_Fiji,</span>
<span id="cb170-130"><a href="#cb170-130" aria-hidden="true" tabindex="-1"></a>                    <span class="at">FP_NS =</span> sigma_error_fromNS_FP, </span>
<span id="cb170-131"><a href="#cb170-131" aria-hidden="true" tabindex="-1"></a>                    <span class="at">FP_NS_all =</span> sigma_error_fromNS_all_FP,</span>
<span id="cb170-132"><a href="#cb170-132" aria-hidden="true" tabindex="-1"></a>                    <span class="at">FP_Slope =</span> sigma_error_fromSlope_FP,</span>
<span id="cb170-133"><a href="#cb170-133" aria-hidden="true" tabindex="-1"></a>                    <span class="at">FP_Slope_all =</span> sigma_error_fromSlope_all_FP),</span>
<span id="cb170-134"><a href="#cb170-134" aria-hidden="true" tabindex="-1"></a>                  <span class="at">name =</span> <span class="st">"Archipelago"</span>, </span>
<span id="cb170-135"><a href="#cb170-135" aria-hidden="true" tabindex="-1"></a>                  <span class="at">value =</span> <span class="st">"Sigma"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb170-136"><a href="#cb170-136" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">Archipelago =</span> <span class="fu">str_remove</span>(Archipelago, <span class="at">pattern=</span><span class="st">"</span><span class="sc">\\</span><span class="st">d+$"</span>))</span>
<span id="cb170-137"><a href="#cb170-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-138"><a href="#cb170-138" aria-hidden="true" tabindex="-1"></a>violins <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(errors, <span class="fu">aes</span>(<span class="at">x =</span> Archipelago, <span class="at">y =</span> Sigma)) <span class="sc">+</span> <span class="fu">geom_violin</span>() <span class="sc">+</span> </span>
<span id="cb170-139"><a href="#cb170-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">1000</span>)) <span class="sc">+</span> </span>
<span id="cb170-140"><a href="#cb170-140" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_point(data = across.archipelagos, </span></span>
<span id="cb170-141"><a href="#cb170-141" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           mapping = aes(x = "Archipelago", y = "Sigma")) +</span></span>
<span id="cb170-142"><a href="#cb170-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>()</span>
<span id="cb170-143"><a href="#cb170-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-144"><a href="#cb170-144" aria-hidden="true" tabindex="-1"></a>boxes <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(errors, <span class="fu">aes</span>(<span class="at">x =</span> Archipelago, <span class="at">y =</span> Sigma)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb170-145"><a href="#cb170-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">1000</span>)) <span class="sc">+</span> </span>
<span id="cb170-146"><a href="#cb170-146" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_point(data = across.archipelagos, </span></span>
<span id="cb170-147"><a href="#cb170-147" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           mapping = aes(x = "Archipelago", y = "Sigma")) +</span></span>
<span id="cb170-148"><a href="#cb170-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="make-figures-for-the-proposal" class="level2">
<h2 class="anchored" data-anchor-id="make-figures-for-the-proposal">Make figures for the proposal</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a>errors.p <span class="ot">&lt;-</span> <span class="fu">enframe</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">New Caledonia</span><span class="st">`</span> <span class="ot">=</span> sigma_error_fromNS_NC,</span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Fiji =</span> sigma_error_fromSlope_Fiji,</span>
<span id="cb171-3"><a href="#cb171-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Societies =</span> sigma_error_fromSlope_FP),</span>
<span id="cb171-4"><a href="#cb171-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">name =</span> <span class="st">"Archipelago"</span>, </span>
<span id="cb171-5"><a href="#cb171-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">value =</span> <span class="st">"Sigma"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb171-6"><a href="#cb171-6" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">Archipelago =</span> <span class="fu">str_remove</span>(Archipelago, <span class="at">pattern=</span><span class="st">"</span><span class="sc">\\</span><span class="st">d+$"</span>))</span>
<span id="cb171-7"><a href="#cb171-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-8"><a href="#cb171-8" aria-hidden="true" tabindex="-1"></a><span class="co">#create a tibble with the point estimates of interest</span></span>
<span id="cb171-9"><a href="#cb171-9" aria-hidden="true" tabindex="-1"></a>point_estimates <span class="ot">&lt;-</span> across.archipelagos.df <span class="sc">%&gt;%</span>  <span class="fu">select</span>(<span class="st">"Archipelago"</span>,<span class="st">"sigma from slope"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb171-10"><a href="#cb171-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">sigma from slope</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">replace</span>(<span class="st">`</span><span class="at">sigma from slope</span><span class="st">`</span>,<span class="dv">2</span>,</span>
<span id="cb171-11"><a href="#cb171-11" aria-hidden="true" tabindex="-1"></a>                                      across.archipelagos.df<span class="sc">$</span><span class="st">`</span><span class="at">sigma from NS</span><span class="st">`</span>[<span class="dv">2</span>])) </span>
<span id="cb171-12"><a href="#cb171-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-13"><a href="#cb171-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-14"><a href="#cb171-14" aria-hidden="true" tabindex="-1"></a>violins.p <span class="ot">&lt;-</span> errors.p <span class="sc">%&gt;%</span> <span class="fu">remove_missing</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb171-15"><a href="#cb171-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Archipelago =</span> <span class="fu">factor</span>(Archipelago,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"New Caledonia"</span>,</span>
<span id="cb171-16"><a href="#cb171-16" aria-hidden="true" tabindex="-1"></a>                                                     <span class="st">"Fiji"</span>,<span class="st">"Societies"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb171-17"><a href="#cb171-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> Archipelago, <span class="at">y =</span> Sigma)) <span class="sc">+</span> </span>
<span id="cb171-18"><a href="#cb171-18" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_point</span>(<span class="at">data =</span> point_estimates, </span>
<span id="cb171-19"><a href="#cb171-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> Archipelago, <span class="at">y =</span> <span class="st">`</span><span class="at">sigma from slope</span><span class="st">`</span>), </span>
<span id="cb171-20"><a href="#cb171-20" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"grey"</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb171-21"><a href="#cb171-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1000</span>) <span class="sc">+</span></span>
<span id="cb171-22"><a href="#cb171-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1000</span>)) <span class="sc">+</span> </span>
<span id="cb171-23"><a href="#cb171-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>()</span>
<span id="cb171-24"><a href="#cb171-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-25"><a href="#cb171-25" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Daruanus_archipelagos_violins.pdf",violins.p)</span></span>
<span id="cb171-26"><a href="#cb171-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-27"><a href="#cb171-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-28"><a href="#cb171-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-29"><a href="#cb171-29" aria-hidden="true" tabindex="-1"></a>kernelplot_archipelagos <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">250</span>)),<span class="fu">aes</span>(x)) <span class="sc">+</span> </span>
<span id="cb171-30"><a href="#cb171-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromNS_NC,<span class="dv">500</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb171-31"><a href="#cb171-31" aria-hidden="true" tabindex="-1"></a>          <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb171-32"><a href="#cb171-32" aria-hidden="true" tabindex="-1"></a>                  <span class="at">colour =</span> <span class="st">"palevioletred"</span>,</span>
<span id="cb171-33"><a href="#cb171-33" aria-hidden="true" tabindex="-1"></a>                  <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.1</span>) }) <span class="sc">+</span></span>
<span id="cb171-34"><a href="#cb171-34" aria-hidden="true" tabindex="-1"></a>   <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromSlope_Fiji,<span class="dv">500</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb171-35"><a href="#cb171-35" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb171-36"><a href="#cb171-36" aria-hidden="true" tabindex="-1"></a>                 <span class="at">colour =</span> <span class="st">"goldenrod"</span>, </span>
<span id="cb171-37"><a href="#cb171-37" aria-hidden="true" tabindex="-1"></a>                 <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.1</span>) }) <span class="sc">+</span></span>
<span id="cb171-38"><a href="#cb171-38" aria-hidden="true" tabindex="-1"></a>   <span class="fu">map</span>(<span class="at">.x =</span> <span class="fu">sample</span>(sigma_error_fromSlope_FP,<span class="dv">500</span>), <span class="at">.f =</span> <span class="cf">function</span>(sigma){</span>
<span id="cb171-39"><a href="#cb171-39" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stat_function</span>(<span class="at">fun =</span> dexp, <span class="at">args =</span> <span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma),</span>
<span id="cb171-40"><a href="#cb171-40" aria-hidden="true" tabindex="-1"></a>                 <span class="at">colour =</span> <span class="st">"lightblue"</span>,</span>
<span id="cb171-41"><a href="#cb171-41" aria-hidden="true" tabindex="-1"></a>                 <span class="at">linetype=</span><span class="dv">1</span>,<span class="at">size=</span><span class="fl">0.1</span>,<span class="at">alpha =</span> <span class="fl">0.1</span>) }) <span class="sc">+</span></span>
<span id="cb171-42"><a href="#cb171-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromNS_NC), <span class="at">linetype=</span><span class="dv">1</span>,</span>
<span id="cb171-43"><a href="#cb171-43" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"New Caledonia"</span>), <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb171-44"><a href="#cb171-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromSlope_Fiji), <span class="at">linetype=</span><span class="dv">1</span>,</span>
<span id="cb171-45"><a href="#cb171-45" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Fiji"</span>), </span>
<span id="cb171-46"><a href="#cb171-46" aria-hidden="true" tabindex="-1"></a>                                                          <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb171-47"><a href="#cb171-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(<span class="at">fun=</span>dexp,<span class="at">args=</span><span class="fu">list</span>(<span class="at">rate =</span> <span class="dv">1</span><span class="sc">/</span>sigma_fromSlope_FP), <span class="at">linetype=</span><span class="dv">1</span>,</span>
<span id="cb171-48"><a href="#cb171-48" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">aes</span>(<span class="at">color=</span><span class="st">"Societies"</span>), </span>
<span id="cb171-49"><a href="#cb171-49" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">show.legend =</span> T) <span class="sc">+</span></span>
<span id="cb171-50"><a href="#cb171-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Alongshore Distance (km)"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Dispersal probability density"</span>) <span class="sc">+</span></span>
<span id="cb171-51"><a href="#cb171-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"Archipelago"</span>,<span class="at">values =</span> </span>
<span id="cb171-52"><a href="#cb171-52" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(<span class="st">`</span><span class="at">New Caledonia</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"red"</span>,</span>
<span id="cb171-53"><a href="#cb171-53" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Fiji =</span> <span class="st">"orange"</span>,</span>
<span id="cb171-54"><a href="#cb171-54" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Societies=</span><span class="st">"blue"</span> </span>
<span id="cb171-55"><a href="#cb171-55" aria-hidden="true" tabindex="-1"></a>                   )) <span class="sc">+</span></span>
<span id="cb171-56"><a href="#cb171-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.02</span>)</span>
<span id="cb171-57"><a href="#cb171-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-58"><a href="#cb171-58" aria-hidden="true" tabindex="-1"></a>kernelplot_archipelagos</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-85-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Darunaus_archipelagos_kernels.pdf", kernelplot_archipelagos)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-11-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-11-1" role="tab" aria-controls="tabset-11-1" aria-selected="true">Violin Plots</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-11-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-11-2" role="tab" aria-controls="tabset-11-2" aria-selected="false">Box Plots</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-11-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-11-3" role="tab" aria-controls="tabset-11-3" aria-selected="false">Final Figure</a></li></ul>
<div class="tab-content">
<div id="tabset-11-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-11-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a>violins</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-86-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-11-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-11-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>boxes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-87-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-11-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-11-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>kernelplot_archipelagos</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-88-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="treemix" class="level1">
<h1>Treemix</h1>
<p>I want to check out what Treemix can do with this dataset, and see if it has any insight into directionality. The treemix manual says:</p>
<blockquote class="blockquote">
<p>the entries for each population are the mean length, variance in length, and number of haplotypes at each microsatellite in each population, again comma-delimited.</p>
</blockquote>
<section id="convert-data" class="level2">
<h2 class="anchored" data-anchor-id="convert-data">Convert data</h2>
<p>Here is code to convert genepop format to treemix format:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>pops <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PNG"</span>,<span class="st">"New Caledonia"</span>,<span class="st">"Fiji"</span>, <span class="st">"Societies"</span>,<span class="st">"Tuamotus"</span>)</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>loci <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Locus_1"</span>,<span class="st">"Locus_2"</span>,<span class="st">"Locus_3"</span>,<span class="st">"Locus_4"</span>,<span class="st">"Locus_5"</span>,<span class="st">"Locus_6"</span>,</span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>       <span class="st">"Locus_7"</span>,<span class="st">"Locus_8"</span>)</span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-5"><a href="#cb176-5" aria-hidden="true" tabindex="-1"></a>Daruanus4pop <span class="ot">&lt;-</span> <span class="fu">read.genepop</span>(<span class="st">"Daruanus/Daruanus_All_8locus_5pops.gen"</span>, </span>
<span id="cb176-6"><a href="#cb176-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncode =</span> 3L)</span>
<span id="cb176-7"><a href="#cb176-7" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(Daruanus4pop<span class="sc">@</span>pop)<span class="ot">&lt;-</span> pops</span>
<span id="cb176-8"><a href="#cb176-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-9"><a href="#cb176-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-10"><a href="#cb176-10" aria-hidden="true" tabindex="-1"></a>Daruanus4pop.gtypes <span class="ot">&lt;-</span> <span class="fu">genind2gtypes</span>(Daruanus4pop)</span>
<span id="cb176-11"><a href="#cb176-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-12"><a href="#cb176-12" aria-hidden="true" tabindex="-1"></a>popmatrix <span class="ot">&lt;-</span> <span class="fu">tibble</span>(Daruanus4pop.gtypes<span class="sc">@</span>data) <span class="sc">%&gt;%</span> </span>
<span id="cb176-13"><a href="#cb176-13" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">mutate</span>(<span class="at">allele =</span> <span class="fu">as.numeric</span>(allele)) <span class="sc">%&gt;%</span> </span>
<span id="cb176-14"><a href="#cb176-14" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">group_by</span>(stratum,locus)</span>
<span id="cb176-15"><a href="#cb176-15" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb176-16"><a href="#cb176-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-17"><a href="#cb176-17" aria-hidden="true" tabindex="-1"></a>popsumlong <span class="ot">&lt;-</span> popmatrix  <span class="sc">%&gt;%</span>  <span class="fu">summarize</span>(<span class="at">mean =</span> <span class="fu">mean</span>(allele,<span class="at">na.rm=</span>T), </span>
<span id="cb176-18"><a href="#cb176-18" aria-hidden="true" tabindex="-1"></a>                      <span class="at">var =</span> <span class="fu">var</span>(allele, <span class="at">na.rm=</span>T),</span>
<span id="cb176-19"><a href="#cb176-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">n =</span> <span class="fu">n</span>())</span>
<span id="cb176-20"><a href="#cb176-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-21"><a href="#cb176-21" aria-hidden="true" tabindex="-1"></a>popsumwide <span class="ot">&lt;-</span> popsumlong <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> stratum, </span>
<span id="cb176-22"><a href="#cb176-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_from =</span> <span class="fu">c</span>(mean,var,n))</span>
<span id="cb176-23"><a href="#cb176-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-24"><a href="#cb176-24" aria-hidden="true" tabindex="-1"></a>popsumwide <span class="ot">&lt;-</span> popsumwide[,<span class="fu">c</span>(<span class="st">"mean_PNG"</span>,<span class="st">"var_PNG"</span>,<span class="st">"n_PNG"</span>,</span>
<span id="cb176-25"><a href="#cb176-25" aria-hidden="true" tabindex="-1"></a>               <span class="st">"mean_New Caledonia"</span>,<span class="st">"var_New Caledonia"</span>,</span>
<span id="cb176-26"><a href="#cb176-26" aria-hidden="true" tabindex="-1"></a>               <span class="st">"n_New Caledonia"</span>,</span>
<span id="cb176-27"><a href="#cb176-27" aria-hidden="true" tabindex="-1"></a>               <span class="st">"mean_Fiji"</span>,<span class="st">"var_Fiji"</span>,<span class="st">"n_Fiji"</span>,</span>
<span id="cb176-28"><a href="#cb176-28" aria-hidden="true" tabindex="-1"></a>               <span class="st">"mean_Societies"</span>,<span class="st">"var_Societies"</span>,<span class="st">"n_Societies"</span>,</span>
<span id="cb176-29"><a href="#cb176-29" aria-hidden="true" tabindex="-1"></a>                <span class="st">"mean_Tuamotus"</span>,<span class="st">"var_Tuamotus"</span>,<span class="st">"n_Tuamotus"</span>)]</span>
<span id="cb176-30"><a href="#cb176-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-31"><a href="#cb176-31" aria-hidden="true" tabindex="-1"></a><span class="co">#write_csv(popsumwide,"Daruanus/treemix/Daruanus_treemix.txt")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Did a search and replace in <code>bbedit</code> to make the final file format, and then gzipped it.</p>
<pre><code>Search: (,[\d\.]+,\d+),
Replace \1 </code></pre>
</section>
<section id="explore-treemix" class="level2">
<h2 class="anchored" data-anchor-id="explore-treemix">Explore Treemix</h2>
<p>Installed <code>Treemix</code> on Argonaute via Mamba. Had to follow <a href="https://stackoverflow.com/questions/22222666/error-while-loading-shared-libraries-libgsl-so-0-cannot-open-shared-object-fil">these</a> instructions to add <code>libgsl.so.25</code> to <code>/etc/ld.so.conf</code> first</p>
<p>Iâ€™m first going to try a variety of m values and root at New Caledonia and Tuamotus.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="kw">(</span><span class="ex">base</span><span class="kw">)</span> <span class="ex">eric@Saraswati</span> ~/github/IBD_Kernels/Daruanus/treemix$ treemix <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_0m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span>                                                                                          </span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a><span class="ex">TreeMix</span> v. 1.13</span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a><span class="va">$Revision</span><span class="ex">:</span> 231 $</span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a><span class="ex">npop:5</span> nsnp:8</span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a><span class="ex">Estimating</span> covariance matrix in 8 blocks of size 1</span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a><span class="ex">SEED:</span> 1670724878</span>
<span id="cb178-9"><a href="#cb178-9" aria-hidden="true" tabindex="-1"></a><span class="ex">Starting</span> from:</span>
<span id="cb178-10"><a href="#cb178-10" aria-hidden="true" tabindex="-1"></a><span class="kw">((</span><span class="ex">Fiji:4.44185,Tuamotus:7.32221</span><span class="kw">)</span><span class="ex">:0,Societies:0</span><span class="kw">);</span></span>
<span id="cb178-11"><a href="#cb178-11" aria-hidden="true" tabindex="-1"></a><span class="ex">Adding</span> PNG [4/5]</span>
<span id="cb178-12"><a href="#cb178-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span><span class="er">(</span><span class="ex">likelihood</span><span class="kw">)</span><span class="bu">:</span> <span class="at">-17.411228</span> </span>
<span id="cb178-13"><a href="#cb178-13" aria-hidden="true" tabindex="-1"></a><span class="kw">(</span><span class="ex">PNG:5.50807,</span><span class="er">(</span><span class="ex">Fiji:1.12789,</span><span class="er">(</span><span class="ex">Tuamotus:4.74938,Societies:0</span><span class="kw">)</span><span class="ex">:5.11134</span><span class="kw">)</span><span class="ex">:5.50807</span><span class="kw">);</span></span>
<span id="cb178-14"><a href="#cb178-14" aria-hidden="true" tabindex="-1"></a><span class="ex">Adding</span> NewCaledonia [5/5]</span>
<span id="cb178-15"><a href="#cb178-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span><span class="er">(</span><span class="ex">likelihood</span><span class="kw">)</span><span class="bu">:</span> <span class="at">-25.594418</span> </span>
<span id="cb178-16"><a href="#cb178-16" aria-hidden="true" tabindex="-1"></a><span class="kw">((</span><span class="ex">Fiji:1.60844,</span><span class="er">(</span><span class="ex">PNG:6.74548,NewCaledonia:0</span><span class="kw">)</span><span class="ex">:3.60694</span><span class="kw">)</span><span class="ex">:2.43321,</span><span class="er">(</span><span class="ex">Tuamotus:4.78333,Societies:0</span><span class="kw">)</span><span class="ex">:2.43321</span><span class="kw">);</span></span>
<span id="cb178-17"><a href="#cb178-17" aria-hidden="true" tabindex="-1"></a><span class="ex">Testing</span> global rearrangements</span>
<span id="cb178-18"><a href="#cb178-18" aria-hidden="true" tabindex="-1"></a><span class="ex">Fiji:1.60844</span> PNG:6.74548</span>
<span id="cb178-19"><a href="#cb178-19" aria-hidden="true" tabindex="-1"></a><span class="ex">-26.530906</span> <span class="at">-25.594418</span></span>
<span id="cb178-20"><a href="#cb178-20" aria-hidden="true" tabindex="-1"></a><span class="ex">Fiji:1.60844</span> Societies:0</span>
<span id="cb178-21"><a href="#cb178-21" aria-hidden="true" tabindex="-1"></a><span class="ex">-28.39933</span> <span class="at">-25.594418</span></span>
<span id="cb178-22"><a href="#cb178-22" aria-hidden="true" tabindex="-1"></a><span class="ex">PNG:6.74548</span> Fiji:1.60844</span>
<span id="cb178-23"><a href="#cb178-23" aria-hidden="true" tabindex="-1"></a><span class="ex">-26.530906</span> <span class="at">-25.594418</span></span>
<span id="cb178-24"><a href="#cb178-24" aria-hidden="true" tabindex="-1"></a><span class="ex">NewCaledonia:0</span> Tuamotus:4.78333</span>
<span id="cb178-25"><a href="#cb178-25" aria-hidden="true" tabindex="-1"></a><span class="ex">-28.219739</span> <span class="at">-25.594418</span></span>
<span id="cb178-26"><a href="#cb178-26" aria-hidden="true" tabindex="-1"></a><span class="ex">Societies:0</span> Fiji:1.60844</span>
<span id="cb178-27"><a href="#cb178-27" aria-hidden="true" tabindex="-1"></a><span class="ex">-28.39933</span> <span class="at">-25.594418</span></span>
<span id="cb178-28"><a href="#cb178-28" aria-hidden="true" tabindex="-1"></a><span class="ex">Set</span> root above PNG</span>
<span id="cb178-29"><a href="#cb178-29" aria-hidden="true" tabindex="-1"></a><span class="ex">DONE.</span></span>
<span id="cb178-30"><a href="#cb178-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-31"><a href="#cb178-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-32"><a href="#cb178-32" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_0m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 0</span>
<span id="cb178-33"><a href="#cb178-33" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_1m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 1</span>
<span id="cb178-34"><a href="#cb178-34" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_2m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 2</span>
<span id="cb178-35"><a href="#cb178-35" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_3m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 3</span>
<span id="cb178-36"><a href="#cb178-36" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_4m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 4</span>
<span id="cb178-37"><a href="#cb178-37" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_5m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 5</span>
<span id="cb178-38"><a href="#cb178-38" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> PNG_6m <span class="at">-micro</span> <span class="at">-root</span> PNG <span class="at">-global</span> <span class="at">-m</span> 6</span>
<span id="cb178-39"><a href="#cb178-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-40"><a href="#cb178-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-41"><a href="#cb178-41" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_0m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 0 </span>
<span id="cb178-42"><a href="#cb178-42" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_1m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 1 </span>
<span id="cb178-43"><a href="#cb178-43" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_2m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 2 </span>
<span id="cb178-44"><a href="#cb178-44" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_3m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 3 </span>
<span id="cb178-45"><a href="#cb178-45" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_4m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 4 </span>
<span id="cb178-46"><a href="#cb178-46" aria-hidden="true" tabindex="-1"></a><span class="ex">treemix</span> <span class="at">-i</span> Daruanus_treemix.txt.gz <span class="at">-o</span> Tua_5m <span class="at">-micro</span> <span class="at">-root</span> Tuamotus <span class="at">-global</span> <span class="at">-m</span> 5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plot" class="level3">
<h3 class="anchored" data-anchor-id="plot">Plot</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"./treemix/5pop_explore"</span>)</span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_0m"</span>)</span>
<span id="cb179-3"><a href="#cb179-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_resid</span>(<span class="st">"PNG_0m"</span>, <span class="st">"../5poporder.txt"</span>)</span>
<span id="cb179-4"><a href="#cb179-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_1m"</span>)</span>
<span id="cb179-5"><a href="#cb179-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_resid</span>(<span class="st">"PNG_1m"</span>, <span class="st">"../5poporder.txt"</span>)</span>
<span id="cb179-6"><a href="#cb179-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_2m"</span>)</span>
<span id="cb179-7"><a href="#cb179-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_resid</span>(<span class="st">"PNG_2m"</span>, <span class="st">"../5poporder.txt"</span>)</span>
<span id="cb179-8"><a href="#cb179-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_3m"</span>)</span>
<span id="cb179-9"><a href="#cb179-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_4m"</span>)</span>
<span id="cb179-10"><a href="#cb179-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_5m"</span>)</span>
<span id="cb179-11"><a href="#cb179-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_6m"</span>)</span>
<span id="cb179-12"><a href="#cb179-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-13"><a href="#cb179-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"PNG_2mse"</span>)</span>
<span id="cb179-14"><a href="#cb179-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_resid</span>(<span class="st">"PNG_2mse"</span>, <span class="st">"../5poporder.txt"</span>)</span>
<span id="cb179-15"><a href="#cb179-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-16"><a href="#cb179-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_0m"</span>)</span>
<span id="cb179-17"><a href="#cb179-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_1m"</span>)</span>
<span id="cb179-18"><a href="#cb179-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_2m"</span>)</span>
<span id="cb179-19"><a href="#cb179-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_3m"</span>)</span>
<span id="cb179-20"><a href="#cb179-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_4m"</span>)</span>
<span id="cb179-21"><a href="#cb179-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_4m"</span>)</span>
<span id="cb179-22"><a href="#cb179-22" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"Tua_5m"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="how-many-migration-events" class="level2">
<h2 class="anchored" data-anchor-id="how-many-migration-events">How many migration events?</h2>
<p>It turns out you can evaluate the likelihood of the Treemix model in the same way that is frequently done for structure using the <span class="citation" data-cites="evannoDetectingNumberClusters2005">Evanno, Regnaut, and Goudet (<a href="#ref-evannoDetectingNumberClusters2005" role="doc-biblioref">2005</a>)</span> method. <span class="citation" data-cites="fitakOptMEstimatingOptimal2021">Fitak (<a href="#ref-fitakOptMEstimatingOptimal2021" role="doc-biblioref">2021</a>)</span> developed the <a href="https://cran.r-project.org/web/packages/OptM/readme/README.html">OptM package</a> to do this, and a few other statistics as well</p>
<section id="run-treemix-at-a-variety-of-m" class="level3">
<h3 class="anchored" data-anchor-id="run-treemix-at-a-variety-of-m">Run Treemix at a variety of m</h3>
<p>Run 5 replicates of Treemix for migration edges of 1 through 10.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> m <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">10</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> n <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">5</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>              <span class="co"># Generate random seed</span></span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>        <span class="va">s</span><span class="op">=</span><span class="va">$RANDOM</span></span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a>        <span class="ex">treemix</span> <span class="at">-i</span> ../Daruanus_treemix.txt.gz <span class="at">-o</span> PNG.<span class="va">${n}</span>.<span class="va">${m}</span> <span class="at">-global</span> <span class="at">-m</span> <span class="va">${m}</span> <span class="at">-seed</span> <span class="va">${s}</span> <span class="at">-micro</span></span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="optm" class="level3">
<h3 class="anchored" data-anchor-id="optm">OptM</h3>
<p>So the OptM algorithm doesnâ€™t work because the standard deviation across replicates is zero, because these are microsats, not SNPs. So I will borrow the <code>read.treemix()</code> function and do the analysis myself.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co">#PNG &lt;- optM("./", tsv="PNG_root_OptM.tsv")</span></span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a>PNG <span class="ot">&lt;-</span> OptM<span class="sc">:::</span><span class="fu">read.treemix</span>(<span class="st">"treemix/5pops_PNG"</span>,<span class="at">orientagraph =</span> F) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(M) <span class="sc">%&gt;%</span> </span>
<span id="cb181-4"><a href="#cb181-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(M) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">LnPD =</span> <span class="fu">mean</span>(LnPD)) <span class="sc">%&gt;%</span> </span>
<span id="cb181-5"><a href="#cb181-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">deltaM =</span> LnPD<span class="sc">-</span> <span class="fu">lag</span>(LnPD))</span>
<span id="cb181-6"><a href="#cb181-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-7"><a href="#cb181-7" aria-hidden="true" tabindex="-1"></a>PNG<span class="sc">$</span>deltaM[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb181-8"><a href="#cb181-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-9"><a href="#cb181-9" aria-hidden="true" tabindex="-1"></a>PNG <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> M, <span class="at">y =</span> LnPD)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">n.breaks=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-93-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a>PNG <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> M, <span class="at">y =</span> deltaM)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>            <span class="fu">scale_x_continuous</span>(<span class="at">n.breaks=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-93-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="plot-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-1">Plot</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./treemix/plotting_funcs.R"</span>)</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_tree</span>(<span class="st">"./treemix/5pops_PNG/PNG.5.1"</span>)</span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_resid</span>(<span class="st">"./treemix/5pops_PNG/PNG.1.1"</span>,<span class="st">"./treemix/5poporder.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="./treemix/OptM_Final_Daruanus_Treemix_Tree.jpg" class="img-fluid" alt="Final Tree with 1 edge output by Treemix"> <img src="./treemix/OptM_Final_Daruanus_Treemix_rooted.jpg" class="img-fluid" alt="Final Tree with 1 edge redrawn at itol.de"></p>
</section>
</section>
</section>
<section id="migrate-n" class="level1">
<h1>Migrate-n</h1>
<p>I also want to measure gene flow between archipelagos with migrate-n, to test the treemix model and other models.</p>
<section id="setup-1" class="level2">
<h2 class="anchored" data-anchor-id="setup-1">Setup</h2>
<p>Hereâ€™s the parmfile I set up</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co">################################################################################</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Parmfile for Migrate 3.6.4 [do not remove these first TWO lines]# generated automatically on</span></span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fri Feb 4 2022</span></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a><span class="va">menu</span><span class="op">=</span>NO</span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a><span class="va">nmlength</span><span class="op">=</span>10</span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a><span class="va">datatype</span><span class="op">=</span>Brownian</span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a><span class="va">weights</span><span class="op">=</span>NO</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a><span class="va">recover</span><span class="op">=</span>NO</span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a><span class="ex">fast-likelihood=NO</span></span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a><span class="ex">inheritance-scalars={1.00000000000000000000}</span></span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a><span class="ex">include-unknown=NO</span></span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a><span class="ex">micro-threshold=10</span></span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a><span class="ex">population-relabel={1</span> 2 3 4}</span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a><span class="va">infile</span><span class="op">=</span>../Daruanus_All_8locus.mig</span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a><span class="ex">random-seed=AUTO</span> <span class="co">#OWN:410568459</span></span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a><span class="va">title</span><span class="op">=</span> <span class="ex">Palythoa</span> tuberculosa <span class="at">-</span> Hawaii</span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a><span class="va">progress</span><span class="op">=</span>YES</span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a><span class="va">logfile</span><span class="op">=</span>YES:logfile.txt</span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a><span class="ex">print-data=NO</span></span>
<span id="cb184-23"><a href="#cb184-23" aria-hidden="true" tabindex="-1"></a><span class="va">outfile</span><span class="op">=</span>outfile.txt</span>
<span id="cb184-24"><a href="#cb184-24" aria-hidden="true" tabindex="-1"></a><span class="ex">pdf-outfile=outfile.pdf</span></span>
<span id="cb184-25"><a href="#cb184-25" aria-hidden="true" tabindex="-1"></a><span class="ex">pdf-terse=YES</span></span>
<span id="cb184-26"><a href="#cb184-26" aria-hidden="true" tabindex="-1"></a><span class="ex">use-M=YES</span></span>
<span id="cb184-27"><a href="#cb184-27" aria-hidden="true" tabindex="-1"></a><span class="ex">print-tree=NONE</span></span>
<span id="cb184-28"><a href="#cb184-28" aria-hidden="true" tabindex="-1"></a><span class="ex">mig-histogram=MIGRATIONEVENTSONLY</span></span>
<span id="cb184-29"><a href="#cb184-29" aria-hidden="true" tabindex="-1"></a><span class="va">skyline</span><span class="op">=</span>NO <span class="co">#needs mig-histogram=ALL:...</span></span>
<span id="cb184-30"><a href="#cb184-30" aria-hidden="true" tabindex="-1"></a><span class="va">mutation</span><span class="op">=</span>CONSTANT</span>
<span id="cb184-31"><a href="#cb184-31" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb184-32"><a href="#cb184-32" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb184-33"><a href="#cb184-33" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb184-34"><a href="#cb184-34" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb184-35"><a href="#cb184-35" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb184-36"><a href="#cb184-36" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb184-37"><a href="#cb184-37" aria-hidden="true" tabindex="-1"></a><span class="va">geo</span><span class="op">=</span>NO</span>
<span id="cb184-38"><a href="#cb184-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-39"><a href="#cb184-39" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-posteriorbins=</span> 500 500</span>
<span id="cb184-40"><a href="#cb184-40" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-posteriormaxtype=TOTAL</span></span>
<span id="cb184-41"><a href="#cb184-41" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-file=YES:bayesfile</span></span>
<span id="cb184-42"><a href="#cb184-42" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-allfile=YES:bayesallfile</span></span>
<span id="cb184-43"><a href="#cb184-43" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-all-posteriors=YES:bayesallposterior</span></span>
<span id="cb184-44"><a href="#cb184-44" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> THETA METROPOLIS-HASTINGS Sampler</span>
<span id="cb184-45"><a href="#cb184-45" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> MIG METROPOLIS-HASTINGS Sampler</span>
<span id="cb184-46"><a href="#cb184-46" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> DIVERGENCE METROPOLIS-HASTINGS Sampler</span>
<span id="cb184-47"><a href="#cb184-47" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> DIVERGENCESTD METROPOLIS-HASTINGS Sampler</span>
<span id="cb184-48"><a href="#cb184-48" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> THETA WEXPPRIOR: 0.0 0.001 0.1000000 0.01000 </span>
<span id="cb184-49"><a href="#cb184-49" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> MIG WEXPPRIOR: 0.000100 1000.000000 10000 100</span>
<span id="cb184-50"><a href="#cb184-50" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> RATE <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0.000000 10000000000.000000 1000000000.000000 </span>
<span id="cb184-51"><a href="#cb184-51" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-hyperpriors=NO</span></span>
<span id="cb184-52"><a href="#cb184-52" aria-hidden="true" tabindex="-1"></a><span class="ex">long-chains=1</span></span>
<span id="cb184-53"><a href="#cb184-53" aria-hidden="true" tabindex="-1"></a><span class="ex">long-inc=100</span></span>
<span id="cb184-54"><a href="#cb184-54" aria-hidden="true" tabindex="-1"></a><span class="ex">long-sample=10000</span></span>
<span id="cb184-55"><a href="#cb184-55" aria-hidden="true" tabindex="-1"></a><span class="ex">burn-in=2000</span>  </span>
<span id="cb184-56"><a href="#cb184-56" aria-hidden="true" tabindex="-1"></a><span class="ex">auto-tune=YES:0.440000</span></span>
<span id="cb184-57"><a href="#cb184-57" aria-hidden="true" tabindex="-1"></a><span class="va">assign</span><span class="op">=</span>NO</span>
<span id="cb184-58"><a href="#cb184-58" aria-hidden="true" tabindex="-1"></a><span class="va">heating</span><span class="op">=</span>YES:1:<span class="dt">{1.000000</span><span class="op">,</span><span class="dt">1.500000</span><span class="op">,</span><span class="dt">3.000000</span><span class="op">,</span><span class="dt">5</span><span class="op">,</span><span class="dt">10</span><span class="op">,</span><span class="dt">1000000.000000}</span></span>
<span id="cb184-59"><a href="#cb184-59" aria-hidden="true" tabindex="-1"></a><span class="ex">heated-swap=YES</span></span>
<span id="cb184-60"><a href="#cb184-60" aria-hidden="true" tabindex="-1"></a><span class="ex">moving-steps=NO</span></span>
<span id="cb184-61"><a href="#cb184-61" aria-hidden="true" tabindex="-1"></a><span class="ex">gelman-convergence=No</span></span>
<span id="cb184-62"><a href="#cb184-62" aria-hidden="true" tabindex="-1"></a><span class="va">replicate</span><span class="op">=</span>YES:3</span>
<span id="cb184-63"><a href="#cb184-63" aria-hidden="true" tabindex="-1"></a><span class="ex">end</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>And the datafile I converted from the 8-locus genpop file with PGDSpider.</p>
<p>The header looks like</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="ex">4</span> 8 . D. aruanus dataset lumped by archipelago with 3 loci removed for HWE violation <span class="at">-</span> EDC 3/28/2022</span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a><span class="co">#@M 2 3 2 2 2 2 2 2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="initial-runs" class="level2">
<h2 class="anchored" data-anchor-id="initial-runs">Initial Runs</h2>
<p>Copied up to Nautilus and ran it with:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="ex">screen</span> <span class="at">-S</span> migrate_testrun</span>
<span id="cb186-2"><a href="#cb186-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-3"><a href="#cb186-3" aria-hidden="true" tabindex="-1"></a><span class="ex">mpirun</span> <span class="at">-np</span> 32 ~/migrate-4.4.4/src/migrate-n-mpi parmfile</span>
<span id="cb186-4"><a href="#cb186-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-5"><a href="#cb186-5" aria-hidden="true" tabindex="-1"></a><span class="co"># And on Montastraea</span></span>
<span id="cb186-6"><a href="#cb186-6" aria-hidden="true" tabindex="-1"></a><span class="ex">mpirun</span> <span class="at">-np</span> 16 ~/eric_data/migrate-3.7.2/src/migrate-n-mpi parmfile</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-12-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-1" role="tab" aria-controls="tabset-12-1" aria-selected="true">Symmetric Stepping Stone</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-12-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-2" role="tab" aria-controls="tabset-12-2" aria-selected="false">East to West Stepping Stone</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-12-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-3" role="tab" aria-controls="tabset-12-3" aria-selected="false">West to East Stepping Stone</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-12-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-4" role="tab" aria-controls="tabset-12-4" aria-selected="false">Smaller Priors</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-12-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-5" role="tab" aria-controls="tabset-12-5" aria-selected="false">Divergence with Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-12-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-12-6" role="tab" aria-controls="tabset-12-6" aria-selected="false">Divergence without gene flow</a></li></ul>
<div class="tab-content">
<div id="tabset-12-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-12-1-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> S 0 0</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="ex">S</span> <span class="pp">*</span> S 0</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> S <span class="pp">*</span> S</span>
<span id="cb187-5"><a href="#cb187-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 S <span class="pp">*</span></span>
<span id="cb187-6"><a href="#cb187-6" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-12-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-12-2-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb188-2"><a href="#cb188-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb188-3"><a href="#cb188-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb188-4"><a href="#cb188-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb188-5"><a href="#cb188-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span></span>
<span id="cb188-6"><a href="#cb188-6" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-12-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-12-3-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> 0 0 0</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb189-5"><a href="#cb189-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb189-6"><a href="#cb189-6" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-12-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-12-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> THETA WEXPPRIOR: 0.0 0.01 1.000000 0.01000 </span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> MIG WEXPPRIOR: 0.0100 100.000000 1000 10</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This definitely improved things. Now all gene flow parameters are around 40! But the model marginal likelihood is all wonky.</p>
</div>
<div id="tabset-12-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-12-5-tab">
<p>Now Iâ€™m going to add divergence with gene flow (going from West to East). Also change the migration priors to uniform.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> THETA <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000 </span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> MIG <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0 100 10</span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLIT <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0 1000  10</span>
<span id="cb191-4"><a href="#cb191-4" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLITSTD <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0 500 10 </span>
<span id="cb191-5"><a href="#cb191-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-6"><a href="#cb191-6" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb191-7"><a href="#cb191-7" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span>       0       0       0</span>
<span id="cb191-8"><a href="#cb191-8" aria-hidden="true" tabindex="-1"></a><span class="ex">D</span>       <span class="pp">*</span>       0       0</span>
<span id="cb191-9"><a href="#cb191-9" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>       D       <span class="pp">*</span>       0</span>
<span id="cb191-10"><a href="#cb191-10" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>       0       D       <span class="pp">*</span></span>
<span id="cb191-11"><a href="#cb191-11" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After <a href="https://groups.google.com/g/migrate-support/c/49njPlwp62E/m/xzy4DkU9AwAJ">this exchange</a> with Peter, I changed the priors to be the same as Theta.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> THETA <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000 </span>
<span id="cb192-2"><a href="#cb192-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> MIG <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0 100 10</span>
<span id="cb192-3"><a href="#cb192-3" aria-hidden="true" tabindex="-1"></a><span class="ex">divergence-distrib</span> = S</span>
<span id="cb192-4"><a href="#cb192-4" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLIT <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000 </span>
<span id="cb192-5"><a href="#cb192-5" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLITSTD <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Results look like there is little to no gene flow among archipelagos</p>
</div>
<div id="tabset-12-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-12-6-tab">
<p>So letâ€™s look at a model of no gene flow</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb193"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb193-1"><a href="#cb193-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb193-2"><a href="#cb193-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span>       0       0       0</span>
<span id="cb193-3"><a href="#cb193-3" aria-hidden="true" tabindex="-1"></a><span class="ex">d</span>       <span class="pp">*</span>       0       0</span>
<span id="cb193-4"><a href="#cb193-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>       d       <span class="pp">*</span>       0</span>
<span id="cb193-5"><a href="#cb193-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span>       0       d       <span class="pp">*</span></span>
<span id="cb193-6"><a href="#cb193-6" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</section>
<section id="final-runs" class="level2">
<h2 class="anchored" data-anchor-id="final-runs">Final Runs</h2>
<p>OK, so after some futzing around with these models, I am now trying 11 models for what I hope will be a publishable analysis.</p>
<p>All built on this parmfile</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co">################################################################################</span></span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Parmfile for Migrate 3.6.4 [do not remove these first TWO lines]# generated automatically on</span></span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fri Feb 4 2022</span></span>
<span id="cb194-4"><a href="#cb194-4" aria-hidden="true" tabindex="-1"></a><span class="va">menu</span><span class="op">=</span>NO</span>
<span id="cb194-5"><a href="#cb194-5" aria-hidden="true" tabindex="-1"></a><span class="va">nmlength</span><span class="op">=</span>10</span>
<span id="cb194-6"><a href="#cb194-6" aria-hidden="true" tabindex="-1"></a><span class="va">datatype</span><span class="op">=</span>Brownian</span>
<span id="cb194-7"><a href="#cb194-7" aria-hidden="true" tabindex="-1"></a><span class="va">weights</span><span class="op">=</span>NO</span>
<span id="cb194-8"><a href="#cb194-8" aria-hidden="true" tabindex="-1"></a><span class="va">recover</span><span class="op">=</span>NO</span>
<span id="cb194-9"><a href="#cb194-9" aria-hidden="true" tabindex="-1"></a><span class="ex">fast-likelihood=NO</span></span>
<span id="cb194-10"><a href="#cb194-10" aria-hidden="true" tabindex="-1"></a><span class="ex">inheritance-scalars={1.00000000000000000000}</span></span>
<span id="cb194-11"><a href="#cb194-11" aria-hidden="true" tabindex="-1"></a><span class="ex">include-unknown=NO</span></span>
<span id="cb194-12"><a href="#cb194-12" aria-hidden="true" tabindex="-1"></a><span class="ex">micro-threshold=10</span></span>
<span id="cb194-13"><a href="#cb194-13" aria-hidden="true" tabindex="-1"></a><span class="ex">population-relabel={1</span> 2 3 4}</span>
<span id="cb194-14"><a href="#cb194-14" aria-hidden="true" tabindex="-1"></a><span class="va">infile</span><span class="op">=</span>../../Daruanus_All_8locus.mig</span>
<span id="cb194-15"><a href="#cb194-15" aria-hidden="true" tabindex="-1"></a><span class="ex">random-seed=AUTO</span> <span class="co">#OWN:410568459</span></span>
<span id="cb194-16"><a href="#cb194-16" aria-hidden="true" tabindex="-1"></a><span class="va">title</span><span class="op">=</span> <span class="ex">Dascyllus</span> aruanus <span class="at">-</span> South Pacific Archipelagos</span>
<span id="cb194-17"><a href="#cb194-17" aria-hidden="true" tabindex="-1"></a><span class="va">progress</span><span class="op">=</span>YES</span>
<span id="cb194-18"><a href="#cb194-18" aria-hidden="true" tabindex="-1"></a><span class="va">logfile</span><span class="op">=</span>YES:logfile.txt</span>
<span id="cb194-19"><a href="#cb194-19" aria-hidden="true" tabindex="-1"></a><span class="ex">print-data=NO</span></span>
<span id="cb194-20"><a href="#cb194-20" aria-hidden="true" tabindex="-1"></a><span class="va">outfile</span><span class="op">=</span>outfile.txt</span>
<span id="cb194-21"><a href="#cb194-21" aria-hidden="true" tabindex="-1"></a><span class="ex">pdf-outfile=outfile.pdf</span></span>
<span id="cb194-22"><a href="#cb194-22" aria-hidden="true" tabindex="-1"></a><span class="ex">pdf-terse=YES</span></span>
<span id="cb194-23"><a href="#cb194-23" aria-hidden="true" tabindex="-1"></a><span class="ex">use-M=NO</span></span>
<span id="cb194-24"><a href="#cb194-24" aria-hidden="true" tabindex="-1"></a><span class="ex">print-tree=NONE</span></span>
<span id="cb194-25"><a href="#cb194-25" aria-hidden="true" tabindex="-1"></a><span class="ex">mig-histogram=MIGRATIONEVENTSONLY</span></span>
<span id="cb194-26"><a href="#cb194-26" aria-hidden="true" tabindex="-1"></a><span class="va">skyline</span><span class="op">=</span>NO <span class="co">#needs mig-histogram=ALL:...</span></span>
<span id="cb194-27"><a href="#cb194-27" aria-hidden="true" tabindex="-1"></a><span class="va">mutation</span><span class="op">=</span>CONSTANT</span>
<span id="cb194-28"><a href="#cb194-28" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb194-29"><a href="#cb194-29" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> S 0 0</span>
<span id="cb194-30"><a href="#cb194-30" aria-hidden="true" tabindex="-1"></a><span class="ex">S</span> <span class="pp">*</span> S 0</span>
<span id="cb194-31"><a href="#cb194-31" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> S <span class="pp">*</span> S</span>
<span id="cb194-32"><a href="#cb194-32" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 S <span class="pp">*</span></span>
<span id="cb194-33"><a href="#cb194-33" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb194-34"><a href="#cb194-34" aria-hidden="true" tabindex="-1"></a><span class="va">geo</span><span class="op">=</span>NO</span>
<span id="cb194-35"><a href="#cb194-35" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-posteriorbins=</span> 2000 2000</span>
<span id="cb194-36"><a href="#cb194-36" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-posteriormaxtype=TOTAL</span></span>
<span id="cb194-37"><a href="#cb194-37" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-file=YES:bayesfile</span></span>
<span id="cb194-38"><a href="#cb194-38" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-allfile=YES:bayesallfile</span></span>
<span id="cb194-39"><a href="#cb194-39" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-all-posteriors=YES:bayesallposterior</span></span>
<span id="cb194-40"><a href="#cb194-40" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> THETA METROPOLIS-HASTINGS Sampler</span>
<span id="cb194-41"><a href="#cb194-41" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> MIG METROPOLIS-HASTINGS Sampler</span>
<span id="cb194-42"><a href="#cb194-42" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> SPLITSTD METROPOLIS-HASTINGS Sampler</span>
<span id="cb194-43"><a href="#cb194-43" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-proposals=</span> SPLIT METROPOLIS-HASTINGS Sampler</span>
<span id="cb194-44"><a href="#cb194-44" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> THETA <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.1 20.00000 0.1000 </span>
<span id="cb194-45"><a href="#cb194-45" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> MIG <span class="pp">*</span> <span class="pp">*</span> UNIFORMPRIOR: 0 100 10</span>
<span id="cb194-46"><a href="#cb194-46" aria-hidden="true" tabindex="-1"></a><span class="ex">divergence-distrib=</span> E</span>
<span id="cb194-47"><a href="#cb194-47" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLIT <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000</span>
<span id="cb194-48"><a href="#cb194-48" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-priors=</span> SPLITSTD <span class="pp">*</span> <span class="pp">*</span> WEXPPRIOR: 0.0 0.01 1.000000 0.01000</span>
<span id="cb194-49"><a href="#cb194-49" aria-hidden="true" tabindex="-1"></a><span class="va">migration</span><span class="op">=</span>PRIOR:10</span>
<span id="cb194-50"><a href="#cb194-50" aria-hidden="true" tabindex="-1"></a><span class="va">rate</span><span class="op">=</span>PRIOR:50</span>
<span id="cb194-51"><a href="#cb194-51" aria-hidden="true" tabindex="-1"></a><span class="va">split</span><span class="op">=</span>PRIOR:10</span>
<span id="cb194-52"><a href="#cb194-52" aria-hidden="true" tabindex="-1"></a><span class="va">splitstd</span><span class="op">=</span>PRIOR:10</span>
<span id="cb194-53"><a href="#cb194-53" aria-hidden="true" tabindex="-1"></a><span class="ex">bayes-hyperpriors=NO</span></span>
<span id="cb194-54"><a href="#cb194-54" aria-hidden="true" tabindex="-1"></a><span class="ex">long-chains=1</span></span>
<span id="cb194-55"><a href="#cb194-55" aria-hidden="true" tabindex="-1"></a><span class="ex">long-inc=100</span></span>
<span id="cb194-56"><a href="#cb194-56" aria-hidden="true" tabindex="-1"></a><span class="ex">long-sample=10000</span></span>
<span id="cb194-57"><a href="#cb194-57" aria-hidden="true" tabindex="-1"></a><span class="ex">burn-in=10000</span>  </span>
<span id="cb194-58"><a href="#cb194-58" aria-hidden="true" tabindex="-1"></a><span class="ex">auto-tune=YES:0.440000</span></span>
<span id="cb194-59"><a href="#cb194-59" aria-hidden="true" tabindex="-1"></a><span class="va">assign</span><span class="op">=</span>NO</span>
<span id="cb194-60"><a href="#cb194-60" aria-hidden="true" tabindex="-1"></a><span class="va">heating</span><span class="op">=</span>YES:1:<span class="dt">{1.000000</span><span class="op">,</span><span class="dt">1.500000</span><span class="op">,</span><span class="dt">3.000000</span><span class="op">,</span><span class="dt">7</span><span class="op">,</span><span class="dt">15</span><span class="op">,</span><span class="dt">1000000.000000}</span></span>
<span id="cb194-61"><a href="#cb194-61" aria-hidden="true" tabindex="-1"></a><span class="ex">heated-swap=YES</span></span>
<span id="cb194-62"><a href="#cb194-62" aria-hidden="true" tabindex="-1"></a><span class="ex">moving-steps=NO</span></span>
<span id="cb194-63"><a href="#cb194-63" aria-hidden="true" tabindex="-1"></a><span class="ex">gelman-convergence=No</span></span>
<span id="cb194-64"><a href="#cb194-64" aria-hidden="true" tabindex="-1"></a><span class="va">replicate</span><span class="op">=</span>NO</span>
<span id="cb194-65"><a href="#cb194-65" aria-hidden="true" tabindex="-1"></a><span class="ex">end</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="models" class="level3">
<h3 class="anchored" data-anchor-id="models">Models</h3>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-13-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-1" role="tab" aria-controls="tabset-13-1" aria-selected="true">Asymmetric Equilibrium Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-2" role="tab" aria-controls="tabset-13-2" aria-selected="false">Symmetric Equilibrium Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-3" role="tab" aria-controls="tabset-13-3" aria-selected="false">n-Island</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-4" role="tab" aria-controls="tabset-13-4" aria-selected="false">Panmixia</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-5" role="tab" aria-controls="tabset-13-5" aria-selected="false">East to West Divergence with Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-6" role="tab" aria-controls="tabset-13-6" aria-selected="false">West to East Divergence with Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-7" role="tab" aria-controls="tabset-13-7" aria-selected="false">East to West Divergence without Gene Flow</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-13-8-tab" data-bs-toggle="tab" data-bs-target="#tabset-13-8" role="tab" aria-controls="tabset-13-8" aria-selected="false">West to East Treemix</a></li></ul>
<div class="tab-content">
<div id="tabset-13-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-13-1-tab">
<p>The â€œstandardâ€ stepping stone model. Gene flow at equilibrium between neighboring archipelagos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb195"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb195-1"><a href="#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb195-2"><a href="#cb195-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0 0</span>
<span id="cb195-3"><a href="#cb195-3" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb195-4"><a href="#cb195-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb195-5"><a href="#cb195-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb195-6"><a href="#cb195-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb195-7"><a href="#cb195-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-2-tab">
<p>Same as above, but gene flow in either direction is symmetric for each pair of archipelagos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb196-2"><a href="#cb196-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> S 0 0 0</span>
<span id="cb196-3"><a href="#cb196-3" aria-hidden="true" tabindex="-1"></a><span class="ex">S</span> <span class="pp">*</span> S 0 0</span>
<span id="cb196-4"><a href="#cb196-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> S <span class="pp">*</span> S 0</span>
<span id="cb196-5"><a href="#cb196-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 S <span class="pp">*</span> S</span>
<span id="cb196-6"><a href="#cb196-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 S <span class="pp">*</span></span>
<span id="cb196-7"><a href="#cb196-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="east-to-west-equilibrium" class="level3">
<h3 class="anchored" data-anchor-id="east-to-west-equilibrium">East To West Equilibrium</h3>
<p>East to West gene flow only. Following the South Equatorial current</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0 0</span>
<span id="cb197-3"><a href="#cb197-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb197-4"><a href="#cb197-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb197-5"><a href="#cb197-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb197-6"><a href="#cb197-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 0 <span class="pp">*</span></span>
<span id="cb197-7"><a href="#cb197-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="west-to-east-equilibrium" class="level3">
<h3 class="anchored" data-anchor-id="west-to-east-equilibrium">West to East Equilibrium</h3>
<p>The other wayâ€¦</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> 0 0 0 0</span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> <span class="pp">*</span> 0 0 0</span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> <span class="pp">*</span> 0 0</span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> <span class="pp">*</span> 0</span>
<span id="cb198-6"><a href="#cb198-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span> <span class="pp">*</span></span>
<span id="cb198-7"><a href="#cb198-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</div>
<div id="tabset-13-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-3-tab">
<p>All populations the same size. Gene flow between all populations at the same rate</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="ex">m</span> m m m m</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a><span class="ex">m</span> m m m m</span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a><span class="ex">m</span> m m m m</span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a><span class="ex">m</span> m m m m</span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a><span class="ex">m</span> m m m m</span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-4-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span></span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-5-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> D 0 0 0</span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> D 0 0</span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> D 0</span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span> D</span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 0 <span class="pp">*</span></span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-6-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> 0 0 0 0</span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a><span class="ex">D</span> <span class="pp">*</span> 0 0 0</span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> D <span class="pp">*</span> 0 0</span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 D <span class="pp">*</span> 0</span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 D <span class="pp">*</span></span>
<span id="cb202-7"><a href="#cb202-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-7-tab">
<div class="cell">
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> d 0 0 0</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> <span class="pp">*</span> d 0 0</span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 <span class="pp">*</span> d 0</span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 <span class="pp">*</span> d</span>
<span id="cb203-6"><a href="#cb203-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 0 <span class="pp">*</span></span>
<span id="cb203-7"><a href="#cb203-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-13-8" class="tab-pane" role="tabpanel" aria-labelledby="tabset-13-8-tab">
<p>Based on the treemix result above. Migration from the Societies to New Caledonia</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="ex">custom-migration={</span></span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a><span class="ex">*</span> 0 0 0 0</span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a><span class="ex">d</span> <span class="pp">*</span> 0 <span class="pp">*</span> 0</span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> d <span class="pp">*</span> 0 0</span>
<span id="cb204-5"><a href="#cb204-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 d <span class="pp">*</span> 0</span>
<span id="cb204-6"><a href="#cb204-6" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> 0 0 d <span class="pp">*</span></span>
<span id="cb204-7"><a href="#cb204-7" aria-hidden="true" tabindex="-1"></a><span class="er">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</section>
<section id="run-it" class="level3">
<h3 class="anchored" data-anchor-id="run-it">Run It</h3>
<p>Copy it up to Nautilus, make 10 replicate folders.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> <span class="at">-r</span> ./migrate/Models2 naut:./Daruanus3/rep1</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> a <span class="kw">in</span> <span class="va">$(</span><span class="fu">seq</span> 2 10<span class="va">)</span><span class="kw">;</span> <span class="cf">do</span> <span class="fu">cp</span> <span class="at">-r</span> rep1 rep<span class="va">$a</span><span class="kw">;</span> <span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="bash-script" class="level4">
<h4 class="anchored" data-anchor-id="bash-script">Bash Script</h4>
<p>So we will do 10 replicates of 3 replicates. This will start at r1, and run all models for that before moving on. Pretty sure this will finish one whole model before moving on to the next one (since all threads are being used for different loci)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Bash Script</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a><span class="co">#!</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> r <span class="kw">in</span> <span class="pp">*</span>/</span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">do</span></span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> starting <span class="va">$r</span> at <span class="va">$(</span><span class="fu">date</span> <span class="va">)</span><span class="op">&gt;&gt;</span> progress.txt</span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a>                <span class="bu">cd</span> <span class="va">$r</span></span>
<span id="cb206-7"><a href="#cb206-7" aria-hidden="true" tabindex="-1"></a>                <span class="bu">echo</span> <span class="va">$r</span></span>
<span id="cb206-8"><a href="#cb206-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">date</span></span>
<span id="cb206-9"><a href="#cb206-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">date</span> <span class="op">&gt;</span> progress.txt</span>
<span id="cb206-10"><a href="#cb206-10" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span> m <span class="kw">in</span> <span class="pp">*</span>/</span>
<span id="cb206-11"><a href="#cb206-11" aria-hidden="true" tabindex="-1"></a>                          <span class="cf">do</span></span>
<span id="cb206-12"><a href="#cb206-12" aria-hidden="true" tabindex="-1"></a>                          <span class="bu">echo</span> starting <span class="va">$m</span> at <span class="va">$(</span><span class="fu">date</span> <span class="va">)</span><span class="op">&gt;&gt;</span> progress.txt</span>
<span id="cb206-13"><a href="#cb206-13" aria-hidden="true" tabindex="-1"></a>                                <span class="bu">cd</span> <span class="va">$m</span></span>
<span id="cb206-14"><a href="#cb206-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">date</span> <span class="op">&gt;</span> date.txt</span>
<span id="cb206-15"><a href="#cb206-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="bu">echo</span> <span class="va">$m</span></span>
<span id="cb206-16"><a href="#cb206-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">date</span></span>
<span id="cb206-17"><a href="#cb206-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="ex">mpirun</span> <span class="at">--use-hwthread-cpus</span> <span class="at">-np</span> 120 ~/migrate-4.4.4/src/migrate-n-mpi parmfile</span>
<span id="cb206-18"><a href="#cb206-18" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">sleep</span> 1</span>
<span id="cb206-19"><a href="#cb206-19" aria-hidden="true" tabindex="-1"></a>                                <span class="bu">cd</span> ..</span>
<span id="cb206-20"><a href="#cb206-20" aria-hidden="true" tabindex="-1"></a>                          <span class="cf">done</span></span>
<span id="cb206-21"><a href="#cb206-21" aria-hidden="true" tabindex="-1"></a>                <span class="bu">cd</span> ..</span>
<span id="cb206-22"><a href="#cb206-22" aria-hidden="true" tabindex="-1"></a>        <span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="copy-it-down" class="level4">
<h4 class="anchored" data-anchor-id="copy-it-down">Copy It Down</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rsync</span> <span class="at">-av</span> <span class="at">-e</span> ssh <span class="at">--exclude</span><span class="op">=</span><span class="st">'bayes*'</span> <span class="at">--exclude</span><span class="op">=</span><span class="st">"*.pdf"</span> <span class="at">--exclude</span><span class="op">=</span><span class="st">"*.mig"</span> ecrandall@nautilus.psu.edu:~/Daruanus3/ output3</span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rsync</span> <span class="at">-av</span> <span class="at">-e</span> ssh ecrandall@nautilus.psu.edu:~/daruanus/rep1/EtoW_DivMig output/rep1/EtoW_DivMig</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="results" class="level3">
<h3 class="anchored" data-anchor-id="results">Results</h3>
<section id="function-library" class="level4">
<h4 class="anchored" data-anchor-id="function-library">Function library</h4>
<p>Starting material for this one is from the Ptuberculosa_migrate.rmd</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a>harvest.model.likelihoods <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">workingDir =</span> workingDir,</span>
<span id="cb208-2"><a href="#cb208-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">outfileName =</span> <span class="st">"outfile.txt"</span>,</span>
<span id="cb208-3"><a href="#cb208-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">multilocus =</span> T){</span>
<span id="cb208-4"><a href="#cb208-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this function harvests model marginal likelihoods for models calculated by</span></span>
<span id="cb208-5"><a href="#cb208-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># the program migrate-n (Beerli &amp; Felsenstein 2001).</span></span>
<span id="cb208-6"><a href="#cb208-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># It takes as input a directory full of directories, </span></span>
<span id="cb208-7"><a href="#cb208-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># each of which contains output from a migrate model, and is named</span></span>
<span id="cb208-8"><a href="#cb208-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># after that model. </span></span>
<span id="cb208-9"><a href="#cb208-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-10"><a href="#cb208-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#initialize a data frame to take the values</span></span>
<span id="cb208-11"><a href="#cb208-11" aria-hidden="true" tabindex="-1"></a>    modelMarglikes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">model=</span><span class="fu">character</span>(),</span>
<span id="cb208-12"><a href="#cb208-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">thermodynamic=</span><span class="fu">numeric</span>(),</span>
<span id="cb208-13"><a href="#cb208-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">bezier.corrected=</span><span class="fu">numeric</span>(), </span>
<span id="cb208-14"><a href="#cb208-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">harmonic=</span><span class="fu">numeric</span>()) </span>
<span id="cb208-15"><a href="#cb208-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># loop through directories in the working directory, each of which is name</span></span>
<span id="cb208-16"><a href="#cb208-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># after a different model</span></span>
<span id="cb208-17"><a href="#cb208-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">list.dirs</span>(workingDir, <span class="at">full.names =</span> F)[<span class="sc">-</span><span class="dv">1</span>]){ <span class="co">#i&lt;-"stepping.stone"</span></span>
<span id="cb208-18"><a href="#cb208-18" aria-hidden="true" tabindex="-1"></a>      modelDir<span class="ot">&lt;-</span><span class="fu">file.path</span>(workingDir,i)</span>
<span id="cb208-19"><a href="#cb208-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(modelDir)</span>
<span id="cb208-20"><a href="#cb208-20" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scan in the outfile, separating at each newline</span></span>
<span id="cb208-21"><a href="#cb208-21" aria-hidden="true" tabindex="-1"></a>      outfile<span class="ot">&lt;-</span><span class="fu">scan</span>(<span class="at">file=</span><span class="fu">file.path</span>(modelDir,outfileName),</span>
<span id="cb208-22"><a href="#cb208-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">what=</span><span class="st">"character"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>) </span>
<span id="cb208-23"><a href="#cb208-23" aria-hidden="true" tabindex="-1"></a>    <span class="co">#find the line with the likelihoods on it and split on runs of spaces</span></span>
<span id="cb208-24"><a href="#cb208-24" aria-hidden="true" tabindex="-1"></a>      marglikeline <span class="ot">&lt;-</span> <span class="fu">str_which</span>(outfile,<span class="st">"Scaling factor"</span>)<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb208-25"><a href="#cb208-25" aria-hidden="true" tabindex="-1"></a>      marglikeline <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(outfile[marglikeline],</span>
<span id="cb208-26"><a href="#cb208-26" aria-hidden="true" tabindex="-1"></a>               <span class="st">"</span><span class="sc">\\</span><span class="st">s+"</span>, <span class="at">perl =</span> T)[[<span class="dv">1</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb208-27"><a href="#cb208-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  if(length(marglikeline)==0){next}</span></span>
<span id="cb208-28"><a href="#cb208-28" aria-hidden="true" tabindex="-1"></a>      marglikes <span class="ot">&lt;-</span> <span class="fu">c</span>(i,marglikeline)</span>
<span id="cb208-29"><a href="#cb208-29" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb208-30"><a href="#cb208-30" aria-hidden="true" tabindex="-1"></a>      modelMarglikes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(modelMarglikes,marglikes, <span class="at">deparse.level =</span> <span class="dv">2</span>)</span>
<span id="cb208-31"><a href="#cb208-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb208-32"><a href="#cb208-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(modelMarglikes) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"model"</span>,<span class="st">"thermodynamic"</span>,</span>
<span id="cb208-33"><a href="#cb208-33" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"bezier.corrected"</span>,<span class="st">"harmonic"</span>)</span>
<span id="cb208-34"><a href="#cb208-34" aria-hidden="true" tabindex="-1"></a>  modelMarglikes[<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(modelMarglikes[<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>], as.numeric)</span>
<span id="cb208-35"><a href="#cb208-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(modelMarglikes)</span>
<span id="cb208-36"><a href="#cb208-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-37"><a href="#cb208-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-38"><a href="#cb208-38" aria-hidden="true" tabindex="-1"></a>bfcalcs<span class="ot">&lt;-</span><span class="cf">function</span>(df,<span class="at">ml=</span><span class="st">"bezier.corrected"</span>){</span>
<span id="cb208-39"><a href="#cb208-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># This calculates log bayes factors on data frames output by</span></span>
<span id="cb208-40"><a href="#cb208-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># harvest.model.likelihoods(), following Johnson and Omland (2004)</span></span>
<span id="cb208-41"><a href="#cb208-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># You may choose the likelihood flavor with</span></span>
<span id="cb208-42"><a href="#cb208-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ml = "bezier.corrected", "thermodynamic" or "harmonic"</span></span>
<span id="cb208-43"><a href="#cb208-43" aria-hidden="true" tabindex="-1"></a>  <span class="co">#df$thermodynamic &lt;- as.numeric(df$thermodynamic)</span></span>
<span id="cb208-44"><a href="#cb208-44" aria-hidden="true" tabindex="-1"></a>  <span class="co">#df$bezier.corrected &lt;- as.numeric(df$bezier.corrected)</span></span>
<span id="cb208-45"><a href="#cb208-45" aria-hidden="true" tabindex="-1"></a>  <span class="co">#df$harmonic &lt;- as.numeric(df$harmonic)</span></span>
<span id="cb208-46"><a href="#cb208-46" aria-hidden="true" tabindex="-1"></a>  mlcol <span class="ot">&lt;-</span> df[[ml]] </span>
<span id="cb208-47"><a href="#cb208-47" aria-hidden="true" tabindex="-1"></a> bmvalue <span class="ot">&lt;-</span> mlcol[<span class="fu">which.max</span>(mlcol)]</span>
<span id="cb208-48"><a href="#cb208-48" aria-hidden="true" tabindex="-1"></a> lbf <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">*</span>(mlcol<span class="sc">-</span>bmvalue)</span>
<span id="cb208-49"><a href="#cb208-49" aria-hidden="true" tabindex="-1"></a> choice <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>mlcol)</span>
<span id="cb208-50"><a href="#cb208-50" aria-hidden="true" tabindex="-1"></a> modelprob <span class="ot">&lt;-</span> <span class="fu">exp</span>(lbf<span class="sc">/</span><span class="dv">2</span>)<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">exp</span>(lbf<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb208-51"><a href="#cb208-51" aria-hidden="true" tabindex="-1"></a> dfall <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df,lbf,choice,modelprob)</span>
<span id="cb208-52"><a href="#cb208-52" aria-hidden="true" tabindex="-1"></a> <span class="fu">return</span>(dfall)</span>
<span id="cb208-53"><a href="#cb208-53" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb208-54"><a href="#cb208-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-55"><a href="#cb208-55" aria-hidden="true" tabindex="-1"></a>migrants.per.gen<span class="ot">&lt;-</span><span class="cf">function</span>(x){</span>
<span id="cb208-56"><a href="#cb208-56" aria-hidden="true" tabindex="-1"></a>  <span class="co">#a function for creating Nm vectors out of m and Theta vectors.</span></span>
<span id="cb208-57"><a href="#cb208-57" aria-hidden="true" tabindex="-1"></a>  <span class="co">#x&lt;-x[[1]]</span></span>
<span id="cb208-58"><a href="#cb208-58" aria-hidden="true" tabindex="-1"></a>  m<span class="ot">&lt;-</span><span class="fu">names</span>(x)[<span class="fu">which</span>(<span class="fu">grepl</span>(<span class="st">"M_"</span>,<span class="fu">names</span>(x)))] <span class="co">#names of m columns</span></span>
<span id="cb208-59"><a href="#cb208-59" aria-hidden="true" tabindex="-1"></a>  <span class="co">#theta&lt;-names(x)[which(grepl("Theta_",names(x)))] #names of theta columns</span></span>
<span id="cb208-60"><a href="#cb208-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(n <span class="cf">in</span> m){</span>
<span id="cb208-61"><a href="#cb208-61" aria-hidden="true" tabindex="-1"></a>    t<span class="ot">&lt;-</span><span class="fu">paste</span>(<span class="st">"Theta"</span>,<span class="fu">strsplit</span>(n,<span class="at">split=</span><span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">3</span>],<span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb208-62"><a href="#cb208-62" aria-hidden="true" tabindex="-1"></a>    x[,<span class="fu">paste</span>(<span class="st">"Nm"</span>,<span class="fu">strsplit</span>(n,<span class="at">split=</span><span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],<span class="fu">strsplit</span>(n,<span class="at">split=</span><span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">3</span>],<span class="at">sep=</span><span class="st">"_"</span>)]<span class="ot">&lt;-</span> x[,<span class="fu">which</span>(<span class="fu">names</span>(x)<span class="sc">==</span>n)]<span class="sc">*</span>x[,<span class="fu">which</span>(<span class="fu">names</span>(x)<span class="sc">==</span>t)] </span>
<span id="cb208-63"><a href="#cb208-63" aria-hidden="true" tabindex="-1"></a>    <span class="co">#this hairy little statement makes a new column named "Nm_X_Y" and then fills it by multiplying the M_X_Y column by the Theta_Y column </span></span>
<span id="cb208-64"><a href="#cb208-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb208-65"><a href="#cb208-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb208-66"><a href="#cb208-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-67"><a href="#cb208-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-68"><a href="#cb208-68" aria-hidden="true" tabindex="-1"></a>remove_prior <span class="ot">&lt;-</span> <span class="cf">function</span>(densityd,prior,<span class="at">threshold =</span> <span class="fl">1e-10</span>, <span class="at">quad_prec =</span> F){</span>
<span id="cb208-69"><a href="#cb208-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># this function removes a prior from the </span></span>
<span id="cb208-70"><a href="#cb208-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y values of a density distributions (density).</span></span>
<span id="cb208-71"><a href="#cb208-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># first it zeros values less than a threshold, then </span></span>
<span id="cb208-72"><a href="#cb208-72" aria-hidden="true" tabindex="-1"></a>  <span class="co"># removes the prior, then renormalizes so the density</span></span>
<span id="cb208-73"><a href="#cb208-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># sums to 1.</span></span>
<span id="cb208-74"><a href="#cb208-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># can optionally use the mpfr package for very small density values.</span></span>
<span id="cb208-75"><a href="#cb208-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Rmpfr)</span>
<span id="cb208-76"><a href="#cb208-76" aria-hidden="true" tabindex="-1"></a>  densityd[<span class="fu">which</span>(densityd <span class="sc">&lt;</span> threshold)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb208-77"><a href="#cb208-77" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quad_prec){</span>
<span id="cb208-78"><a href="#cb208-78" aria-hidden="true" tabindex="-1"></a>    densityd <span class="ot">&lt;-</span> <span class="fu">mpfr</span>(densityd,<span class="at">precBits =</span> <span class="dv">128</span>)</span>
<span id="cb208-79"><a href="#cb208-79" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb208-80"><a href="#cb208-80" aria-hidden="true" tabindex="-1"></a>  new <span class="ot">&lt;-</span> (densityd<span class="sc">/</span>prior)<span class="sc">/</span><span class="fu">sum</span>(densityd<span class="sc">/</span>prior)</span>
<span id="cb208-81"><a href="#cb208-81" aria-hidden="true" tabindex="-1"></a>  new <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(new)</span>
<span id="cb208-82"><a href="#cb208-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(new)</span>
<span id="cb208-83"><a href="#cb208-83" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-84"><a href="#cb208-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-85"><a href="#cb208-85" aria-hidden="true" tabindex="-1"></a>sum_over_loci <span class="ot">&lt;-</span> <span class="cf">function</span>(df,parameter){</span>
<span id="cb208-86"><a href="#cb208-86" aria-hidden="true" tabindex="-1"></a>      <span class="co">#this function takes a data frame of probability densities for many loci</span></span>
<span id="cb208-87"><a href="#cb208-87" aria-hidden="true" tabindex="-1"></a>      <span class="co"># that have had the prior removed,</span></span>
<span id="cb208-88"><a href="#cb208-88" aria-hidden="true" tabindex="-1"></a>      <span class="co"># together with with a logged prior named "logPrior",</span></span>
<span id="cb208-89"><a href="#cb208-89" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  as well as the name of a parameter (e.g. "Theta_1")</span></span>
<span id="cb208-90"><a href="#cb208-90" aria-hidden="true" tabindex="-1"></a>      <span class="co"># and sums the densities over loci.</span></span>
<span id="cb208-91"><a href="#cb208-91" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Rmpfr package allows quadruple precision for </span></span>
<span id="cb208-92"><a href="#cb208-92" aria-hidden="true" tabindex="-1"></a>      <span class="co">#calcs on very small numbers.</span></span>
<span id="cb208-93"><a href="#cb208-93" aria-hidden="true" tabindex="-1"></a>    <span class="fu">require</span>(Rmpfr)</span>
<span id="cb208-94"><a href="#cb208-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-95"><a href="#cb208-95" aria-hidden="true" tabindex="-1"></a>    <span class="co">#add a teeny-tiny amount to all values to avoid zeros</span></span>
<span id="cb208-96"><a href="#cb208-96" aria-hidden="true" tabindex="-1"></a>    df2 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(parameter), </span>
<span id="cb208-97"><a href="#cb208-97" aria-hidden="true" tabindex="-1"></a>                        <span class="at">.fns=</span> <span class="cf">function</span>(x) x <span class="sc">+</span> <span class="fl">1e-11</span>))  <span class="sc">%&gt;%</span> </span>
<span id="cb208-98"><a href="#cb208-98" aria-hidden="true" tabindex="-1"></a>      <span class="co">#log all values</span></span>
<span id="cb208-99"><a href="#cb208-99" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(parameter)),</span>
<span id="cb208-100"><a href="#cb208-100" aria-hidden="true" tabindex="-1"></a>                  <span class="at">.fns=</span>log)) <span class="sc">%&gt;%</span> </span>
<span id="cb208-101"><a href="#cb208-101" aria-hidden="true" tabindex="-1"></a>      <span class="co"># convert the df to rowwise so that rows can be summed</span></span>
<span id="cb208-102"><a href="#cb208-102" aria-hidden="true" tabindex="-1"></a>      <span class="co"># and then sum across the row, including the prior</span></span>
<span id="cb208-103"><a href="#cb208-103" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb208-104"><a href="#cb208-104" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">sum_param_prior =</span> </span>
<span id="cb208-105"><a href="#cb208-105" aria-hidden="true" tabindex="-1"></a>          <span class="fu">sum</span>(<span class="fu">c_across</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(parameter,<span class="st">"logPrior"</span>))))) <span class="sc">%&gt;%</span> </span>
<span id="cb208-106"><a href="#cb208-106" aria-hidden="true" tabindex="-1"></a>    <span class="co">#convert back to a regular df</span></span>
<span id="cb208-107"><a href="#cb208-107" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ungroup</span>()</span>
<span id="cb208-108"><a href="#cb208-108" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb208-109"><a href="#cb208-109" aria-hidden="true" tabindex="-1"></a>    <span class="co">#need to convert to quadruple precision because </span></span>
<span id="cb208-110"><a href="#cb208-110" aria-hidden="true" tabindex="-1"></a>    <span class="co">#these will exponentiate to very small numbers.</span></span>
<span id="cb208-111"><a href="#cb208-111" aria-hidden="true" tabindex="-1"></a>    sum_param_prior_exp <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="fu">mpfr</span>(df2<span class="sc">$</span>sum_param_prior, <span class="at">precBits =</span> <span class="dv">128</span>))</span>
<span id="cb208-112"><a href="#cb208-112" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardize by dividing by the sum of the column</span></span>
<span id="cb208-113"><a href="#cb208-113" aria-hidden="true" tabindex="-1"></a>    sum_param_prior_standardized <span class="ot">&lt;-</span></span>
<span id="cb208-114"><a href="#cb208-114" aria-hidden="true" tabindex="-1"></a>             sum_param_prior_exp<span class="sc">/</span><span class="fu">sum</span>(sum_param_prior_exp)</span>
<span id="cb208-115"><a href="#cb208-115" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb208-116"><a href="#cb208-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-117"><a href="#cb208-117" aria-hidden="true" tabindex="-1"></a>    <span class="co">#drop the intermediate columns (no longer needed), change the standardized</span></span>
<span id="cb208-118"><a href="#cb208-118" aria-hidden="true" tabindex="-1"></a>    <span class="co"># output back to double precision so that it can be incorporated into the df</span></span>
<span id="cb208-119"><a href="#cb208-119" aria-hidden="true" tabindex="-1"></a>    <span class="co"># rename the summed column after the parameter</span></span>
<span id="cb208-120"><a href="#cb208-120" aria-hidden="true" tabindex="-1"></a>      df3 <span class="ot">&lt;-</span> df2 <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(sum_param_prior)) <span class="sc">%&gt;%</span></span>
<span id="cb208-121"><a href="#cb208-121" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">summed_over_loci =</span></span>
<span id="cb208-122"><a href="#cb208-122" aria-hidden="true" tabindex="-1"></a>          <span class="fu">as.numeric</span>(sum_param_prior_standardized)) <span class="sc">%&gt;%</span> </span>
<span id="cb208-123"><a href="#cb208-123" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rename_with</span>(<span class="at">.fn =</span> <span class="sc">~</span> <span class="fu">paste</span>(parameter), </span>
<span id="cb208-124"><a href="#cb208-124" aria-hidden="true" tabindex="-1"></a>                      <span class="at">.cols =</span> summed_over_loci)</span>
<span id="cb208-125"><a href="#cb208-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df3)</span>
<span id="cb208-126"><a href="#cb208-126" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-127"><a href="#cb208-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-128"><a href="#cb208-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-129"><a href="#cb208-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-130"><a href="#cb208-130" aria-hidden="true" tabindex="-1"></a>summarize_posterior <span class="ot">&lt;-</span> <span class="cf">function</span>(posterior, </span>
<span id="cb208-131"><a href="#cb208-131" aria-hidden="true" tabindex="-1"></a>                  <span class="at">parameter_type =</span> <span class="fu">c</span>(<span class="st">"Theta"</span>,<span class="st">"M"</span>,<span class="st">"Nm"</span>,<span class="st">"D"</span>),</span>
<span id="cb208-132"><a href="#cb208-132" aria-hidden="true" tabindex="-1"></a>                  prior,</span>
<span id="cb208-133"><a href="#cb208-133" aria-hidden="true" tabindex="-1"></a>                  <span class="at">n=</span><span class="dv">16384</span>,</span>
<span id="cb208-134"><a href="#cb208-134" aria-hidden="true" tabindex="-1"></a>                  ...){</span>
<span id="cb208-135"><a href="#cb208-135" aria-hidden="true" tabindex="-1"></a>  <span class="co"># this function takes a Migrate-n posterior "bayesallfile" as a dataframe</span></span>
<span id="cb208-136"><a href="#cb208-136" aria-hidden="true" tabindex="-1"></a>  <span class="co"># as well as one of the parameter types, and the prior on that parameter</span></span>
<span id="cb208-137"><a href="#cb208-137" aria-hidden="true" tabindex="-1"></a>  <span class="co"># as a tibble of x and y values. </span></span>
<span id="cb208-138"><a href="#cb208-138" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Currently only exponential priors supported</span></span>
<span id="cb208-139"><a href="#cb208-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># it will create densities for each parameter of the given type,</span></span>
<span id="cb208-140"><a href="#cb208-140" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove the prior from each, sum across loci, and re-add the prior (once)</span></span>
<span id="cb208-141"><a href="#cb208-141" aria-hidden="true" tabindex="-1"></a>  parameters <span class="ot">&lt;-</span> <span class="fu">names</span>(posterior) <span class="sc">%&gt;%</span></span>
<span id="cb208-142"><a href="#cb208-142" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">str_subset</span>(parameter_type)</span>
<span id="cb208-143"><a href="#cb208-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-144"><a href="#cb208-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a tibble with x values for a density plot</span></span>
<span id="cb208-145"><a href="#cb208-145" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  of the chosen number of points</span></span>
<span id="cb208-146"><a href="#cb208-146" aria-hidden="true" tabindex="-1"></a>  dens <span class="ot">&lt;-</span> prior</span>
<span id="cb208-147"><a href="#cb208-147" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-148"><a href="#cb208-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"calculating densities"</span>)</span>
<span id="cb208-149"><a href="#cb208-149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># calculate densities for each parameter of a given type at each locus</span></span>
<span id="cb208-150"><a href="#cb208-150" aria-hidden="true" tabindex="-1"></a>  dens <span class="ot">&lt;-</span> posterior <span class="sc">%&gt;%</span> </span>
<span id="cb208-151"><a href="#cb208-151" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">"StepsR"</span>,<span class="st">"Locus"</span>,</span>
<span id="cb208-152"><a href="#cb208-152" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">paste0</span>(parameter_type,<span class="st">"_"</span>)))) <span class="sc">%&gt;%</span> </span>
<span id="cb208-153"><a href="#cb208-153" aria-hidden="true" tabindex="-1"></a>          <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"Locus"</span>, <span class="at">values_from =</span> </span>
<span id="cb208-154"><a href="#cb208-154" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">starts_with</span>(<span class="fu">paste0</span>(parameter_type,<span class="st">"_"</span>)),</span>
<span id="cb208-155"><a href="#cb208-155" aria-hidden="true" tabindex="-1"></a>                          <span class="at">names_sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb208-156"><a href="#cb208-156" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="fu">paste0</span>(parameter_type,<span class="st">"_"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb208-157"><a href="#cb208-157" aria-hidden="true" tabindex="-1"></a>          <span class="fu">map_dfc</span>(<span class="cf">function</span>(x) <span class="fu">density</span>(x, <span class="at">n =</span> n, <span class="at">from =</span> <span class="dv">0</span>,</span>
<span id="cb208-158"><a href="#cb208-158" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">bw =</span> <span class="st">"nrd0"</span>)<span class="sc">$</span>y) <span class="sc">%&gt;%</span></span>
<span id="cb208-159"><a href="#cb208-159" aria-hidden="true" tabindex="-1"></a>          <span class="fu">bind_cols</span>(dens)</span>
<span id="cb208-160"><a href="#cb208-160" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-161"><a href="#cb208-161" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create, standardize, log and remove prior</span></span>
<span id="cb208-162"><a href="#cb208-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-163"><a href="#cb208-163" aria-hidden="true" tabindex="-1"></a>  <span class="co">#check for NA values</span></span>
<span id="cb208-164"><a href="#cb208-164" aria-hidden="true" tabindex="-1"></a>  <span class="co">#df %&gt;% </span></span>
<span id="cb208-165"><a href="#cb208-165" aria-hidden="true" tabindex="-1"></a>  <span class="co">#select_if(function(x) any(is.na(x))) %&gt;% </span></span>
<span id="cb208-166"><a href="#cb208-166" aria-hidden="true" tabindex="-1"></a>  <span class="co">#summarise_each(funs(sum(is.na(.)))) -&gt; extra_NA</span></span>
<span id="cb208-167"><a href="#cb208-167" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-168"><a href="#cb208-168" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb208-169"><a href="#cb208-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">"removing prior"</span>)</span>
<span id="cb208-170"><a href="#cb208-170" aria-hidden="true" tabindex="-1"></a>  dens2 <span class="ot">&lt;-</span> dens <span class="sc">%&gt;%</span> </span>
<span id="cb208-171"><a href="#cb208-171" aria-hidden="true" tabindex="-1"></a>        <span class="co">#remove the prior, standardize</span></span>
<span id="cb208-172"><a href="#cb208-172" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(parameter_type), </span>
<span id="cb208-173"><a href="#cb208-173" aria-hidden="true" tabindex="-1"></a>                  <span class="sc">~</span> <span class="fu">remove_prior</span>(<span class="at">densityd =</span> .x,</span>
<span id="cb208-174"><a href="#cb208-174" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">prior =</span> dens<span class="sc">$</span>prior,</span>
<span id="cb208-175"><a href="#cb208-175" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">threshold =</span> <span class="fl">1e-10</span>,</span>
<span id="cb208-176"><a href="#cb208-176" aria-hidden="true" tabindex="-1"></a>                                  ...) ))</span>
<span id="cb208-177"><a href="#cb208-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-178"><a href="#cb208-178" aria-hidden="true" tabindex="-1"></a>  dens3 <span class="ot">&lt;-</span> dens2</span>
<span id="cb208-179"><a href="#cb208-179" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb208-180"><a href="#cb208-180" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(p <span class="cf">in</span> parameters){</span>
<span id="cb208-181"><a href="#cb208-181" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(p)  </span>
<span id="cb208-182"><a href="#cb208-182" aria-hidden="true" tabindex="-1"></a>    dens3 <span class="ot">&lt;-</span> <span class="fu">sum_over_loci</span>(<span class="at">df =</span> dens3, <span class="at">parameter =</span> p)</span>
<span id="cb208-183"><a href="#cb208-183" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb208-184"><a href="#cb208-184" aria-hidden="true" tabindex="-1"></a>  <span class="co"># trying to do the above loop with purrr:map_dfc</span></span>
<span id="cb208-185"><a href="#cb208-185" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dens4 &lt;- parameters %&gt;% </span></span>
<span id="cb208-186"><a href="#cb208-186" aria-hidden="true" tabindex="-1"></a>  <span class="co">#        map_dfc(.f = ~ sum_over_loci(df = dens2, parameter = .x))</span></span>
<span id="cb208-187"><a href="#cb208-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dens3)</span>
<span id="cb208-188"><a href="#cb208-188" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-189"><a href="#cb208-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-190"><a href="#cb208-190" aria-hidden="true" tabindex="-1"></a>posterior_stats <span class="ot">&lt;-</span> <span class="cf">function</span>(df,parameter){</span>
<span id="cb208-191"><a href="#cb208-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(spatstat.core)</span>
<span id="cb208-192"><a href="#cb208-192" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"x"</span>,parameter))) <span class="sc">%&gt;%</span> <span class="fu">as.list</span>(<span class="at">bw =</span> <span class="dv">1</span>)</span>
<span id="cb208-193"><a href="#cb208-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(p) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"x"</span>, <span class="st">"y"</span>)</span>
<span id="cb208-194"><a href="#cb208-194" aria-hidden="true" tabindex="-1"></a>  p<span class="sc">$</span>bw <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb208-195"><a href="#cb208-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">attr</span>(p, <span class="st">"class"</span>) <span class="ot">&lt;-</span> <span class="st">"density"</span></span>
<span id="cb208-196"><a href="#cb208-196" aria-hidden="true" tabindex="-1"></a>  qu <span class="ot">&lt;-</span> <span class="fu">quantile.density</span>(p, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>, <span class="fl">0.975</span>))</span>
<span id="cb208-197"><a href="#cb208-197" aria-hidden="true" tabindex="-1"></a>  wmo <span class="ot">&lt;-</span> p<span class="sc">$</span>x[<span class="fu">which</span>(p<span class="sc">$</span>y<span class="sc">==</span><span class="fu">max</span>(p<span class="sc">$</span>y))]</span>
<span id="cb208-198"><a href="#cb208-198" aria-hidden="true" tabindex="-1"></a>  wme <span class="ot">&lt;-</span> <span class="fu">weighted.mean</span>(p<span class="sc">$</span>x, p<span class="sc">$</span>y)</span>
<span id="cb208-199"><a href="#cb208-199" aria-hidden="true" tabindex="-1"></a>  wsd <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">weighted.var</span>(p<span class="sc">$</span>x, p<span class="sc">$</span>y))</span>
<span id="cb208-200"><a href="#cb208-200" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">c</span>(qu,<span class="at">mode =</span> wmo, <span class="at">mean =</span> wme, <span class="at">sd =</span> wsd)</span>
<span id="cb208-201"><a href="#cb208-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats)</span>
<span id="cb208-202"><a href="#cb208-202" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="model-marginal-likelihoods" class="level4">
<h4 class="anchored" data-anchor-id="model-marginal-likelihoods">Model Marginal Likelihoods</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a>runDir <span class="ot">&lt;-</span> <span class="st">"./migrate/output2"</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a>likelist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(r <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>){</span>
<span id="cb209-5"><a href="#cb209-5" aria-hidden="true" tabindex="-1"></a>  rep <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"rep"</span>,r)</span>
<span id="cb209-6"><a href="#cb209-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(rep)</span>
<span id="cb209-7"><a href="#cb209-7" aria-hidden="true" tabindex="-1"></a>  likelist[[rep]] <span class="ot">&lt;-</span> <span class="fu">harvest.model.likelihoods</span>(<span class="at">workingDir=</span>                                             <span class="fu">file.path</span>(runDir,rep))</span>
<span id="cb209-8"><a href="#cb209-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb209-9"><a href="#cb209-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-10"><a href="#cb209-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Model selection for each replicate...</span></span>
<span id="cb209-11"><a href="#cb209-11" aria-hidden="true" tabindex="-1"></a>likelist <span class="sc">%&gt;%</span> <span class="fu">map</span>(bfcalcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The final model marginal likelihood estimates based on the mean of 10 replicates. The best model is West to East divergence with migration as specified by Treemix (a single migration event from Societies to New Caledonia)!</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a>like.df <span class="ot">&lt;-</span>  likelist <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(model)</span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a>means <span class="ot">&lt;-</span> like.df <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">bezier.corrected =</span> <span class="fu">mean</span>(bezier.corrected))</span>
<span id="cb210-4"><a href="#cb210-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb210-5"><a href="#cb210-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-6"><a href="#cb210-6" aria-hidden="true" tabindex="-1"></a>final_model <span class="ot">&lt;-</span> <span class="fu">bfcalcs</span>(means)</span>
<span id="cb210-7"><a href="#cb210-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-8"><a href="#cb210-8" aria-hidden="true" tabindex="-1"></a>final_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["model"],"name":[1],"type":["chr"],"align":["left"]},{"label":["bezier.corrected"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["lbf"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["choice"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["modelprob"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"2way.SS_Equilibrium","2":"-40148607","3":"-1111726.48","4":"9","5":"0"},{"1":"EtoW_DivMig","2":"-39728954","3":"-272420.61","4":"5","5":"0"},{"1":"EtoW_DivOnly","2":"-39648958","3":"-112428.41","4":"4","5":"0"},{"1":"EtoW_treemix","2":"-39632763","3":"-80039.79","4":"2","5":"0"},{"1":"EtoW.SS_Equilibrium","2":"-39963458","3":"-741428.37","4":"7","5":"0"},{"1":"n-Island","2":"-40239658","3":"-1293829.65","4":"10","5":"0"},{"1":"Panmixia","2":"-60090371","3":"-40995255.92","4":"11","5":"0"},{"1":"WtoE_DivMig","2":"-39758536","3":"-331584.48","4":"6","5":"0"},{"1":"WtoE_DivOnly","2":"-39646758","3":"-108028.60","4":"3","5":"0"},{"1":"WtoE_treemix","2":"-39592743","3":"0.00","4":"1","5":"1"},{"1":"WtoE.SS_Equilibrium","2":"-39973016","3":"-760545.54","4":"8","5":"0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<section id="t-test" class="level5">
<h5 class="anchored" data-anchor-id="t-test">T-Test</h5>
<p>Difference between West to East Treemix and the second-place East to West Treemix model is not significant, due to fairly large variability in marginal likelhoods. Need to consider this in the discussion.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a>top.choice <span class="ot">&lt;-</span> final_model<span class="sc">$</span>model[<span class="fu">which</span>(final_model<span class="sc">$</span>choice <span class="sc">==</span><span class="dv">1</span>)]</span>
<span id="cb211-2"><a href="#cb211-2" aria-hidden="true" tabindex="-1"></a>second.choice <span class="ot">&lt;-</span> final_model<span class="sc">$</span>model[<span class="fu">which</span>(final_model<span class="sc">$</span>choice <span class="sc">==</span><span class="dv">2</span>)]</span>
<span id="cb211-3"><a href="#cb211-3" aria-hidden="true" tabindex="-1"></a>third.choice <span class="ot">&lt;-</span> final_model<span class="sc">$</span>model[<span class="fu">which</span>(final_model<span class="sc">$</span>choice <span class="sc">==</span><span class="dv">3</span>)]</span>
<span id="cb211-4"><a href="#cb211-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-5"><a href="#cb211-5" aria-hidden="true" tabindex="-1"></a><span class="fu">permTS</span>(like.df<span class="sc">$</span>bezier.corrected[<span class="fu">which</span>(like.df<span class="sc">$</span>model <span class="sc">==</span> top.choice)],</span>
<span id="cb211-6"><a href="#cb211-6" aria-hidden="true" tabindex="-1"></a>       like.df<span class="sc">$</span>bezier.corrected[<span class="fu">which</span>(like.df<span class="sc">$</span>model <span class="sc">==</span> second.choice)],</span>
<span id="cb211-7"><a href="#cb211-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">alternative =</span> <span class="st">"greater"</span>, <span class="at">method =</span> <span class="st">"exact.ce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Exact Permutation Test (complete enumeration)

data:  GROUP 1 and GROUP 2
p-value = 0.1183
alternative hypothesis: true mean GROUP 1 - mean GROUP 2 is greater than 0
sample estimates:
mean GROUP 1 - mean GROUP 2 
                    40019.9 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">permTS</span>(like.df<span class="sc">$</span>bezier.corrected[<span class="fu">which</span>(like.df<span class="sc">$</span>model <span class="sc">==</span> top.choice)],</span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a>       like.df<span class="sc">$</span>bezier.corrected[<span class="fu">which</span>(like.df<span class="sc">$</span>model <span class="sc">==</span> third.choice)],</span>
<span id="cb213-3"><a href="#cb213-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">alternative =</span> <span class="st">"greater"</span>, <span class="at">method =</span> <span class="st">"exact.ce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Exact Permutation Test (complete enumeration)

data:  GROUP 1 and GROUP 2
p-value = 0.09168
alternative hypothesis: true mean GROUP 1 - mean GROUP 2 is greater than 0
sample estimates:
mean GROUP 1 - mean GROUP 2 
                    54014.3 </code></pre>
</div>
</div>
</section>
<section id="model-selection-figures" class="level5">
<h5 class="anchored" data-anchor-id="model-selection-figures">Model Selection Figures</h5>
<p>And a figure summarizing all this</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a>models <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"2-way Stepping-Stone Equilibrium"</span>,</span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">"E to W Divergence with Migration"</span>,</span>
<span id="cb215-3"><a href="#cb215-3" aria-hidden="true" tabindex="-1"></a>            <span class="st">"E to W Divergence Only"</span>,</span>
<span id="cb215-4"><a href="#cb215-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">"E to W Treemix"</span>, </span>
<span id="cb215-5"><a href="#cb215-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">"E to W Stepping-Stone Equilibrium"</span>,</span>
<span id="cb215-6"><a href="#cb215-6" aria-hidden="true" tabindex="-1"></a>            <span class="st">"n-Island"</span>,</span>
<span id="cb215-7"><a href="#cb215-7" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Panmixia"</span>,</span>
<span id="cb215-8"><a href="#cb215-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">"W to E Divergence with Migration"</span>,</span>
<span id="cb215-9"><a href="#cb215-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">"W to E Divergence Only"</span>,</span>
<span id="cb215-10"><a href="#cb215-10" aria-hidden="true" tabindex="-1"></a>            <span class="st">"W to E Treemix"</span>, </span>
<span id="cb215-11"><a href="#cb215-11" aria-hidden="true" tabindex="-1"></a>            <span class="st">"W to E Stepping-Stone Equilibrium"</span>)</span>
<span id="cb215-12"><a href="#cb215-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-13"><a href="#cb215-13" aria-hidden="true" tabindex="-1"></a>likesPlot <span class="ot">&lt;-</span> likelist <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(model) <span class="sc">%&gt;%</span> </span>
<span id="cb215-14"><a href="#cb215-14" aria-hidden="true" tabindex="-1"></a>              <span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>model, <span class="at">y =</span> bezier.corrected)) <span class="sc">+</span></span>
<span id="cb215-15"><a href="#cb215-15" aria-hidden="true" tabindex="-1"></a>              <span class="fu">geom_violin</span>(<span class="at">draw_quantiles =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb215-16"><a href="#cb215-16" aria-hidden="true" tabindex="-1"></a>              <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb215-17"><a href="#cb215-17" aria-hidden="true" tabindex="-1"></a>              <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> models) <span class="sc">+</span></span>
<span id="cb215-18"><a href="#cb215-18" aria-hidden="true" tabindex="-1"></a>              <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Metapopulation Model"</span>, <span class="at">y =</span> <span class="st">"Bezier Corrected Marginal Likelihood"</span>) <span class="sc">+</span></span>
<span id="cb215-19"><a href="#cb215-19" aria-hidden="true" tabindex="-1"></a>              <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>))</span>
<span id="cb215-20"><a href="#cb215-20" aria-hidden="true" tabindex="-1"></a>likesPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-123-1.png" class="img-fluid" width="960"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a>likesPlot2 <span class="ot">&lt;-</span> likesPlot <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">4.05e7</span>, <span class="sc">-</span><span class="fl">3.94e7</span>)</span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>likesPlot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Daruanus_IBD_migrate_files/figure-html/unnamed-chunk-123-2.png" class="img-fluid" width="960"></p>
</div>
</div>
</section>
</section>
<section id="parameter-estimates" class="level4">
<h4 class="anchored" data-anchor-id="parameter-estimates">Parameter Estimates</h4>
<section id="priors" class="level5">
<h5 class="anchored" data-anchor-id="priors">Priors</h5>
<p>Here are the priors Iâ€™m using for models, as a reminder.</p>
<pre><code>bayes-priors= THETA * * WEXPPRIOR: 0.0 0.1 1.000000 0.1000 
bayes-priors= MIG * * UNIFORMPRIOR: 0 300 30
divergence_distrib = E
bayes-priors= SPLIT * * WEXPPRIOR: 0.0 0.01 1.000000 0.01000
bayes-priors= SPLITSTD * * WEXPPRIOR: 0.0 0.01 1.000000 0.01000

</code></pre>



</section>
</section>
</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-doNeEstimatorV2Reimplementation2014" class="csl-entry" role="doc-biblioentry">
Do, C., R. S. Waples, D. Peel, G. M. Macbeth, B. J. Tillett, and J. R. Ovenden. 2014. <span>â€œ<span>NeEstimator</span> V2: Re-Implementation of Software for the Estimation of Contemporary Effective Population Size from Genetic Data.â€</span> <em>Molecular Ecology Resources</em> 14 (1): 209â€“14. <a href="https://doi.org/10.1111/1755-0998.12157">https://doi.org/10.1111/1755-0998.12157</a>.
</div>
<div id="ref-evannoDetectingNumberClusters2005" class="csl-entry" role="doc-biblioentry">
Evanno, G., S. Regnaut, and J. Goudet. 2005. <span>â€œDetecting the Number of Clusters of Individuals Using the Software <span>STRUCTURE</span>: A Simulation Study.â€</span> <em>Molecular Ecology</em> 14 (8): 2611â€“20. <a href="https://doi.org/10.1111/j.1365-294X.2005.02553.x">https://doi.org/10.1111/j.1365-294X.2005.02553.x</a>.
</div>
<div id="ref-fitakOptMEstimatingOptimal2021" class="csl-entry" role="doc-biblioentry">
Fitak, Robert R. 2021. <span>â€œ<span><em>OptM</em></span> : Estimating the Optimal Number of Migration Edges on Population Trees Using <span><em>Treemix</em></span>.â€</span> <em>Biology Methods and Protocols</em> 6 (1): bpab017. <a href="https://doi.org/10.1093/biomethods/bpab017">https://doi.org/10.1093/biomethods/bpab017</a>.
</div>
<div id="ref-jonesImprovedConfidenceIntervals2016" class="csl-entry" role="doc-biblioentry">
Jones, A. T., J. R. Ovenden, and Y.-G. Wang. 2016. <span>â€œImproved Confidence Intervals for the Linkage Disequilibrium Method for Estimating Effective Population Size.â€</span> <em>Heredity</em> 117 (4): 217â€“23. <a href="https://doi.org/10.1038/hdy.2016.19">https://doi.org/10.1038/hdy.2016.19</a>.
</div>
<div id="ref-lebloisMigraiNeManual2020" class="csl-entry" role="doc-biblioentry">
Leblois, RaphaÃ«l, and FranÃ§ois Rousset. 2020. <span>â€œ<span>MigraiNe</span> 0.6 <span>Manual</span>.â€</span>
</div>
<div id="ref-naaykensIsolationDistanceGenetic2022" class="csl-entry" role="doc-biblioentry">
Naaykens, Taylor, and Cassidy C. Dâ€™Aloia. 2022. <span>â€œIsolation-by-Distance and Genetic Parentage Analysis Provide Similar Larval Dispersal Estimates.â€</span> <em>Molecular Ecology</em> 31 (11): 3072â€“82. <a href="https://doi.org/10.1111/mec.16465">https://doi.org/10.1111/mec.16465</a>.
</div>
<div id="ref-neelEstimationEffectivePopulation2013" class="csl-entry" role="doc-biblioentry">
Neel, M. C., K. McKelvey, N. Ryman, M. W. Lloyd, R. Short Bull, F. W. Allendorf, M. K. Schwartz, and R. S. Waples. 2013. <span>â€œEstimation of Effective Population Size in Continuously Distributed Populations: There Goes the Neighborhood.â€</span> <em>Heredity</em>, May, 1â€“11. <a href="https://doi.org/10.1038/hdy.2013.37">https://doi.org/10.1038/hdy.2013.37</a>.
</div>
<div id="ref-pinskyUsingIsolationDistance2010" class="csl-entry" role="doc-biblioentry">
Pinsky, Malin L., Humberto R. Montes, and Stephen R. Palumbi. 2010. <span>â€œUsing Isolation by Distance and Effective Density to Estimate Dispersal Scales in Anemonefish.â€</span> <em>Evolution</em> 64 (9): 2688â€“2700. <a href="https://doi.org/czskhg">https://doi.org/czskhg</a>.
</div>
<div id="ref-pinskyMarineDispersalScales2017" class="csl-entry" role="doc-biblioentry">
Pinsky, Malin L., Pablo Saenz-Agudelo, OcÃ©ane C. Salles, Glenn R. Almany, Michael Bode, Michael L. Berumen, Serge AndrÃ©fouÃ«t, Simon R. Thorrold, Geoffrey P. Jones, and Serge Planes. 2017. <span>â€œMarine <span>Dispersal Scales Are Congruent</span> over <span>Evolutionary</span> and <span>Ecological Time</span>.â€</span> <em>Current Biology : CB</em>, December, 1â€“16. <a href="https://doi.org/10.1016/j.cub.2016.10.053">https://doi.org/10.1016/j.cub.2016.10.053</a>.
</div>
<div id="ref-roussetGeneticDifferentiationEstimation1997" class="csl-entry" role="doc-biblioentry">
Rousset, F. 1997. <span>â€œ<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1207888">Genetic Differentiation and Estimation of Gene Flow from <span class="nocase">F-statistics</span> Under Isolation by Distance.</a>â€</span> <em>Genetics</em> 145 (4): 1219â€“28.
</div>
<div id="ref-waplesTestingHardyWeinberg2014" class="csl-entry" role="doc-biblioentry">
Waples, Robin S. 2014. <span>â€œTesting for <span>Hardy</span>: <span>Have We Lost</span> the <span>Plot</span>?â€</span> <em>Journal Of Heredity</em> 106 (1): 1â€“19. <a href="https://doi.org/10.1093/jhered/esu062">https://doi.org/10.1093/jhered/esu062</a>.
</div>
<div id="ref-waplesEffectsOverlappingGenerations2014" class="csl-entry" role="doc-biblioentry">
Waples, Robin S., Tiago Antao, and Gordon Luikart. 2014. <span>â€œEffects of Overlapping Generations on Linkage Disequilibrium Estimates of Effective Population Size.â€</span> <em>Genetics</em> 197 (2): 769â€“80. <a href="https://doi.org/10.1534/genetics.114.164822">https://doi.org/10.1534/genetics.114.164822</a>.
</div>
<div id="ref-waplesLinkageDisequilibriumEstimates2010" class="csl-entry" role="doc-biblioentry">
Waples, Robin S., and Chi Do. 2010. <span>â€œLinkage Disequilibrium Estimates of Contemporary <span class="nocase">N-e</span> Using Highly Variable Genetic Markers: A Largely Untapped Resource for Applied Conservation and Evolution.â€</span> <em>Evolutionary Applications</em> 3 (3): 244â€“62. <a href="https://doi.org/10.1111/j.1752-4571.2009.00104.x">https://doi.org/10.1111/j.1752-4571.2009.00104.x</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>